{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style=\"color:#4682B4\">NCBI Taxonomy Data Profile</span>\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The [NCBI Taxonomy](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi) is a reference database that contains the names and hierarchically-arranged phylogenetic classifications of organisms created in 1991. The NCBI Taxonomy database is organized in a tree for a hierarchical data structure, where each <span style=\"color:#4682B4\">**_node_**</span> of the tree represents a <span style=\"color:#4682B4\">**_taxon_**</span> and each entry has a primary name, secondary names, and a unique taxonomic identifier. The NCBI Taxonomy database is critical to link nucleotides and protein sequences from the International Nucleotide Sequence Database Collaboration (INSDC) and other biological databases which rely on data from INSDC. These linkages can be made using either the organism name or taxonomic ID.\n",
    "\n",
    "The database is provided remotely by the National Center for Biotechnology Information (NCBI). INSDC partners send any requests for new names to NCBI Taxonomy curators before data is released. The database incorporates phylogenetic and taxonomic knowledge from published literature, web-based databases, and the advice of sequence submitters and outside taxonomy experts."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <span style=\"color:#4682B4\">Why NCBI Taxonomy</span>\n",
    "We chose to use the NCBI Taxonomy database because it is the <span style=\"color:#4682B4\">**_sole source for taxonomic classification_**</span> for the INSDC and forms the backbone for many other resources at the NCBI. The NCBI Taxonomy database contains formal and informal organism names and classifications for every sequence in INSDC's datasets (more than 160,000 organisms); these associations between pathogen name and genetic and genomic data are foundational for public health intelligence efforts. The inclusion of informal names also allows us to link pathogens (and their corresponding genetic information) to case reports and other non-traditional data sources which may use names which exist outside of the codes of nomenclature (i.e. \"COVID-19\" instead of SARS-CoV-2). Additionally, more than 150 external partners maintain links to the NCBI Taxonomy database, with specialty datasets of their own."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <span style=\"color:#4682B4\">Accessing the NCBI Taxonomy database</span>\n",
    "\n",
    "There are three methods for accessing the NCBI Taxonomy. First is the NCBI Taxonomy Browser, which is a web-page that allows users to search for organisms, visualize the hierarchy at custom levels of classification, and summarize organism information, such as lineage, in a taxon-specific page. \n",
    "\n",
    "In contrast to the Taxonomy Browser, Entrez supports Boolean queries and common search fields across all NCBI databases. There are also several public APIs that allow programmatic access to the Entrez databases; <span style=\"color:#4682B4\">**_we used E-utilities,_**</span> a suite of server-side programs that accept a fixed URL syntax for search, link, and retrieval. The available E-Utils include: EInfo, ESearch, EPost, ESumary, EFetch, ELink, EGQuery, ESpell, and ECitMatch. While many are tested in this notebook, full documentation is available [here](https://www.ncbi.nlm.nih.gov/books/NBK25500/).\n",
    "\n",
    "A third option is to download the complete database as a full text taxdump file (in .dmp format). While some externally developed packages, including Bio2RDF, use the data dump to formulate global SPARQL queries (transforming the relational database into RDF), it appears that these tools may be out of date. We chose not to utilize the taxdump file to facilitate access because it appears that version control is not well documented and the data may quickly become out of date. \n",
    "\n",
    "The API access via E-Utilities appears to be the most stable option for accessing NCBI data, as of current.\n",
    "\n",
    "This notebook examines the information stored within the NCBI Taxonomy and the quality of metadata that is retrievable via publicly available APIs.\n",
    "\n",
    "![](2022-05-05-14-44-58.png) \n",
    "\n",
    "NCBI Taxonomy Browser page\n",
    "\n",
    "![](2022-05-05-14-46-37.png)\n",
    "\n",
    "NCBI EFetch Utility in Entrez Taxonomy database"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <span style=\"color:gray\">_Extracting data from NCBI Taxonomy database_</span>\n",
    "\n",
    "There are 39 Entrez databases. To return a list of all Entrez database names and identify the one we want to query, we use the following:\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking python executable path\n",
      "/Users/haileyrobertson/Documents/GitHub/kr2-graph/build_graph_2/notebooks/env/bin/python\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>pre { line-height: 125%; }\n",
       "td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       ".highlight .hll { background-color: #ffffcc }\n",
       ".highlight { background: #f8f8f8; }\n",
       ".highlight .c { color: #3D7B7B; font-style: italic } /* Comment */\n",
       ".highlight .err { border: 1px solid #FF0000 } /* Error */\n",
       ".highlight .k { color: #008000; font-weight: bold } /* Keyword */\n",
       ".highlight .o { color: #666666 } /* Operator */\n",
       ".highlight .ch { color: #3D7B7B; font-style: italic } /* Comment.Hashbang */\n",
       ".highlight .cm { color: #3D7B7B; font-style: italic } /* Comment.Multiline */\n",
       ".highlight .cp { color: #9C6500 } /* Comment.Preproc */\n",
       ".highlight .cpf { color: #3D7B7B; font-style: italic } /* Comment.PreprocFile */\n",
       ".highlight .c1 { color: #3D7B7B; font-style: italic } /* Comment.Single */\n",
       ".highlight .cs { color: #3D7B7B; font-style: italic } /* Comment.Special */\n",
       ".highlight .gd { color: #A00000 } /* Generic.Deleted */\n",
       ".highlight .ge { font-style: italic } /* Generic.Emph */\n",
       ".highlight .gr { color: #E40000 } /* Generic.Error */\n",
       ".highlight .gh { color: #000080; font-weight: bold } /* Generic.Heading */\n",
       ".highlight .gi { color: #008400 } /* Generic.Inserted */\n",
       ".highlight .go { color: #717171 } /* Generic.Output */\n",
       ".highlight .gp { color: #000080; font-weight: bold } /* Generic.Prompt */\n",
       ".highlight .gs { font-weight: bold } /* Generic.Strong */\n",
       ".highlight .gu { color: #800080; font-weight: bold } /* Generic.Subheading */\n",
       ".highlight .gt { color: #0044DD } /* Generic.Traceback */\n",
       ".highlight .kc { color: #008000; font-weight: bold } /* Keyword.Constant */\n",
       ".highlight .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */\n",
       ".highlight .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */\n",
       ".highlight .kp { color: #008000 } /* Keyword.Pseudo */\n",
       ".highlight .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */\n",
       ".highlight .kt { color: #B00040 } /* Keyword.Type */\n",
       ".highlight .m { color: #666666 } /* Literal.Number */\n",
       ".highlight .s { color: #BA2121 } /* Literal.String */\n",
       ".highlight .na { color: #687822 } /* Name.Attribute */\n",
       ".highlight .nb { color: #008000 } /* Name.Builtin */\n",
       ".highlight .nc { color: #0000FF; font-weight: bold } /* Name.Class */\n",
       ".highlight .no { color: #880000 } /* Name.Constant */\n",
       ".highlight .nd { color: #AA22FF } /* Name.Decorator */\n",
       ".highlight .ni { color: #717171; font-weight: bold } /* Name.Entity */\n",
       ".highlight .ne { color: #CB3F38; font-weight: bold } /* Name.Exception */\n",
       ".highlight .nf { color: #0000FF } /* Name.Function */\n",
       ".highlight .nl { color: #767600 } /* Name.Label */\n",
       ".highlight .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */\n",
       ".highlight .nt { color: #008000; font-weight: bold } /* Name.Tag */\n",
       ".highlight .nv { color: #19177C } /* Name.Variable */\n",
       ".highlight .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */\n",
       ".highlight .w { color: #bbbbbb } /* Text.Whitespace */\n",
       ".highlight .mb { color: #666666 } /* Literal.Number.Bin */\n",
       ".highlight .mf { color: #666666 } /* Literal.Number.Float */\n",
       ".highlight .mh { color: #666666 } /* Literal.Number.Hex */\n",
       ".highlight .mi { color: #666666 } /* Literal.Number.Integer */\n",
       ".highlight .mo { color: #666666 } /* Literal.Number.Oct */\n",
       ".highlight .sa { color: #BA2121 } /* Literal.String.Affix */\n",
       ".highlight .sb { color: #BA2121 } /* Literal.String.Backtick */\n",
       ".highlight .sc { color: #BA2121 } /* Literal.String.Char */\n",
       ".highlight .dl { color: #BA2121 } /* Literal.String.Delimiter */\n",
       ".highlight .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */\n",
       ".highlight .s2 { color: #BA2121 } /* Literal.String.Double */\n",
       ".highlight .se { color: #AA5D1F; font-weight: bold } /* Literal.String.Escape */\n",
       ".highlight .sh { color: #BA2121 } /* Literal.String.Heredoc */\n",
       ".highlight .si { color: #A45A77; font-weight: bold } /* Literal.String.Interpol */\n",
       ".highlight .sx { color: #008000 } /* Literal.String.Other */\n",
       ".highlight .sr { color: #A45A77 } /* Literal.String.Regex */\n",
       ".highlight .s1 { color: #BA2121 } /* Literal.String.Single */\n",
       ".highlight .ss { color: #19177C } /* Literal.String.Symbol */\n",
       ".highlight .bp { color: #008000 } /* Name.Builtin.Pseudo */\n",
       ".highlight .fm { color: #0000FF } /* Name.Function.Magic */\n",
       ".highlight .vc { color: #19177C } /* Name.Variable.Class */\n",
       ".highlight .vg { color: #19177C } /* Name.Variable.Global */\n",
       ".highlight .vi { color: #19177C } /* Name.Variable.Instance */\n",
       ".highlight .vm { color: #19177C } /* Name.Variable.Magic */\n",
       ".highlight .il { color: #666666 } /* Literal.Number.Integer.Long */</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import sys\n",
    "\n",
    "print(\"Checking python executable path\")\n",
    "print(sys.executable)\n",
    "# append the path of the parent directory to access project packages\n",
    "sys.path.append(\"..\")\n",
    "\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# import project packages\n",
    "import ncbi\n",
    "\n",
    "# Set up notebook code formatting:\n",
    "import inspect\n",
    "from IPython.display import display, Code\n",
    "from pygments.formatters import HtmlFormatter\n",
    "from IPython.core.display import HTML\n",
    "formatter = HtmlFormatter()\n",
    "display(HTML(f'<style>{ formatter.get_style_defs(\".highlight\") }</style>'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "pubmed\n",
      "protein\n",
      "nuccore\n",
      "ipg\n",
      "nucleotide\n",
      "structure\n",
      "genome\n",
      "annotinfo\n",
      "assembly\n",
      "bioproject\n",
      "biosample\n",
      "blastdbinfo\n",
      "books\n",
      "cdd\n",
      "clinvar\n",
      "gap\n",
      "gapplus\n",
      "grasp\n",
      "dbvar\n",
      "gene\n",
      "gds\n",
      "geoprofiles\n",
      "homologene\n",
      "medgen\n",
      "mesh\n",
      "ncbisearch\n",
      "nlmcatalog\n",
      "omim\n",
      "orgtrack\n",
      "pmc\n",
      "popset\n",
      "proteinclusters\n",
      "pcassay\n",
      "protfam\n",
      "pccompound\n",
      "pcsubstance\n",
      "seqannot\n",
      "snp\n",
      "sra\n",
      "taxonomy\n",
      "biocollections\n",
      "gtr\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "entrez_url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi\"\n",
    "entrez_response = requests.get(entrez_url)\n",
    "all_db = BeautifulSoup(entrez_response.content, features=\"xml\").getText()\n",
    "print(all_db)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To gather statistics about the <span style=\"color:#4682B4\">**_taxonomy_**</span> database and look at the available fields, we specify the taxonomy URL using the EInfo utility:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Field Abbreviation</th>\n",
       "      <th>Full Field Name</th>\n",
       "      <th>Field Description</th>\n",
       "      <th>Term Count</th>\n",
       "      <th>Is Date</th>\n",
       "      <th>Is Numerical</th>\n",
       "      <th>Single Token</th>\n",
       "      <th>Hierarchy</th>\n",
       "      <th>Is Hidden</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALL</td>\n",
       "      <td>All Fields</td>\n",
       "      <td>All terms from all searchable fields</td>\n",
       "      <td>36780484</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UID</td>\n",
       "      <td>Taxonomy ID</td>\n",
       "      <td>Unique number assigned to publication</td>\n",
       "      <td>0</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FILT</td>\n",
       "      <td>Filter</td>\n",
       "      <td>Limits the records</td>\n",
       "      <td>303</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SCIN</td>\n",
       "      <td>Scientific Name</td>\n",
       "      <td>Scientific name of organism</td>\n",
       "      <td>2421008</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>COMN</td>\n",
       "      <td>Common Name</td>\n",
       "      <td>Common name of organism</td>\n",
       "      <td>47324</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TXSY</td>\n",
       "      <td>Synonym</td>\n",
       "      <td>Synonym of organism name</td>\n",
       "      <td>222019</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ALLN</td>\n",
       "      <td>All Names</td>\n",
       "      <td>All aliases for organism</td>\n",
       "      <td>3687515</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NXLV</td>\n",
       "      <td>Next Level</td>\n",
       "      <td>Immediate parent in taxonomic hierarchy</td>\n",
       "      <td>373744</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SBTR</td>\n",
       "      <td>Subtree</td>\n",
       "      <td>Any parent node in taxonomic hierarchy</td>\n",
       "      <td>6105938</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LNGE</td>\n",
       "      <td>Lineage</td>\n",
       "      <td>Lineage in taxonomic hierarchy</td>\n",
       "      <td>3687515</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GC</td>\n",
       "      <td>Nuclear Genetic Code</td>\n",
       "      <td>Nuclear genetic code</td>\n",
       "      <td>32</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MGC</td>\n",
       "      <td>Mitochondrial Genetic Code</td>\n",
       "      <td>Mitochondrial genetic code</td>\n",
       "      <td>30</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PGC</td>\n",
       "      <td>Plastid Genetic Code</td>\n",
       "      <td>Plastid genetic code</td>\n",
       "      <td>4</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TXDV</td>\n",
       "      <td>Division</td>\n",
       "      <td>GenBank division</td>\n",
       "      <td>24</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>RANK</td>\n",
       "      <td>Rank</td>\n",
       "      <td>Hierarchical position (e.g., order, genus)</td>\n",
       "      <td>43</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>EDAT</td>\n",
       "      <td>Entrez Date</td>\n",
       "      <td>Date record first accessible through Entrez</td>\n",
       "      <td>9086</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MDAT</td>\n",
       "      <td>Modification Date</td>\n",
       "      <td>Date of last update</td>\n",
       "      <td>6975</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PROP</td>\n",
       "      <td>Properties</td>\n",
       "      <td>Property defined on particular node (e.g., ter...</td>\n",
       "      <td>147</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>WORD</td>\n",
       "      <td>Text Word</td>\n",
       "      <td>Free text associated with record</td>\n",
       "      <td>28312189</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NTOK</td>\n",
       "      <td>Name Tokens</td>\n",
       "      <td>Name tokens associated with organism names</td>\n",
       "      <td>9745991</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HGC</td>\n",
       "      <td>Hydrogenosome Genetic Code</td>\n",
       "      <td>Hydrogenosome genetic code</td>\n",
       "      <td>4</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Field Abbreviation             Full Field Name  \\\n",
       "0                 ALL                  All Fields   \n",
       "1                 UID                 Taxonomy ID   \n",
       "2                FILT                      Filter   \n",
       "3                SCIN             Scientific Name   \n",
       "4                COMN                 Common Name   \n",
       "5                TXSY                     Synonym   \n",
       "6                ALLN                   All Names   \n",
       "7                NXLV                  Next Level   \n",
       "8                SBTR                     Subtree   \n",
       "9                LNGE                     Lineage   \n",
       "10                 GC        Nuclear Genetic Code   \n",
       "11                MGC  Mitochondrial Genetic Code   \n",
       "12                PGC        Plastid Genetic Code   \n",
       "13               TXDV                    Division   \n",
       "14               RANK                        Rank   \n",
       "15               EDAT                 Entrez Date   \n",
       "16               MDAT           Modification Date   \n",
       "17               PROP                  Properties   \n",
       "18               WORD                   Text Word   \n",
       "19               NTOK                 Name Tokens   \n",
       "20                HGC  Hydrogenosome Genetic Code   \n",
       "\n",
       "                                    Field Description Term Count Is Date  \\\n",
       "0                All terms from all searchable fields   36780484       N   \n",
       "1               Unique number assigned to publication          0       N   \n",
       "2                                  Limits the records        303       N   \n",
       "3                         Scientific name of organism    2421008       N   \n",
       "4                             Common name of organism      47324       N   \n",
       "5                            Synonym of organism name     222019       N   \n",
       "6                            All aliases for organism    3687515       N   \n",
       "7             Immediate parent in taxonomic hierarchy     373744       N   \n",
       "8              Any parent node in taxonomic hierarchy    6105938       N   \n",
       "9                      Lineage in taxonomic hierarchy    3687515       N   \n",
       "10                               Nuclear genetic code         32       N   \n",
       "11                         Mitochondrial genetic code         30       N   \n",
       "12                               Plastid genetic code          4       N   \n",
       "13                                   GenBank division         24       N   \n",
       "14         Hierarchical position (e.g., order, genus)         43       N   \n",
       "15        Date record first accessible through Entrez       9086       Y   \n",
       "16                                Date of last update       6975       Y   \n",
       "17  Property defined on particular node (e.g., ter...        147       N   \n",
       "18                   Free text associated with record   28312189       N   \n",
       "19         Name tokens associated with organism names    9745991       N   \n",
       "20                         Hydrogenosome genetic code          4       N   \n",
       "\n",
       "   Is Numerical Single Token Hierarchy Is Hidden  \n",
       "0             N            N         N         N  \n",
       "1             Y            Y         N         Y  \n",
       "2             N            Y         N         N  \n",
       "3             N            Y         N         N  \n",
       "4             N            Y         N         N  \n",
       "5             N            Y         N         N  \n",
       "6             N            Y         N         N  \n",
       "7             N            Y         N         N  \n",
       "8             N            Y         N         N  \n",
       "9             N            Y         N         N  \n",
       "10            N            Y         N         N  \n",
       "11            N            Y         N         N  \n",
       "12            N            Y         N         N  \n",
       "13            N            Y         N         N  \n",
       "14            N            Y         N         N  \n",
       "15            N            Y         N         N  \n",
       "16            N            Y         N         N  \n",
       "17            N            Y         N         N  \n",
       "18            N            Y         N         N  \n",
       "19            N            Y         N         N  \n",
       "20            N            Y         N         N  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "einfo_url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi?db=taxonomy\"\n",
    "einfo_response = requests.get(einfo_url)\n",
    "taxon_info = BeautifulSoup(einfo_response.content, features='xml')\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "# Extracting the field data\n",
    "field_tag = taxon_info.FieldList\n",
    "\n",
    "field = field_tag.find_all('Name')  \n",
    "full_name = field_tag.find_all('FullName')  \n",
    "desc = field_tag.find_all('Description')  \n",
    "term_count = field_tag.find_all('TermCount')\n",
    "is_date = field_tag.find_all('IsDate')\n",
    "is_numerical = field_tag.find_all('IsNumerical')\n",
    "single_token = field_tag.find_all('SingleToken')\n",
    "hierarchy = field_tag.find_all('Hierarchy')\n",
    "is_hidden = field_tag.find_all('IsHidden')\n",
    "\n",
    "field_data = []\n",
    "  \n",
    "# Loop to store the data in a list named 'field_data'\n",
    "for i in range(0, len(field)):\n",
    "    rows = [field[i].get_text(),full_name[i].get_text(),desc[i].get_text(),\n",
    "    term_count[i].get_text(), is_date[i].get_text(), is_numerical[i].get_text(), \n",
    "    single_token[i].get_text(),hierarchy[i].get_text(),is_hidden[i].get_text()]\n",
    "    field_data.append(rows)\n",
    "  \n",
    "# Converting the list into dataframe\n",
    "field_df = pd.DataFrame(field_data, columns=['Field Abbreviation','Full Field Name', 'Field Description',\n",
    "'Term Count', 'Is Date','Is Numerical','Single Token',\n",
    "'Hierarchy','Is Hidden'], dtype = str)\n",
    "\n",
    "display(field_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To learn about other Entrez databases associated with the <span style=\"color:#4682B4\">**_taxonomy_**</span> database entries, we call the EInfo utility again, this time specifying the LinkList:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Link Name</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>taxonomy_assembly_exp</td>\n",
       "      <td>Assembly records associated with taxonomy reco...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>taxonomy_bioproject_exp</td>\n",
       "      <td>BioProject records associated with taxonomy re...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>taxonomy_biosample_exp</td>\n",
       "      <td>BioSample records associated with taxonomy rec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>taxonomy_biosystems_exp</td>\n",
       "      <td>BioSystems records associated with taxonomy re...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>taxonomy_books</td>\n",
       "      <td>Books</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>taxonomy_cdd_exp</td>\n",
       "      <td>CDD records associated with taxonomy record (e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>taxonomy_clone_exp</td>\n",
       "      <td>Links to Clone DB (exploded for higher taxa)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>taxonomy_dbvar_exp</td>\n",
       "      <td>dbVar records associated with taxonomy record ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>taxonomy_gds_exp</td>\n",
       "      <td>GEO DataSet records associated with taxonomy r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>taxonomy_gene_exp</td>\n",
       "      <td>Gene records associated with taxonomy record (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>taxonomy_genome</td>\n",
       "      <td>Genome sequences associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>taxonomy_geoprofiles</td>\n",
       "      <td>GEO records associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>taxonomy_homologene_exp</td>\n",
       "      <td>HomoloGene records associated with taxonomy re...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>taxonomy_mesh</td>\n",
       "      <td>Related MeSH term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>taxonomy_nucest</td>\n",
       "      <td>EST sequences associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>taxonomy_nucgss</td>\n",
       "      <td>GSS sequences associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>taxonomy_nucleotide_exp</td>\n",
       "      <td>Nucleotide sequences associated with taxonomy ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>taxonomy_pcassay</td>\n",
       "      <td>Related PubChem BioAssay</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>taxonomy_pccompound</td>\n",
       "      <td>PubChem Compounds</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>taxonomy_pcsubstance</td>\n",
       "      <td>PubChem Substances</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>taxonomy_pmc</td>\n",
       "      <td>Free full text articles in PMC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>taxonomy_popset</td>\n",
       "      <td>Popset sequences associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>taxonomy_probe_exp</td>\n",
       "      <td>Probe records associated with taxonomy record ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>taxonomy_protein_exp</td>\n",
       "      <td>Protein sequences associated with taxonomy rec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>taxonomy_protein_species2wp</td>\n",
       "      <td>Autonomous proteins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>taxonomy_proteinclusters_exp</td>\n",
       "      <td>Protein Clusters records associated with taxon...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>taxonomy_pubmed</td>\n",
       "      <td>Link to related PubMed articles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>taxonomy_pubmed_entrez</td>\n",
       "      <td>Related PubMed articles computed using other E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>taxonomy_snp_exp</td>\n",
       "      <td>Snp records associated with taxonomy record (e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>taxonomy_sra_exp</td>\n",
       "      <td>Sra records associated with taxonomy record (e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>taxonomy_structure_exp</td>\n",
       "      <td>Structure sequences associated with taxonomy r...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Link Name  \\\n",
       "0          taxonomy_assembly_exp   \n",
       "1        taxonomy_bioproject_exp   \n",
       "2         taxonomy_biosample_exp   \n",
       "3        taxonomy_biosystems_exp   \n",
       "4                 taxonomy_books   \n",
       "5               taxonomy_cdd_exp   \n",
       "6             taxonomy_clone_exp   \n",
       "7             taxonomy_dbvar_exp   \n",
       "8               taxonomy_gds_exp   \n",
       "9              taxonomy_gene_exp   \n",
       "10               taxonomy_genome   \n",
       "11          taxonomy_geoprofiles   \n",
       "12       taxonomy_homologene_exp   \n",
       "13                 taxonomy_mesh   \n",
       "14               taxonomy_nucest   \n",
       "15               taxonomy_nucgss   \n",
       "16       taxonomy_nucleotide_exp   \n",
       "17              taxonomy_pcassay   \n",
       "18           taxonomy_pccompound   \n",
       "19          taxonomy_pcsubstance   \n",
       "20                  taxonomy_pmc   \n",
       "21               taxonomy_popset   \n",
       "22            taxonomy_probe_exp   \n",
       "23          taxonomy_protein_exp   \n",
       "24   taxonomy_protein_species2wp   \n",
       "25  taxonomy_proteinclusters_exp   \n",
       "26               taxonomy_pubmed   \n",
       "27        taxonomy_pubmed_entrez   \n",
       "28              taxonomy_snp_exp   \n",
       "29              taxonomy_sra_exp   \n",
       "30        taxonomy_structure_exp   \n",
       "\n",
       "                                          Description  \n",
       "0   Assembly records associated with taxonomy reco...  \n",
       "1   BioProject records associated with taxonomy re...  \n",
       "2   BioSample records associated with taxonomy rec...  \n",
       "3   BioSystems records associated with taxonomy re...  \n",
       "4                                               Books  \n",
       "5   CDD records associated with taxonomy record (e...  \n",
       "6        Links to Clone DB (exploded for higher taxa)  \n",
       "7   dbVar records associated with taxonomy record ...  \n",
       "8   GEO DataSet records associated with taxonomy r...  \n",
       "9   Gene records associated with taxonomy record (...  \n",
       "10   Genome sequences associated with taxonomy record  \n",
       "11        GEO records associated with taxonomy record  \n",
       "12  HomoloGene records associated with taxonomy re...  \n",
       "13                                  Related MeSH term  \n",
       "14      EST sequences associated with taxonomy record  \n",
       "15      GSS sequences associated with taxonomy record  \n",
       "16  Nucleotide sequences associated with taxonomy ...  \n",
       "17                           Related PubChem BioAssay  \n",
       "18                                  PubChem Compounds  \n",
       "19                                 PubChem Substances  \n",
       "20                     Free full text articles in PMC  \n",
       "21   Popset sequences associated with taxonomy record  \n",
       "22  Probe records associated with taxonomy record ...  \n",
       "23  Protein sequences associated with taxonomy rec...  \n",
       "24                                Autonomous proteins  \n",
       "25  Protein Clusters records associated with taxon...  \n",
       "26                    Link to related PubMed articles  \n",
       "27  Related PubMed articles computed using other E...  \n",
       "28  Snp records associated with taxonomy record (e...  \n",
       "29  Sra records associated with taxonomy record (e...  \n",
       "30  Structure sequences associated with taxonomy r...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Extracting the associated links\n",
    "link_tag = taxon_info.LinkList\n",
    "\n",
    "link_name = link_tag.find_all('Name')  \n",
    "link_desc = link_tag.find_all('Description')  \n",
    "\n",
    "link_data = []\n",
    "  \n",
    "# Loop to store the data in a list named 'link_data'\n",
    "for i in range(0, len(link_name)):\n",
    "    link_rows = [link_name[i].get_text(), link_desc[i].get_text()]\n",
    "    link_data.append(link_rows)\n",
    "  \n",
    "# Converting the list into dataframe\n",
    "link_df = pd.DataFrame(link_data, columns=['Link Name', 'Description'], dtype = str)\n",
    "\n",
    "display(link_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To extract data about organisms from these fields, we created a Python package <span style=\"color:#4682B4\">**_ncbi._**</span> This module is composed of three main functions demonstrated below, <span style=\"color:#4682B4\">**_api_soup_**</span>, <span style=\"color:#4682B4\">**_id_search_**</span>, and <span style=\"color:#4682B4\">**_get_metadata._**</span>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>pre { line-height: 125%; }\n",
       "td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       ".output_html .hll { background-color: #ffffcc }\n",
       ".output_html { background: #f8f8f8; }\n",
       ".output_html .c { color: #3D7B7B; font-style: italic } /* Comment */\n",
       ".output_html .err { border: 1px solid #FF0000 } /* Error */\n",
       ".output_html .k { color: #008000; font-weight: bold } /* Keyword */\n",
       ".output_html .o { color: #666666 } /* Operator */\n",
       ".output_html .ch { color: #3D7B7B; font-style: italic } /* Comment.Hashbang */\n",
       ".output_html .cm { color: #3D7B7B; font-style: italic } /* Comment.Multiline */\n",
       ".output_html .cp { color: #9C6500 } /* Comment.Preproc */\n",
       ".output_html .cpf { color: #3D7B7B; font-style: italic } /* Comment.PreprocFile */\n",
       ".output_html .c1 { color: #3D7B7B; font-style: italic } /* Comment.Single */\n",
       ".output_html .cs { color: #3D7B7B; font-style: italic } /* Comment.Special */\n",
       ".output_html .gd { color: #A00000 } /* Generic.Deleted */\n",
       ".output_html .ge { font-style: italic } /* Generic.Emph */\n",
       ".output_html .gr { color: #E40000 } /* Generic.Error */\n",
       ".output_html .gh { color: #000080; font-weight: bold } /* Generic.Heading */\n",
       ".output_html .gi { color: #008400 } /* Generic.Inserted */\n",
       ".output_html .go { color: #717171 } /* Generic.Output */\n",
       ".output_html .gp { color: #000080; font-weight: bold } /* Generic.Prompt */\n",
       ".output_html .gs { font-weight: bold } /* Generic.Strong */\n",
       ".output_html .gu { color: #800080; font-weight: bold } /* Generic.Subheading */\n",
       ".output_html .gt { color: #0044DD } /* Generic.Traceback */\n",
       ".output_html .kc { color: #008000; font-weight: bold } /* Keyword.Constant */\n",
       ".output_html .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */\n",
       ".output_html .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */\n",
       ".output_html .kp { color: #008000 } /* Keyword.Pseudo */\n",
       ".output_html .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */\n",
       ".output_html .kt { color: #B00040 } /* Keyword.Type */\n",
       ".output_html .m { color: #666666 } /* Literal.Number */\n",
       ".output_html .s { color: #BA2121 } /* Literal.String */\n",
       ".output_html .na { color: #687822 } /* Name.Attribute */\n",
       ".output_html .nb { color: #008000 } /* Name.Builtin */\n",
       ".output_html .nc { color: #0000FF; font-weight: bold } /* Name.Class */\n",
       ".output_html .no { color: #880000 } /* Name.Constant */\n",
       ".output_html .nd { color: #AA22FF } /* Name.Decorator */\n",
       ".output_html .ni { color: #717171; font-weight: bold } /* Name.Entity */\n",
       ".output_html .ne { color: #CB3F38; font-weight: bold } /* Name.Exception */\n",
       ".output_html .nf { color: #0000FF } /* Name.Function */\n",
       ".output_html .nl { color: #767600 } /* Name.Label */\n",
       ".output_html .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */\n",
       ".output_html .nt { color: #008000; font-weight: bold } /* Name.Tag */\n",
       ".output_html .nv { color: #19177C } /* Name.Variable */\n",
       ".output_html .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */\n",
       ".output_html .w { color: #bbbbbb } /* Text.Whitespace */\n",
       ".output_html .mb { color: #666666 } /* Literal.Number.Bin */\n",
       ".output_html .mf { color: #666666 } /* Literal.Number.Float */\n",
       ".output_html .mh { color: #666666 } /* Literal.Number.Hex */\n",
       ".output_html .mi { color: #666666 } /* Literal.Number.Integer */\n",
       ".output_html .mo { color: #666666 } /* Literal.Number.Oct */\n",
       ".output_html .sa { color: #BA2121 } /* Literal.String.Affix */\n",
       ".output_html .sb { color: #BA2121 } /* Literal.String.Backtick */\n",
       ".output_html .sc { color: #BA2121 } /* Literal.String.Char */\n",
       ".output_html .dl { color: #BA2121 } /* Literal.String.Delimiter */\n",
       ".output_html .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */\n",
       ".output_html .s2 { color: #BA2121 } /* Literal.String.Double */\n",
       ".output_html .se { color: #AA5D1F; font-weight: bold } /* Literal.String.Escape */\n",
       ".output_html .sh { color: #BA2121 } /* Literal.String.Heredoc */\n",
       ".output_html .si { color: #A45A77; font-weight: bold } /* Literal.String.Interpol */\n",
       ".output_html .sx { color: #008000 } /* Literal.String.Other */\n",
       ".output_html .sr { color: #A45A77 } /* Literal.String.Regex */\n",
       ".output_html .s1 { color: #BA2121 } /* Literal.String.Single */\n",
       ".output_html .ss { color: #19177C } /* Literal.String.Symbol */\n",
       ".output_html .bp { color: #008000 } /* Name.Builtin.Pseudo */\n",
       ".output_html .fm { color: #0000FF } /* Name.Function.Magic */\n",
       ".output_html .vc { color: #19177C } /* Name.Variable.Class */\n",
       ".output_html .vg { color: #19177C } /* Name.Variable.Global */\n",
       ".output_html .vi { color: #19177C } /* Name.Variable.Instance */\n",
       ".output_html .vm { color: #19177C } /* Name.Variable.Magic */\n",
       ".output_html .il { color: #666666 } /* Literal.Number.Integer.Long */</style><div class=\"highlight\"><pre><span></span><span class=\"k\">def</span> <span class=\"nf\">api_soup</span><span class=\"p\">(</span><span class=\"n\">eutil</span><span class=\"p\">,</span> <span class=\"n\">params</span><span class=\"p\">):</span>\n",
       "    <span class=\"sd\">&quot;&quot;&quot;</span>\n",
       "<span class=\"sd\">    Retrieve NCBI Eutils response XML, and</span>\n",
       "<span class=\"sd\">    parse it into a beautifulsoup object</span>\n",
       "<span class=\"sd\">    &quot;&quot;&quot;</span>\n",
       "\n",
       "    <span class=\"n\">url</span> <span class=\"o\">=</span> <span class=\"sa\">f</span><span class=\"s2\">&quot;https://eutils.ncbi.nlm.nih.gov/entrez/eutils/</span><span class=\"si\">{</span><span class=\"n\">eutil</span><span class=\"si\">}</span><span class=\"s2\">.fcgi&quot;</span>\n",
       "    <span class=\"n\">response</span> <span class=\"o\">=</span> <span class=\"n\">requests</span><span class=\"o\">.</span><span class=\"n\">get</span><span class=\"p\">(</span><span class=\"n\">url</span><span class=\"p\">,</span> <span class=\"n\">params</span><span class=\"p\">)</span>\n",
       "    <span class=\"n\">soup</span> <span class=\"o\">=</span> <span class=\"n\">BeautifulSoup</span><span class=\"p\">(</span><span class=\"n\">response</span><span class=\"o\">.</span><span class=\"n\">content</span><span class=\"p\">,</span> <span class=\"n\">features</span><span class=\"o\">=</span><span class=\"s2\">&quot;xml&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"k\">return</span> <span class=\"n\">soup</span>\n",
       "</pre></div>\n"
      ],
      "text/latex": [
       "\\begin{Verbatim}[commandchars=\\\\\\{\\}]\n",
       "\\PY{k}{def} \\PY{n+nf}{api\\PYZus{}soup}\\PY{p}{(}\\PY{n}{eutil}\\PY{p}{,} \\PY{n}{params}\\PY{p}{)}\\PY{p}{:}\n",
       "    \\PY{l+s+sd}{\\PYZdq{}\\PYZdq{}\\PYZdq{}}\n",
       "\\PY{l+s+sd}{    Retrieve NCBI Eutils response XML, and}\n",
       "\\PY{l+s+sd}{    parse it into a beautifulsoup object}\n",
       "\\PY{l+s+sd}{    \\PYZdq{}\\PYZdq{}\\PYZdq{}}\n",
       "\n",
       "    \\PY{n}{url} \\PY{o}{=} \\PY{l+s+sa}{f}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{https://eutils.ncbi.nlm.nih.gov/entrez/eutils/}\\PY{l+s+si}{\\PYZob{}}\\PY{n}{eutil}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{.fcgi}\\PY{l+s+s2}{\\PYZdq{}}\n",
       "    \\PY{n}{response} \\PY{o}{=} \\PY{n}{requests}\\PY{o}{.}\\PY{n}{get}\\PY{p}{(}\\PY{n}{url}\\PY{p}{,} \\PY{n}{params}\\PY{p}{)}\n",
       "    \\PY{n}{soup} \\PY{o}{=} \\PY{n}{BeautifulSoup}\\PY{p}{(}\\PY{n}{response}\\PY{o}{.}\\PY{n}{content}\\PY{p}{,} \\PY{n}{features}\\PY{o}{=}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{xml}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "    \\PY{k}{return} \\PY{n}{soup}\n",
       "\\end{Verbatim}\n"
      ],
      "text/plain": [
       "def api_soup(eutil, params):\n",
       "    \"\"\"\n",
       "    Retrieve NCBI Eutils response XML, and\n",
       "    parse it into a beautifulsoup object\n",
       "    \"\"\"\n",
       "\n",
       "    url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/{eutil}.fcgi\"\n",
       "    response = requests.get(url, params)\n",
       "    soup = BeautifulSoup(response.content, features=\"xml\")\n",
       "\n",
       "    return soup"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Code(inspect.getsource(ncbi.api_soup), language='python')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>pre { line-height: 125%; }\n",
       "td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       ".output_html .hll { background-color: #ffffcc }\n",
       ".output_html { background: #f8f8f8; }\n",
       ".output_html .c { color: #3D7B7B; font-style: italic } /* Comment */\n",
       ".output_html .err { border: 1px solid #FF0000 } /* Error */\n",
       ".output_html .k { color: #008000; font-weight: bold } /* Keyword */\n",
       ".output_html .o { color: #666666 } /* Operator */\n",
       ".output_html .ch { color: #3D7B7B; font-style: italic } /* Comment.Hashbang */\n",
       ".output_html .cm { color: #3D7B7B; font-style: italic } /* Comment.Multiline */\n",
       ".output_html .cp { color: #9C6500 } /* Comment.Preproc */\n",
       ".output_html .cpf { color: #3D7B7B; font-style: italic } /* Comment.PreprocFile */\n",
       ".output_html .c1 { color: #3D7B7B; font-style: italic } /* Comment.Single */\n",
       ".output_html .cs { color: #3D7B7B; font-style: italic } /* Comment.Special */\n",
       ".output_html .gd { color: #A00000 } /* Generic.Deleted */\n",
       ".output_html .ge { font-style: italic } /* Generic.Emph */\n",
       ".output_html .gr { color: #E40000 } /* Generic.Error */\n",
       ".output_html .gh { color: #000080; font-weight: bold } /* Generic.Heading */\n",
       ".output_html .gi { color: #008400 } /* Generic.Inserted */\n",
       ".output_html .go { color: #717171 } /* Generic.Output */\n",
       ".output_html .gp { color: #000080; font-weight: bold } /* Generic.Prompt */\n",
       ".output_html .gs { font-weight: bold } /* Generic.Strong */\n",
       ".output_html .gu { color: #800080; font-weight: bold } /* Generic.Subheading */\n",
       ".output_html .gt { color: #0044DD } /* Generic.Traceback */\n",
       ".output_html .kc { color: #008000; font-weight: bold } /* Keyword.Constant */\n",
       ".output_html .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */\n",
       ".output_html .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */\n",
       ".output_html .kp { color: #008000 } /* Keyword.Pseudo */\n",
       ".output_html .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */\n",
       ".output_html .kt { color: #B00040 } /* Keyword.Type */\n",
       ".output_html .m { color: #666666 } /* Literal.Number */\n",
       ".output_html .s { color: #BA2121 } /* Literal.String */\n",
       ".output_html .na { color: #687822 } /* Name.Attribute */\n",
       ".output_html .nb { color: #008000 } /* Name.Builtin */\n",
       ".output_html .nc { color: #0000FF; font-weight: bold } /* Name.Class */\n",
       ".output_html .no { color: #880000 } /* Name.Constant */\n",
       ".output_html .nd { color: #AA22FF } /* Name.Decorator */\n",
       ".output_html .ni { color: #717171; font-weight: bold } /* Name.Entity */\n",
       ".output_html .ne { color: #CB3F38; font-weight: bold } /* Name.Exception */\n",
       ".output_html .nf { color: #0000FF } /* Name.Function */\n",
       ".output_html .nl { color: #767600 } /* Name.Label */\n",
       ".output_html .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */\n",
       ".output_html .nt { color: #008000; font-weight: bold } /* Name.Tag */\n",
       ".output_html .nv { color: #19177C } /* Name.Variable */\n",
       ".output_html .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */\n",
       ".output_html .w { color: #bbbbbb } /* Text.Whitespace */\n",
       ".output_html .mb { color: #666666 } /* Literal.Number.Bin */\n",
       ".output_html .mf { color: #666666 } /* Literal.Number.Float */\n",
       ".output_html .mh { color: #666666 } /* Literal.Number.Hex */\n",
       ".output_html .mi { color: #666666 } /* Literal.Number.Integer */\n",
       ".output_html .mo { color: #666666 } /* Literal.Number.Oct */\n",
       ".output_html .sa { color: #BA2121 } /* Literal.String.Affix */\n",
       ".output_html .sb { color: #BA2121 } /* Literal.String.Backtick */\n",
       ".output_html .sc { color: #BA2121 } /* Literal.String.Char */\n",
       ".output_html .dl { color: #BA2121 } /* Literal.String.Delimiter */\n",
       ".output_html .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */\n",
       ".output_html .s2 { color: #BA2121 } /* Literal.String.Double */\n",
       ".output_html .se { color: #AA5D1F; font-weight: bold } /* Literal.String.Escape */\n",
       ".output_html .sh { color: #BA2121 } /* Literal.String.Heredoc */\n",
       ".output_html .si { color: #A45A77; font-weight: bold } /* Literal.String.Interpol */\n",
       ".output_html .sx { color: #008000 } /* Literal.String.Other */\n",
       ".output_html .sr { color: #A45A77 } /* Literal.String.Regex */\n",
       ".output_html .s1 { color: #BA2121 } /* Literal.String.Single */\n",
       ".output_html .ss { color: #19177C } /* Literal.String.Symbol */\n",
       ".output_html .bp { color: #008000 } /* Name.Builtin.Pseudo */\n",
       ".output_html .fm { color: #0000FF } /* Name.Function.Magic */\n",
       ".output_html .vc { color: #19177C } /* Name.Variable.Class */\n",
       ".output_html .vg { color: #19177C } /* Name.Variable.Global */\n",
       ".output_html .vi { color: #19177C } /* Name.Variable.Instance */\n",
       ".output_html .vm { color: #19177C } /* Name.Variable.Magic */\n",
       ".output_html .il { color: #666666 } /* Literal.Number.Integer.Long */</style><div class=\"highlight\"><pre><span></span><span class=\"k\">def</span> <span class=\"nf\">id_search</span><span class=\"p\">(</span><span class=\"n\">name</span><span class=\"p\">):</span>\n",
       "    <span class=\"sd\">&quot;&quot;&quot;Get ID from text search, using NCBI esearch eutil&quot;&quot;&quot;</span>\n",
       "\n",
       "    <span class=\"n\">logger</span><span class=\"o\">.</span><span class=\"n\">info</span><span class=\"p\">(</span><span class=\"sa\">f</span><span class=\"s2\">&quot;Searching ncbi for term </span><span class=\"si\">{</span><span class=\"n\">name</span><span class=\"si\">}</span><span class=\"s2\">&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"n\">params</span> <span class=\"o\">=</span> <span class=\"p\">{</span><span class=\"s2\">&quot;db&quot;</span><span class=\"p\">:</span> <span class=\"s2\">&quot;Taxonomy&quot;</span><span class=\"p\">,</span> <span class=\"s2\">&quot;term&quot;</span><span class=\"p\">:</span> <span class=\"n\">name</span><span class=\"p\">}</span>\n",
       "\n",
       "    <span class=\"n\">soup</span> <span class=\"o\">=</span> <span class=\"n\">ncbi</span><span class=\"o\">.</span><span class=\"n\">api_soup</span><span class=\"p\">(</span><span class=\"s2\">&quot;esearch&quot;</span><span class=\"p\">,</span> <span class=\"n\">params</span><span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"k\">try</span><span class=\"p\">:</span>\n",
       "        <span class=\"n\">ncbi_id</span> <span class=\"o\">=</span> <span class=\"n\">soup</span><span class=\"o\">.</span><span class=\"n\">find</span><span class=\"p\">(</span><span class=\"s2\">&quot;Id&quot;</span><span class=\"p\">)</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">()</span>\n",
       "\n",
       "    <span class=\"k\">except</span> <span class=\"ne\">AttributeError</span><span class=\"p\">:</span>\n",
       "        <span class=\"n\">errors</span> <span class=\"o\">=</span> <span class=\"n\">soup</span><span class=\"o\">.</span><span class=\"n\">find</span><span class=\"p\">(</span><span class=\"s2\">&quot;ErrorList&quot;</span><span class=\"p\">)</span>\n",
       "        <span class=\"n\">warnings</span> <span class=\"o\">=</span> <span class=\"n\">soup</span><span class=\"o\">.</span><span class=\"n\">find</span><span class=\"p\">(</span><span class=\"s2\">&quot;WarningList&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "        <span class=\"k\">for</span> <span class=\"n\">error</span> <span class=\"ow\">in</span> <span class=\"n\">errors</span><span class=\"o\">.</span><span class=\"n\">children</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">logger</span><span class=\"o\">.</span><span class=\"n\">error</span><span class=\"p\">(</span><span class=\"sa\">f</span><span class=\"s2\">&quot;</span><span class=\"si\">{</span><span class=\"n\">error</span><span class=\"o\">.</span><span class=\"n\">name</span><span class=\"si\">}</span><span class=\"s2\">: </span><span class=\"si\">{</span><span class=\"n\">error</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">()</span><span class=\"si\">}</span><span class=\"s2\">&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "        <span class=\"k\">for</span> <span class=\"n\">warning</span> <span class=\"ow\">in</span> <span class=\"n\">warnings</span><span class=\"o\">.</span><span class=\"n\">children</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">logger</span><span class=\"o\">.</span><span class=\"n\">warning</span><span class=\"p\">(</span><span class=\"sa\">f</span><span class=\"s2\">&quot;</span><span class=\"si\">{</span><span class=\"n\">warning</span><span class=\"o\">.</span><span class=\"n\">name</span><span class=\"si\">}</span><span class=\"s2\">: </span><span class=\"si\">{</span><span class=\"n\">warning</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">()</span><span class=\"si\">}</span><span class=\"s2\">&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "        <span class=\"k\">return</span> <span class=\"kc\">None</span>\n",
       "\n",
       "    <span class=\"k\">return</span> <span class=\"n\">ncbi_id</span>\n",
       "</pre></div>\n"
      ],
      "text/latex": [
       "\\begin{Verbatim}[commandchars=\\\\\\{\\}]\n",
       "\\PY{k}{def} \\PY{n+nf}{id\\PYZus{}search}\\PY{p}{(}\\PY{n}{name}\\PY{p}{)}\\PY{p}{:}\n",
       "    \\PY{l+s+sd}{\\PYZdq{}\\PYZdq{}\\PYZdq{}Get ID from text search, using NCBI esearch eutil\\PYZdq{}\\PYZdq{}\\PYZdq{}}\n",
       "\n",
       "    \\PY{n}{logger}\\PY{o}{.}\\PY{n}{info}\\PY{p}{(}\\PY{l+s+sa}{f}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Searching ncbi for term }\\PY{l+s+si}{\\PYZob{}}\\PY{n}{name}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "    \\PY{n}{params} \\PY{o}{=} \\PY{p}{\\PYZob{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{db}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Taxonomy}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{,} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{term}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{name}\\PY{p}{\\PYZcb{}}\n",
       "\n",
       "    \\PY{n}{soup} \\PY{o}{=} \\PY{n}{ncbi}\\PY{o}{.}\\PY{n}{api\\PYZus{}soup}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{esearch}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{,} \\PY{n}{params}\\PY{p}{)}\n",
       "\n",
       "    \\PY{k}{try}\\PY{p}{:}\n",
       "        \\PY{n}{ncbi\\PYZus{}id} \\PY{o}{=} \\PY{n}{soup}\\PY{o}{.}\\PY{n}{find}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Id}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\n",
       "\n",
       "    \\PY{k}{except} \\PY{n+ne}{AttributeError}\\PY{p}{:}\n",
       "        \\PY{n}{errors} \\PY{o}{=} \\PY{n}{soup}\\PY{o}{.}\\PY{n}{find}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{ErrorList}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "        \\PY{n}{warnings} \\PY{o}{=} \\PY{n}{soup}\\PY{o}{.}\\PY{n}{find}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{WarningList}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "        \\PY{k}{for} \\PY{n}{error} \\PY{o+ow}{in} \\PY{n}{errors}\\PY{o}{.}\\PY{n}{children}\\PY{p}{:}\n",
       "            \\PY{n}{logger}\\PY{o}{.}\\PY{n}{error}\\PY{p}{(}\\PY{l+s+sa}{f}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+si}{\\PYZob{}}\\PY{n}{error}\\PY{o}{.}\\PY{n}{name}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{: }\\PY{l+s+si}{\\PYZob{}}\\PY{n}{error}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "        \\PY{k}{for} \\PY{n}{warning} \\PY{o+ow}{in} \\PY{n}{warnings}\\PY{o}{.}\\PY{n}{children}\\PY{p}{:}\n",
       "            \\PY{n}{logger}\\PY{o}{.}\\PY{n}{warning}\\PY{p}{(}\\PY{l+s+sa}{f}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+si}{\\PYZob{}}\\PY{n}{warning}\\PY{o}{.}\\PY{n}{name}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{: }\\PY{l+s+si}{\\PYZob{}}\\PY{n}{warning}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "        \\PY{k}{return} \\PY{k+kc}{None}\n",
       "\n",
       "    \\PY{k}{return} \\PY{n}{ncbi\\PYZus{}id}\n",
       "\\end{Verbatim}\n"
      ],
      "text/plain": [
       "def id_search(name):\n",
       "    \"\"\"Get ID from text search, using NCBI esearch eutil\"\"\"\n",
       "\n",
       "    logger.info(f\"Searching ncbi for term {name}\")\n",
       "\n",
       "    params = {\"db\": \"Taxonomy\", \"term\": name}\n",
       "\n",
       "    soup = ncbi.api_soup(\"esearch\", params)\n",
       "\n",
       "    try:\n",
       "        ncbi_id = soup.find(\"Id\").getText()\n",
       "\n",
       "    except AttributeError:\n",
       "        errors = soup.find(\"ErrorList\")\n",
       "        warnings = soup.find(\"WarningList\")\n",
       "\n",
       "        for error in errors.children:\n",
       "            logger.error(f\"{error.name}: {error.getText()}\")\n",
       "\n",
       "        for warning in warnings.children:\n",
       "            logger.warning(f\"{warning.name}: {warning.getText()}\")\n",
       "\n",
       "        return None\n",
       "\n",
       "    return ncbi_id"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Code(inspect.getsource(ncbi.id_search), language='python')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>pre { line-height: 125%; }\n",
       "td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       ".output_html .hll { background-color: #ffffcc }\n",
       ".output_html { background: #f8f8f8; }\n",
       ".output_html .c { color: #3D7B7B; font-style: italic } /* Comment */\n",
       ".output_html .err { border: 1px solid #FF0000 } /* Error */\n",
       ".output_html .k { color: #008000; font-weight: bold } /* Keyword */\n",
       ".output_html .o { color: #666666 } /* Operator */\n",
       ".output_html .ch { color: #3D7B7B; font-style: italic } /* Comment.Hashbang */\n",
       ".output_html .cm { color: #3D7B7B; font-style: italic } /* Comment.Multiline */\n",
       ".output_html .cp { color: #9C6500 } /* Comment.Preproc */\n",
       ".output_html .cpf { color: #3D7B7B; font-style: italic } /* Comment.PreprocFile */\n",
       ".output_html .c1 { color: #3D7B7B; font-style: italic } /* Comment.Single */\n",
       ".output_html .cs { color: #3D7B7B; font-style: italic } /* Comment.Special */\n",
       ".output_html .gd { color: #A00000 } /* Generic.Deleted */\n",
       ".output_html .ge { font-style: italic } /* Generic.Emph */\n",
       ".output_html .gr { color: #E40000 } /* Generic.Error */\n",
       ".output_html .gh { color: #000080; font-weight: bold } /* Generic.Heading */\n",
       ".output_html .gi { color: #008400 } /* Generic.Inserted */\n",
       ".output_html .go { color: #717171 } /* Generic.Output */\n",
       ".output_html .gp { color: #000080; font-weight: bold } /* Generic.Prompt */\n",
       ".output_html .gs { font-weight: bold } /* Generic.Strong */\n",
       ".output_html .gu { color: #800080; font-weight: bold } /* Generic.Subheading */\n",
       ".output_html .gt { color: #0044DD } /* Generic.Traceback */\n",
       ".output_html .kc { color: #008000; font-weight: bold } /* Keyword.Constant */\n",
       ".output_html .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */\n",
       ".output_html .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */\n",
       ".output_html .kp { color: #008000 } /* Keyword.Pseudo */\n",
       ".output_html .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */\n",
       ".output_html .kt { color: #B00040 } /* Keyword.Type */\n",
       ".output_html .m { color: #666666 } /* Literal.Number */\n",
       ".output_html .s { color: #BA2121 } /* Literal.String */\n",
       ".output_html .na { color: #687822 } /* Name.Attribute */\n",
       ".output_html .nb { color: #008000 } /* Name.Builtin */\n",
       ".output_html .nc { color: #0000FF; font-weight: bold } /* Name.Class */\n",
       ".output_html .no { color: #880000 } /* Name.Constant */\n",
       ".output_html .nd { color: #AA22FF } /* Name.Decorator */\n",
       ".output_html .ni { color: #717171; font-weight: bold } /* Name.Entity */\n",
       ".output_html .ne { color: #CB3F38; font-weight: bold } /* Name.Exception */\n",
       ".output_html .nf { color: #0000FF } /* Name.Function */\n",
       ".output_html .nl { color: #767600 } /* Name.Label */\n",
       ".output_html .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */\n",
       ".output_html .nt { color: #008000; font-weight: bold } /* Name.Tag */\n",
       ".output_html .nv { color: #19177C } /* Name.Variable */\n",
       ".output_html .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */\n",
       ".output_html .w { color: #bbbbbb } /* Text.Whitespace */\n",
       ".output_html .mb { color: #666666 } /* Literal.Number.Bin */\n",
       ".output_html .mf { color: #666666 } /* Literal.Number.Float */\n",
       ".output_html .mh { color: #666666 } /* Literal.Number.Hex */\n",
       ".output_html .mi { color: #666666 } /* Literal.Number.Integer */\n",
       ".output_html .mo { color: #666666 } /* Literal.Number.Oct */\n",
       ".output_html .sa { color: #BA2121 } /* Literal.String.Affix */\n",
       ".output_html .sb { color: #BA2121 } /* Literal.String.Backtick */\n",
       ".output_html .sc { color: #BA2121 } /* Literal.String.Char */\n",
       ".output_html .dl { color: #BA2121 } /* Literal.String.Delimiter */\n",
       ".output_html .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */\n",
       ".output_html .s2 { color: #BA2121 } /* Literal.String.Double */\n",
       ".output_html .se { color: #AA5D1F; font-weight: bold } /* Literal.String.Escape */\n",
       ".output_html .sh { color: #BA2121 } /* Literal.String.Heredoc */\n",
       ".output_html .si { color: #A45A77; font-weight: bold } /* Literal.String.Interpol */\n",
       ".output_html .sx { color: #008000 } /* Literal.String.Other */\n",
       ".output_html .sr { color: #A45A77 } /* Literal.String.Regex */\n",
       ".output_html .s1 { color: #BA2121 } /* Literal.String.Single */\n",
       ".output_html .ss { color: #19177C } /* Literal.String.Symbol */\n",
       ".output_html .bp { color: #008000 } /* Name.Builtin.Pseudo */\n",
       ".output_html .fm { color: #0000FF } /* Name.Function.Magic */\n",
       ".output_html .vc { color: #19177C } /* Name.Variable.Class */\n",
       ".output_html .vg { color: #19177C } /* Name.Variable.Global */\n",
       ".output_html .vi { color: #19177C } /* Name.Variable.Instance */\n",
       ".output_html .vm { color: #19177C } /* Name.Variable.Magic */\n",
       ".output_html .il { color: #666666 } /* Literal.Number.Integer.Long */</style><div class=\"highlight\"><pre><span></span><span class=\"k\">def</span> <span class=\"nf\">get_metadata</span><span class=\"p\">(</span><span class=\"n\">ncbi_id</span><span class=\"p\">):</span>\n",
       "    <span class=\"sd\">&quot;&quot;&quot;Request metadata by NCBI taxonomy ID, and return cleaned object&quot;&quot;&quot;</span>\n",
       "\n",
       "    <span class=\"n\">params</span> <span class=\"o\">=</span> <span class=\"p\">{</span><span class=\"s2\">&quot;db&quot;</span><span class=\"p\">:</span> <span class=\"s2\">&quot;Taxonomy&quot;</span><span class=\"p\">,</span> <span class=\"s2\">&quot;id&quot;</span><span class=\"p\">:</span> <span class=\"n\">ncbi_id</span><span class=\"p\">}</span>\n",
       "    <span class=\"n\">soup</span> <span class=\"o\">=</span> <span class=\"n\">ncbi</span><span class=\"o\">.</span><span class=\"n\">api_soup</span><span class=\"p\">(</span><span class=\"s2\">&quot;efetch&quot;</span><span class=\"p\">,</span> <span class=\"n\">params</span><span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"n\">taxon</span> <span class=\"o\">=</span> <span class=\"n\">soup</span><span class=\"o\">.</span><span class=\"n\">TaxaSet</span><span class=\"o\">.</span><span class=\"n\">Taxon</span>\n",
       "\n",
       "    <span class=\"n\">taxon_metadata</span> <span class=\"o\">=</span> <span class=\"p\">{</span>\n",
       "        <span class=\"s2\">&quot;ScientificName&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">ScientificName</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;ParentTaxId&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">ParentTaxId</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;Rank&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">Rank</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;Division&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">Division</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;GeneticCode&quot;</span><span class=\"p\">:</span> <span class=\"p\">{</span><span class=\"s2\">&quot;GCId&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">GCId</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span> <span class=\"s2\">&quot;GCName&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">GCName</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">()},</span>\n",
       "        <span class=\"s2\">&quot;MitoGeneticCode&quot;</span><span class=\"p\">:</span> <span class=\"p\">{</span>\n",
       "            <span class=\"s2\">&quot;MGCId&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">MGCId</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "            <span class=\"s2\">&quot;MGCName&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">MGCName</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"p\">},</span>\n",
       "        <span class=\"s2\">&quot;Lineage&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">Lineage</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;CreateDate&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">CreateDate</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;UpdateDate&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">UpdateDate</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;PubDate&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">PubDate</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "    <span class=\"p\">}</span>\n",
       "\n",
       "    <span class=\"k\">if</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">otherNames</span><span class=\"p\">:</span>\n",
       "        <span class=\"n\">taxon</span><span class=\"p\">[</span><span class=\"s2\">&quot;OtherNames&quot;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"p\">(</span><span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">OtherNames</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),)</span>\n",
       "\n",
       "    <span class=\"c1\"># parse lineage</span>\n",
       "    <span class=\"n\">lineage_ex</span> <span class=\"o\">=</span> <span class=\"p\">[]</span>\n",
       "    <span class=\"k\">for</span> <span class=\"n\">taxon</span> <span class=\"ow\">in</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">LineageEx</span><span class=\"o\">.</span><span class=\"n\">children</span><span class=\"p\">:</span>\n",
       "        <span class=\"k\">if</span> <span class=\"nb\">isinstance</span><span class=\"p\">(</span><span class=\"n\">taxon</span><span class=\"p\">,</span> <span class=\"n\">Tag</span><span class=\"p\">):</span>\n",
       "            <span class=\"n\">lineage_ex</span><span class=\"o\">.</span><span class=\"n\">append</span><span class=\"p\">(</span>\n",
       "                <span class=\"p\">{</span>\n",
       "                    <span class=\"s2\">&quot;TaxId&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">TaxId</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "                    <span class=\"s2\">&quot;ScientificName&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">ScientificName</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "                    <span class=\"s2\">&quot;Rank&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">Rank</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "                <span class=\"p\">}</span>\n",
       "            <span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"n\">taxon_metadata</span><span class=\"p\">[</span><span class=\"s2\">&quot;LineageEx&quot;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"n\">lineage_ex</span>\n",
       "\n",
       "    <span class=\"k\">return</span> <span class=\"n\">taxon_metadata</span>\n",
       "</pre></div>\n"
      ],
      "text/latex": [
       "\\begin{Verbatim}[commandchars=\\\\\\{\\}]\n",
       "\\PY{k}{def} \\PY{n+nf}{get\\PYZus{}metadata}\\PY{p}{(}\\PY{n}{ncbi\\PYZus{}id}\\PY{p}{)}\\PY{p}{:}\n",
       "    \\PY{l+s+sd}{\\PYZdq{}\\PYZdq{}\\PYZdq{}Request metadata by NCBI taxonomy ID, and return cleaned object\\PYZdq{}\\PYZdq{}\\PYZdq{}}\n",
       "\n",
       "    \\PY{n}{params} \\PY{o}{=} \\PY{p}{\\PYZob{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{db}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Taxonomy}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{,} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{id}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{ncbi\\PYZus{}id}\\PY{p}{\\PYZcb{}}\n",
       "    \\PY{n}{soup} \\PY{o}{=} \\PY{n}{ncbi}\\PY{o}{.}\\PY{n}{api\\PYZus{}soup}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{efetch}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{,} \\PY{n}{params}\\PY{p}{)}\n",
       "\n",
       "    \\PY{n}{taxon} \\PY{o}{=} \\PY{n}{soup}\\PY{o}{.}\\PY{n}{TaxaSet}\\PY{o}{.}\\PY{n}{Taxon}\n",
       "\n",
       "    \\PY{n}{taxon\\PYZus{}metadata} \\PY{o}{=} \\PY{p}{\\PYZob{}}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{ScientificName}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{ScientificName}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{ParentTaxId}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{ParentTaxId}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Rank}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{Rank}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Division}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{Division}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{GeneticCode}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{p}{\\PYZob{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{GCId}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{GCId}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{GCName}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{GCName}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{\\PYZcb{}}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{MitoGeneticCode}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{p}{\\PYZob{}}\n",
       "            \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{MGCId}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{MGCId}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "            \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{MGCName}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{MGCName}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{p}{\\PYZcb{}}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Lineage}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{Lineage}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{CreateDate}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{CreateDate}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{UpdateDate}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{UpdateDate}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{PubDate}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{PubDate}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "    \\PY{p}{\\PYZcb{}}\n",
       "\n",
       "    \\PY{k}{if} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{otherNames}\\PY{p}{:}\n",
       "        \\PY{n}{taxon}\\PY{p}{[}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{OtherNames}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{]} \\PY{o}{=} \\PY{p}{(}\\PY{n}{taxon}\\PY{o}{.}\\PY{n}{OtherNames}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\\PY{p}{)}\n",
       "\n",
       "    \\PY{c+c1}{\\PYZsh{} parse lineage}\n",
       "    \\PY{n}{lineage\\PYZus{}ex} \\PY{o}{=} \\PY{p}{[}\\PY{p}{]}\n",
       "    \\PY{k}{for} \\PY{n}{taxon} \\PY{o+ow}{in} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{LineageEx}\\PY{o}{.}\\PY{n}{children}\\PY{p}{:}\n",
       "        \\PY{k}{if} \\PY{n+nb}{isinstance}\\PY{p}{(}\\PY{n}{taxon}\\PY{p}{,} \\PY{n}{Tag}\\PY{p}{)}\\PY{p}{:}\n",
       "            \\PY{n}{lineage\\PYZus{}ex}\\PY{o}{.}\\PY{n}{append}\\PY{p}{(}\n",
       "                \\PY{p}{\\PYZob{}}\n",
       "                    \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{TaxId}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{TaxId}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "                    \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{ScientificName}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{ScientificName}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "                    \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Rank}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{Rank}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "                \\PY{p}{\\PYZcb{}}\n",
       "            \\PY{p}{)}\n",
       "\n",
       "    \\PY{n}{taxon\\PYZus{}metadata}\\PY{p}{[}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{LineageEx}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{]} \\PY{o}{=} \\PY{n}{lineage\\PYZus{}ex}\n",
       "\n",
       "    \\PY{k}{return} \\PY{n}{taxon\\PYZus{}metadata}\n",
       "\\end{Verbatim}\n"
      ],
      "text/plain": [
       "def get_metadata(ncbi_id):\n",
       "    \"\"\"Request metadata by NCBI taxonomy ID, and return cleaned object\"\"\"\n",
       "\n",
       "    params = {\"db\": \"Taxonomy\", \"id\": ncbi_id}\n",
       "    soup = ncbi.api_soup(\"efetch\", params)\n",
       "\n",
       "    taxon = soup.TaxaSet.Taxon\n",
       "\n",
       "    taxon_metadata = {\n",
       "        \"ScientificName\": taxon.ScientificName.getText(),\n",
       "        \"ParentTaxId\": taxon.ParentTaxId.getText(),\n",
       "        \"Rank\": taxon.Rank.getText(),\n",
       "        \"Division\": taxon.Division.getText(),\n",
       "        \"GeneticCode\": {\"GCId\": taxon.GCId.getText(), \"GCName\": taxon.GCName.getText()},\n",
       "        \"MitoGeneticCode\": {\n",
       "            \"MGCId\": taxon.MGCId.getText(),\n",
       "            \"MGCName\": taxon.MGCName.getText(),\n",
       "        },\n",
       "        \"Lineage\": taxon.Lineage.getText(),\n",
       "        \"CreateDate\": taxon.CreateDate.getText(),\n",
       "        \"UpdateDate\": taxon.UpdateDate.getText(),\n",
       "        \"PubDate\": taxon.PubDate.getText(),\n",
       "    }\n",
       "\n",
       "    if taxon.otherNames:\n",
       "        taxon[\"OtherNames\"] = (taxon.OtherNames.getText(),)\n",
       "\n",
       "    # parse lineage\n",
       "    lineage_ex = []\n",
       "    for taxon in taxon.LineageEx.children:\n",
       "        if isinstance(taxon, Tag):\n",
       "            lineage_ex.append(\n",
       "                {\n",
       "                    \"TaxId\": taxon.TaxId.getText(),\n",
       "                    \"ScientificName\": taxon.ScientificName.getText(),\n",
       "                    \"Rank\": taxon.Rank.getText(),\n",
       "                }\n",
       "            )\n",
       "\n",
       "    taxon_metadata[\"LineageEx\"] = lineage_ex\n",
       "\n",
       "    return taxon_metadata"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Code(inspect.getsource(ncbi.get_metadata), language='python')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <span style=\"color:gray\">_Accessing what is in the NCBI Taxonomy database_</span>\n",
    "\n",
    "The NCBI Taxonomy documentation indicates that the data model is built around a central framework called NameBank, and each entry includes a <span style=\"color:#4682B4\">**_primary name_**</span>, <span style=\"color:#4682B4\">**_secondary names_**</span>, a <span style=\"color:#4682B4\">**_taxonomy identifier_**</span>, <span style=\"color:#4682B4\">**_name entity identifiers_**</span> along with other various metadata about the lineage, genetic code, and linked Entrez records.\n",
    "\n",
    "We first tested our database access using a sample query with the 'esearch' utility for the Entrez NCBI Taxonomy database using formal organism names. \n",
    "\n",
    "<span style=\"color:#4682B4\">**_Input: Entrez text query (&term); Entrez database (&db)_**</span>\n",
    "\n",
    "<span style=\"color:#4682B4\">**_Expected Output: List of UIDs matching the Entrez query_**</span>\n",
    "\n",
    "\n",
    "Sample data: Get the Taxonomic IDs (TaxID) for 'influenza A subtype h1n1' 'alphainfluenzavirus', and 'orthomyxoviridae:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-05-05 17:36:40.312 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term influenza A subtype h1n1\n",
      "2022-05-05 17:36:40.798 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term alphainfluenzavirus\n",
      "2022-05-05 17:36:40.993 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term orthomyxoviridae\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['114727', '197911', '11308']\n"
     ]
    }
   ],
   "source": [
    "term_list = ['influenza A subtype h1n1','alphainfluenzavirus','orthomyxoviridae']\n",
    "id_list = []\n",
    "\n",
    "for each in term_list:\n",
    "    id_list.append(ncbi.id_search(each))\n",
    "\n",
    "print(id_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using this ID, we can then pass it through the 'efetch' utility to get the full records from the NCBI Taxonomy database about organisms in XML format.\n",
    "\n",
    "<span style=\"color:#4682B4\">**_Input: List of UIDs (&id); Entrez database (&db); Retrieval type (&rettype); Retrieval mode (&retmode)_**</span>\n",
    "\n",
    "<span style=\"color:#4682B4\">**_Expected Output: Formatted data records as specified_**</span>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "{'ScientificName': 'H1N1 subtype', 'ParentTaxId': '11320', 'Rank': 'serotype', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales; Orthomyxoviridae; Alphainfluenzavirus; Influenza A virus', 'CreateDate': '2000/02/07 12:42:00', 'UpdateDate': '2020/04/07 15:24:16', 'PubDate': '2000/02/07 12:42:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}, {'TaxId': '11308', 'ScientificName': 'Orthomyxoviridae', 'Rank': 'family'}, {'TaxId': '197911', 'ScientificName': 'Alphainfluenzavirus', 'Rank': 'genus'}, {'TaxId': '11320', 'ScientificName': 'Influenza A virus', 'Rank': 'species'}]}\n",
      "\n",
      "{'ScientificName': 'Alphainfluenzavirus', 'ParentTaxId': '11308', 'Rank': 'genus', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales; Orthomyxoviridae', 'CreateDate': '2002/05/08 12:00:00', 'UpdateDate': '2020/04/07 15:24:16', 'PubDate': '2002/06/12 19:01:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}, {'TaxId': '11308', 'ScientificName': 'Orthomyxoviridae', 'Rank': 'family'}]}\n",
      "\n",
      "{'ScientificName': 'Orthomyxoviridae', 'ParentTaxId': '2499411', 'Rank': 'family', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales', 'CreateDate': '1995/02/27 09:24:00', 'UpdateDate': '2002/05/08 12:00:00', 'PubDate': '1993/04/23 01:00:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}]}\n"
     ]
    }
   ],
   "source": [
    "for each in id_list: \n",
    "    print('\\n' + str(ncbi.get_metadata(each)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The query can handle multiple terms at once and return lists of IDs and full records for each ID. \n",
    "\n",
    "This same query can also be performed using informal names and synonyms for the organisms, as shown below. Together, these features will allow us to pass in disease and species names from other data sources and map them back to the formal Taxonomy."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Diving deeper into Entrez with the species 'Influenzavirus A subtype H1N1' (H1N1):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-05-05 17:39:41.171 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term H1N1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: 114727\n",
      "\n",
      "{'ScientificName': 'H1N1 subtype', 'ParentTaxId': '11320', 'Rank': 'serotype', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales; Orthomyxoviridae; Alphainfluenzavirus; Influenza A virus', 'CreateDate': '2000/02/07 12:42:00', 'UpdateDate': '2020/04/07 15:24:16', 'PubDate': '2000/02/07 12:42:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}, {'TaxId': '11308', 'ScientificName': 'Orthomyxoviridae', 'Rank': 'family'}, {'TaxId': '197911', 'ScientificName': 'Alphainfluenzavirus', 'Rank': 'genus'}, {'TaxId': '11320', 'ScientificName': 'Influenza A virus', 'Rank': 'species'}]}\n"
     ]
    }
   ],
   "source": [
    "h1n1_id = ncbi.id_search('H1N1')\n",
    "print(f'ID: {h1n1_id}\\n')\n",
    "\n",
    "h1n1_data = ncbi.get_metadata(h1n1_id)\n",
    "print(h1n1_data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Specifically, we can pull out the lineage of H1N1 to visualize the hierarchy (especially as nodes) and obtain the IDs for higher ranks. Such queries allow us to use a shortest path analysis to identify the family which contains H1N1, where we specify the start point (H1N1) and the end node type (Family), following edges backwards to return exactly one parent."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TaxId</th>\n",
       "      <th>ScientificName</th>\n",
       "      <th>Rank</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10239</td>\n",
       "      <td>Viruses</td>\n",
       "      <td>superkingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2559587</td>\n",
       "      <td>Riboviria</td>\n",
       "      <td>clade</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2732396</td>\n",
       "      <td>Orthornavirae</td>\n",
       "      <td>kingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2497569</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>phylum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2497571</td>\n",
       "      <td>Polyploviricotina</td>\n",
       "      <td>subphylum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2497577</td>\n",
       "      <td>Insthoviricetes</td>\n",
       "      <td>class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2499411</td>\n",
       "      <td>Articulavirales</td>\n",
       "      <td>order</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>11308</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>197911</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>genus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11320</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>species</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     TaxId       ScientificName          Rank\n",
       "0    10239              Viruses  superkingdom\n",
       "1  2559587            Riboviria         clade\n",
       "2  2732396        Orthornavirae       kingdom\n",
       "3  2497569      Negarnaviricota        phylum\n",
       "4  2497571    Polyploviricotina     subphylum\n",
       "5  2497577      Insthoviricetes         class\n",
       "6  2499411      Articulavirales         order\n",
       "7    11308     Orthomyxoviridae        family\n",
       "8   197911  Alphainfluenzavirus         genus\n",
       "9    11320    Influenza A virus       species"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extracting the lineage\n",
    "lineage = h1n1_data[\"LineageEx\"]\n",
    "\n",
    "lineage_df = pd.DataFrame(lineage)\n",
    "lineage_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using the Taxonomy ID for H1N1, we can use the ELink utility to then access information from any of the other Entrez databases that link with Taxonomy. \n",
    "\n",
    "In the example below, we pull out gene sequence data from the NCBI Gene database:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Entrez Gene Summary: Uronyl 2-sulfotransferase transfers sulfate to the 2-position of uronyl residues, such as iduronyl residues in dermatan sulfate and glucuronyl residues in chondroitin sulfate (Kobayashi et al., 1999 [PubMed 10187838]).[supplied by OMIM, Mar 2008]\n"
     ]
    }
   ],
   "source": [
    "h1n1_gene_id = ncbi.api_soup('elink',{\"db\":\"taxonomy\",\"dbfrom\":\"gene\",\"id\":h1n1_id})\n",
    "h1n1_gene_id = h1n1_gene_id.LinkSetDb.Id.getText()\n",
    "\n",
    "gene_output = ncbi.api_soup(\"efetch\",{\"db\":\"gene\",\"id\":h1n1_gene_id,\"retmode\":\"xml\"})\n",
    "# print(gene_output)\n",
    "gene_output = gene_output.Entrezgene_summary.getText()\n",
    "print(\"Entrez Gene Summary: \" + gene_output)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also use the EGQuery utility to see how many records in all Entrez databases refer to the formal term \"Influenza A virus subtype H1N1\" and compare that with how many mention the informal term \"H1N1\":"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Database Name</th>\n",
       "      <th>Records with Term \"Influenza A virus subtype H1N1\"</th>\n",
       "      <th>Records with Term \"H1N1\"</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pubmed</td>\n",
       "      <td>16901</td>\n",
       "      <td>22948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pmc</td>\n",
       "      <td>6917</td>\n",
       "      <td>63269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mesh</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>books</td>\n",
       "      <td>135</td>\n",
       "      <td>1515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pubmedhealth</td>\n",
       "      <td>Error</td>\n",
       "      <td>Error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>omim</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ncbisearch</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>nuccore</td>\n",
       "      <td>4213</td>\n",
       "      <td>226250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>nucgss</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>nucest</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>protein</td>\n",
       "      <td>1651</td>\n",
       "      <td>307623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>genome</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>structure</td>\n",
       "      <td>28</td>\n",
       "      <td>451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>taxonomy</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>snp</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>dbvar</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>gene</td>\n",
       "      <td>22</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>sra</td>\n",
       "      <td>2077</td>\n",
       "      <td>8805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>biosystems</td>\n",
       "      <td>Error</td>\n",
       "      <td>Error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>unigene</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>cdd</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>clone</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>popset</td>\n",
       "      <td>593</td>\n",
       "      <td>628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>geoprofiles</td>\n",
       "      <td>0</td>\n",
       "      <td>279625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>gds</td>\n",
       "      <td>1042</td>\n",
       "      <td>11865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>homologene</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>pccompound</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>pcsubstance</td>\n",
       "      <td>0</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>pcassay</td>\n",
       "      <td>183</td>\n",
       "      <td>3405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>nlmcatalog</td>\n",
       "      <td>97</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>probe</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>gap</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>proteinclusters</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>bioproject</td>\n",
       "      <td>6</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>biosample</td>\n",
       "      <td>913</td>\n",
       "      <td>6801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>biocollections</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Database Name Records with Term \"Influenza A virus subtype H1N1\"  \\\n",
       "0            pubmed                                              16901   \n",
       "1               pmc                                               6917   \n",
       "2              mesh                                                  1   \n",
       "3             books                                                135   \n",
       "4      pubmedhealth                                              Error   \n",
       "5              omim                                                  0   \n",
       "6        ncbisearch                                                  6   \n",
       "7           nuccore                                               4213   \n",
       "8            nucgss                                                  0   \n",
       "9            nucest                                                  0   \n",
       "10          protein                                               1651   \n",
       "11           genome                                                  0   \n",
       "12        structure                                                 28   \n",
       "13         taxonomy                                                  1   \n",
       "14              snp                                                  0   \n",
       "15            dbvar                                                  0   \n",
       "16             gene                                                 22   \n",
       "17              sra                                               2077   \n",
       "18       biosystems                                              Error   \n",
       "19          unigene                                                  0   \n",
       "20              cdd                                                  0   \n",
       "21            clone                                                  0   \n",
       "22           popset                                                593   \n",
       "23      geoprofiles                                                  0   \n",
       "24              gds                                               1042   \n",
       "25       homologene                                                  0   \n",
       "26       pccompound                                                  0   \n",
       "27      pcsubstance                                                  0   \n",
       "28          pcassay                                                183   \n",
       "29       nlmcatalog                                                 97   \n",
       "30            probe                                                  0   \n",
       "31              gap                                                  0   \n",
       "32  proteinclusters                                                  0   \n",
       "33       bioproject                                                  6   \n",
       "34        biosample                                                913   \n",
       "35   biocollections                                                  5   \n",
       "\n",
       "   Records with Term \"H1N1\"  \n",
       "0                     22948  \n",
       "1                     63269  \n",
       "2                        17  \n",
       "3                      1515  \n",
       "4                     Error  \n",
       "5                        13  \n",
       "6                         9  \n",
       "7                    226250  \n",
       "8                         0  \n",
       "9                         0  \n",
       "10                   307623  \n",
       "11                        1  \n",
       "12                      451  \n",
       "13                        1  \n",
       "14                        0  \n",
       "15                        0  \n",
       "16                      158  \n",
       "17                     8805  \n",
       "18                    Error  \n",
       "19                        0  \n",
       "20                        0  \n",
       "21                        0  \n",
       "22                      628  \n",
       "23                   279625  \n",
       "24                    11865  \n",
       "25                        0  \n",
       "26                        3  \n",
       "27                      489  \n",
       "28                     3405  \n",
       "29                      158  \n",
       "30                        0  \n",
       "31                       98  \n",
       "32                        0  \n",
       "33                      304  \n",
       "34                     6801  \n",
       "35                        0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "query_list = ['H1N1', 'Influenza A virus subtype H1N1']\n",
    "dfs = []\n",
    "\n",
    "for each in query_list:\n",
    "\n",
    "    query = ncbi.api_soup('egquery', {\"term\": each})\n",
    "    query_tag = query.eGQueryResult\n",
    "\n",
    "    query_name = query_tag.find_all('DbName')  \n",
    "    query_count = query_tag.find_all('Count')  \n",
    "\n",
    "    query_data = []\n",
    "    \n",
    "    # Loop to store the data in a list named 'lineage_data'\n",
    "    for i in range(0, len(query_name)):\n",
    "        query_rows = [query_name[i].get_text(), query_count[i].get_text()]\n",
    "        query_data.append(query_rows)\n",
    "    \n",
    "    # Converting the list into dataframe\n",
    "    query_df = pd.DataFrame(query_data, columns=['Database Name', 'Records with Term \"' + each + '\"'], dtype = str)\n",
    "\n",
    "    dfs.append(query_df)\n",
    "\n",
    "df = pd.merge(right=dfs[0], left=dfs[1])\n",
    "\n",
    "display(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It should be noted that the taxonomy database only has 1 record that refers to either, and that they are the same record. This shows the core mapping function of the taxonomy database -- all other Entrez linkages map back to the same taxonomy.\n",
    "\n",
    "Additional tests reveal that NCBI is strict about terms used for implementation of the ESearch utility. For instance, searching for the term 'Influenza Virus A' reveals that the phrase is not found in the NCBI Taxonomy:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<ErrorList>\n",
      " <PhraseNotFound>\n",
      "  Influenza\n",
      " </PhraseNotFound>\n",
      " <PhraseNotFound>\n",
      "  Virus\n",
      " </PhraseNotFound>\n",
      " <PhraseNotFound>\n",
      "  A\n",
      " </PhraseNotFound>\n",
      "</ErrorList>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "correction = ncbi.api_soup('esearch', {\"term\": 'Influenza Virus A', \"db\":\"taxonomy\"})\n",
    "print(correction.ErrorList.prettify())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "However, the terms 'Influenzavirus A' and 'Influenza A virus' can both be found using ESearch:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<IdList>\n",
      " <Id>\n",
      "  197911\n",
      " </Id>\n",
      "</IdList>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "id_response = ncbi.api_soup('esearch', {\"term\": 'Influenzavirus A', \"db\":\"taxonomy\"})\n",
    "print(id_response.IdList.prettify())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<IdList>\n",
      " <Id>\n",
      "  11320\n",
      " </Id>\n",
      "</IdList>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "id_response = ncbi.api_soup('esearch', {\"term\": 'Influenza A virus', \"db\":\"taxonomy\"})\n",
    "print(id_response.IdList.prettify())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "While at first glance, it may appear that ESearch doesn't handle synonyms as expected. However, these tests actually reveal that NCBI has a highly controlled taxonomy which means we can rely upon it as a data source. The reason 'Influenza Virus A' is not supported by ESearch is because it could be a misnomer for either the genus (Alphainfluenzavirus, informal equivalent is \"Influenzavirus A\") or the species (Influenza A Virus). \n",
    "\n",
    "The ESpell utility provided by NCBI offers a way to view possible spellings for a particular organism to help reduce the risk of using invalid search terms in ESearch, as demonstrated below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<CorrectedQuery>\n",
      " influenzavirus a\n",
      "</CorrectedQuery>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "correction = ncbi.api_soup('espell', {\"term\": 'Influenzavirus A', \"db\":\"taxonomy\"})\n",
    "print(correction.CorrectedQuery.prettify())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "However, ESpell does not function perfectly. Human work is required for review of many synonyms in NCBI, as there are unpredictable mistakes which could pose issues to data integrity. \n",
    "\n",
    "For instance, searching \"Influenza Virus B\" (expected output: \"Influenzavirus B\") returns \"Influenza Virus A\":"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<CorrectedQuery>\n",
      " influenza virus a\n",
      "</CorrectedQuery>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "correction = ncbi.api_soup('espell', {\"term\": 'Influenza Virus B', \"db\":\"taxonomy\"})\n",
    "print(correction.CorrectedQuery.prettify())"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "d7315c49e695b70a80fa10d86a25c9d146976f258accde30ac4a8950a94324f1"
  },
  "kernelspec": {
   "display_name": "Python 3.9.12 ('env': venv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
