{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style=\"color:#4682B4\">NCBI Taxonomy Data Profile</span>\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The [NCBI Taxonomy](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi) is a reference database that contains the names and hierarchically-arranged phylogenetic classifications of organisms created in 1991. The NCBI Taxonomy database is organized in a tree for a hierarchical data structure, where each <span style=\"color:#4682B4\">**_node_**</span> of the tree represents a <span style=\"color:#4682B4\">**_taxon_**</span> and each entry has a primary name, secondary names, and a unique taxonomic identifier. The NCBI Taxonomy database is critical to link nucleotides and protein sequences from the International Nucleotide Sequence Database Collaboration (INSDC) and other biological databases which rely on data from INSDC. These linkages can be made using either the organism name or taxonomic ID.\n",
    "\n",
    "The database is provided remotely by the National Center for Biotechnology Information (NCBI). INSDC partners send any requests for new names to NCBI Taxonomy curators before data is released. The database incorporates phylogenetic and taxonomic knowledge from published literature, web-based databases, and the advice of sequence submitters and outside taxonomy experts."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <span style=\"color:#4682B4\">Why NCBI Taxonomy</span>\n",
    "We chose to use the NCBI Taxonomy database because it is the <span style=\"color:#4682B4\">**_sole source for taxonomic classification_**</span> for the INSDC and forms the backbone for many other resources at the NCBI. The NCBI Taxonomy database contains formal and informal organism names and classifications for every sequence in INSDC's datasets (more than 160,000 organisms); these associations between pathogen name and genetic and genomic data are foundational for public health intelligence efforts. The inclusion of informal names also allows us to link pathogens (and their corresponding genetic information) to case reports and other non-traditional data sources which may use names which exist outside of the codes of nomenclature (i.e. \"COVID-19\" instead of SARS-CoV-2). Additionally, more than 150 external partners maintain links to the NCBI Taxonomy database, with specialty datasets of their own."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <span style=\"color:#4682B4\">Accessing the NCBI Taxonomy database</span>\n",
    "\n",
    "There are three methods for accessing the NCBI Taxonomy. First is the NCBI Taxonomy Browser, which is a web-page that allows users to search for organisms, visualize the hierarchy at custom levels of classification, and summarize organism information, such as lineage, in a taxon-specific page. \n",
    "\n",
    "In contrast to the Taxonomy Browser, Entrez supports Boolean queries and common search fields across all NCBI databases. There are also several public APIs that allow programmatic access to the Entrez databases; <span style=\"color:#4682B4\">**_we used E-utilities,_**</span> a suite of server-side programs that accept a fixed URL syntax for search, link, and retrieval. The available E-Utils include: EInfo, ESearch, EPost, ESumary, EFetch, ELink, EGQuery, ESpell, and ECitMatch. While many are tested in this notebook, full documentation is available [here](https://www.ncbi.nlm.nih.gov/books/NBK25500/).\n",
    "\n",
    "A third option is to download the complete database as a full text taxdump file (in .dmp format). While some externally developed packages, including Bio2RDF, use the data dump to formulate global SPARQL queries (transforming the relational database into RDF), it appears that these tools may be out of date. We chose not to utilize the taxdump file to facilitate access because it appears that version control is not well documented and the data may quickly become out of date. \n",
    "\n",
    "The API access via E-Utilities appears to be the most stable option for accessing NCBI data, as of current.\n",
    "\n",
    "This notebook examines the information stored within the NCBI Taxonomy and the quality of metadata that is retrievable via publicly available APIs.\n",
    "\n",
    "![](2022-05-05-14-44-58.png) \n",
    "\n",
    "NCBI Taxonomy Browser page\n",
    "\n",
    "![](2022-05-05-14-46-37.png)\n",
    "\n",
    "NCBI EFetch Utility in Entrez Taxonomy database"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <span style=\"color:gray\">_Extracting data from NCBI Taxonomy database_</span>\n",
    "\n",
    "There are 39 Entrez databases. To return a list of all Entrez database names and identify the one we want to query, we use the following:\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking python executable path\n",
      "/Users/haileyrobertson/Documents/GitHub/kr2-graph/build_graph_2/notebooks/env/bin/python\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>pre { line-height: 125%; }\n",
       "td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       ".highlight .hll { background-color: #ffffcc }\n",
       ".highlight { background: #f8f8f8; }\n",
       ".highlight .c { color: #3D7B7B; font-style: italic } /* Comment */\n",
       ".highlight .err { border: 1px solid #FF0000 } /* Error */\n",
       ".highlight .k { color: #008000; font-weight: bold } /* Keyword */\n",
       ".highlight .o { color: #666666 } /* Operator */\n",
       ".highlight .ch { color: #3D7B7B; font-style: italic } /* Comment.Hashbang */\n",
       ".highlight .cm { color: #3D7B7B; font-style: italic } /* Comment.Multiline */\n",
       ".highlight .cp { color: #9C6500 } /* Comment.Preproc */\n",
       ".highlight .cpf { color: #3D7B7B; font-style: italic } /* Comment.PreprocFile */\n",
       ".highlight .c1 { color: #3D7B7B; font-style: italic } /* Comment.Single */\n",
       ".highlight .cs { color: #3D7B7B; font-style: italic } /* Comment.Special */\n",
       ".highlight .gd { color: #A00000 } /* Generic.Deleted */\n",
       ".highlight .ge { font-style: italic } /* Generic.Emph */\n",
       ".highlight .gr { color: #E40000 } /* Generic.Error */\n",
       ".highlight .gh { color: #000080; font-weight: bold } /* Generic.Heading */\n",
       ".highlight .gi { color: #008400 } /* Generic.Inserted */\n",
       ".highlight .go { color: #717171 } /* Generic.Output */\n",
       ".highlight .gp { color: #000080; font-weight: bold } /* Generic.Prompt */\n",
       ".highlight .gs { font-weight: bold } /* Generic.Strong */\n",
       ".highlight .gu { color: #800080; font-weight: bold } /* Generic.Subheading */\n",
       ".highlight .gt { color: #0044DD } /* Generic.Traceback */\n",
       ".highlight .kc { color: #008000; font-weight: bold } /* Keyword.Constant */\n",
       ".highlight .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */\n",
       ".highlight .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */\n",
       ".highlight .kp { color: #008000 } /* Keyword.Pseudo */\n",
       ".highlight .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */\n",
       ".highlight .kt { color: #B00040 } /* Keyword.Type */\n",
       ".highlight .m { color: #666666 } /* Literal.Number */\n",
       ".highlight .s { color: #BA2121 } /* Literal.String */\n",
       ".highlight .na { color: #687822 } /* Name.Attribute */\n",
       ".highlight .nb { color: #008000 } /* Name.Builtin */\n",
       ".highlight .nc { color: #0000FF; font-weight: bold } /* Name.Class */\n",
       ".highlight .no { color: #880000 } /* Name.Constant */\n",
       ".highlight .nd { color: #AA22FF } /* Name.Decorator */\n",
       ".highlight .ni { color: #717171; font-weight: bold } /* Name.Entity */\n",
       ".highlight .ne { color: #CB3F38; font-weight: bold } /* Name.Exception */\n",
       ".highlight .nf { color: #0000FF } /* Name.Function */\n",
       ".highlight .nl { color: #767600 } /* Name.Label */\n",
       ".highlight .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */\n",
       ".highlight .nt { color: #008000; font-weight: bold } /* Name.Tag */\n",
       ".highlight .nv { color: #19177C } /* Name.Variable */\n",
       ".highlight .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */\n",
       ".highlight .w { color: #bbbbbb } /* Text.Whitespace */\n",
       ".highlight .mb { color: #666666 } /* Literal.Number.Bin */\n",
       ".highlight .mf { color: #666666 } /* Literal.Number.Float */\n",
       ".highlight .mh { color: #666666 } /* Literal.Number.Hex */\n",
       ".highlight .mi { color: #666666 } /* Literal.Number.Integer */\n",
       ".highlight .mo { color: #666666 } /* Literal.Number.Oct */\n",
       ".highlight .sa { color: #BA2121 } /* Literal.String.Affix */\n",
       ".highlight .sb { color: #BA2121 } /* Literal.String.Backtick */\n",
       ".highlight .sc { color: #BA2121 } /* Literal.String.Char */\n",
       ".highlight .dl { color: #BA2121 } /* Literal.String.Delimiter */\n",
       ".highlight .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */\n",
       ".highlight .s2 { color: #BA2121 } /* Literal.String.Double */\n",
       ".highlight .se { color: #AA5D1F; font-weight: bold } /* Literal.String.Escape */\n",
       ".highlight .sh { color: #BA2121 } /* Literal.String.Heredoc */\n",
       ".highlight .si { color: #A45A77; font-weight: bold } /* Literal.String.Interpol */\n",
       ".highlight .sx { color: #008000 } /* Literal.String.Other */\n",
       ".highlight .sr { color: #A45A77 } /* Literal.String.Regex */\n",
       ".highlight .s1 { color: #BA2121 } /* Literal.String.Single */\n",
       ".highlight .ss { color: #19177C } /* Literal.String.Symbol */\n",
       ".highlight .bp { color: #008000 } /* Name.Builtin.Pseudo */\n",
       ".highlight .fm { color: #0000FF } /* Name.Function.Magic */\n",
       ".highlight .vc { color: #19177C } /* Name.Variable.Class */\n",
       ".highlight .vg { color: #19177C } /* Name.Variable.Global */\n",
       ".highlight .vi { color: #19177C } /* Name.Variable.Instance */\n",
       ".highlight .vm { color: #19177C } /* Name.Variable.Magic */\n",
       ".highlight .il { color: #666666 } /* Literal.Number.Integer.Long */</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import sys\n",
    "\n",
    "print(\"Checking python executable path\")\n",
    "print(sys.executable)\n",
    "# append the path of the parent directory to access project packages\n",
    "sys.path.append(\"..\")\n",
    "\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# import project packages\n",
    "import ncbi\n",
    "\n",
    "# Set up notebook code formatting:\n",
    "import inspect\n",
    "from IPython.display import display, Code\n",
    "from pygments.formatters import HtmlFormatter\n",
    "from IPython.core.display import HTML\n",
    "formatter = HtmlFormatter()\n",
    "display(HTML(f'<style>{ formatter.get_style_defs(\".highlight\") }</style>'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "pubmed\n",
      "protein\n",
      "nuccore\n",
      "ipg\n",
      "nucleotide\n",
      "structure\n",
      "genome\n",
      "annotinfo\n",
      "assembly\n",
      "bioproject\n",
      "biosample\n",
      "blastdbinfo\n",
      "books\n",
      "cdd\n",
      "clinvar\n",
      "gap\n",
      "gapplus\n",
      "grasp\n",
      "dbvar\n",
      "gene\n",
      "gds\n",
      "geoprofiles\n",
      "homologene\n",
      "medgen\n",
      "mesh\n",
      "ncbisearch\n",
      "nlmcatalog\n",
      "omim\n",
      "orgtrack\n",
      "pmc\n",
      "popset\n",
      "proteinclusters\n",
      "pcassay\n",
      "protfam\n",
      "pccompound\n",
      "pcsubstance\n",
      "seqannot\n",
      "snp\n",
      "sra\n",
      "taxonomy\n",
      "biocollections\n",
      "gtr\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "entrez_url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi\"\n",
    "entrez_response = requests.get(entrez_url)\n",
    "all_db = BeautifulSoup(entrez_response.content, features=\"xml\").getText()\n",
    "print(all_db)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To gather statistics about the <span style=\"color:#4682B4\">**_taxonomy_**</span> database and look at the available fields, we specify the taxonomy URL using the EInfo utility:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Field Abbreviation</th>\n",
       "      <th>Full Field Name</th>\n",
       "      <th>Field Description</th>\n",
       "      <th>Term Count</th>\n",
       "      <th>Is Date</th>\n",
       "      <th>Is Numerical</th>\n",
       "      <th>Single Token</th>\n",
       "      <th>Hierarchy</th>\n",
       "      <th>Is Hidden</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALL</td>\n",
       "      <td>All Fields</td>\n",
       "      <td>All terms from all searchable fields</td>\n",
       "      <td>36724884</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UID</td>\n",
       "      <td>Taxonomy ID</td>\n",
       "      <td>Unique number assigned to publication</td>\n",
       "      <td>0</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FILT</td>\n",
       "      <td>Filter</td>\n",
       "      <td>Limits the records</td>\n",
       "      <td>303</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SCIN</td>\n",
       "      <td>Scientific Name</td>\n",
       "      <td>Scientific name of organism</td>\n",
       "      <td>2420508</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>COMN</td>\n",
       "      <td>Common Name</td>\n",
       "      <td>Common name of organism</td>\n",
       "      <td>47323</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TXSY</td>\n",
       "      <td>Synonym</td>\n",
       "      <td>Synonym of organism name</td>\n",
       "      <td>221997</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ALLN</td>\n",
       "      <td>All Names</td>\n",
       "      <td>All aliases for organism</td>\n",
       "      <td>3686478</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NXLV</td>\n",
       "      <td>Next Level</td>\n",
       "      <td>Immediate parent in taxonomic hierarchy</td>\n",
       "      <td>373348</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SBTR</td>\n",
       "      <td>Subtree</td>\n",
       "      <td>Any parent node in taxonomic hierarchy</td>\n",
       "      <td>6104404</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LNGE</td>\n",
       "      <td>Lineage</td>\n",
       "      <td>Lineage in taxonomic hierarchy</td>\n",
       "      <td>3686478</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GC</td>\n",
       "      <td>Nuclear Genetic Code</td>\n",
       "      <td>Nuclear genetic code</td>\n",
       "      <td>32</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MGC</td>\n",
       "      <td>Mitochondrial Genetic Code</td>\n",
       "      <td>Mitochondrial genetic code</td>\n",
       "      <td>30</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PGC</td>\n",
       "      <td>Plastid Genetic Code</td>\n",
       "      <td>Plastid genetic code</td>\n",
       "      <td>4</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TXDV</td>\n",
       "      <td>Division</td>\n",
       "      <td>GenBank division</td>\n",
       "      <td>24</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>RANK</td>\n",
       "      <td>Rank</td>\n",
       "      <td>Hierarchical position (e.g., order, genus)</td>\n",
       "      <td>43</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>EDAT</td>\n",
       "      <td>Entrez Date</td>\n",
       "      <td>Date record first accessible through Entrez</td>\n",
       "      <td>9085</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MDAT</td>\n",
       "      <td>Modification Date</td>\n",
       "      <td>Date of last update</td>\n",
       "      <td>6974</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PROP</td>\n",
       "      <td>Properties</td>\n",
       "      <td>Property defined on particular node (e.g., ter...</td>\n",
       "      <td>147</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>WORD</td>\n",
       "      <td>Text Word</td>\n",
       "      <td>Free text associated with record</td>\n",
       "      <td>28258632</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NTOK</td>\n",
       "      <td>Name Tokens</td>\n",
       "      <td>Name tokens associated with organism names</td>\n",
       "      <td>9743538</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HGC</td>\n",
       "      <td>Hydrogenosome Genetic Code</td>\n",
       "      <td>Hydrogenosome genetic code</td>\n",
       "      <td>4</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Field Abbreviation             Full Field Name  \\\n",
       "0                 ALL                  All Fields   \n",
       "1                 UID                 Taxonomy ID   \n",
       "2                FILT                      Filter   \n",
       "3                SCIN             Scientific Name   \n",
       "4                COMN                 Common Name   \n",
       "5                TXSY                     Synonym   \n",
       "6                ALLN                   All Names   \n",
       "7                NXLV                  Next Level   \n",
       "8                SBTR                     Subtree   \n",
       "9                LNGE                     Lineage   \n",
       "10                 GC        Nuclear Genetic Code   \n",
       "11                MGC  Mitochondrial Genetic Code   \n",
       "12                PGC        Plastid Genetic Code   \n",
       "13               TXDV                    Division   \n",
       "14               RANK                        Rank   \n",
       "15               EDAT                 Entrez Date   \n",
       "16               MDAT           Modification Date   \n",
       "17               PROP                  Properties   \n",
       "18               WORD                   Text Word   \n",
       "19               NTOK                 Name Tokens   \n",
       "20                HGC  Hydrogenosome Genetic Code   \n",
       "\n",
       "                                    Field Description Term Count Is Date  \\\n",
       "0                All terms from all searchable fields   36724884       N   \n",
       "1               Unique number assigned to publication          0       N   \n",
       "2                                  Limits the records        303       N   \n",
       "3                         Scientific name of organism    2420508       N   \n",
       "4                             Common name of organism      47323       N   \n",
       "5                            Synonym of organism name     221997       N   \n",
       "6                            All aliases for organism    3686478       N   \n",
       "7             Immediate parent in taxonomic hierarchy     373348       N   \n",
       "8              Any parent node in taxonomic hierarchy    6104404       N   \n",
       "9                      Lineage in taxonomic hierarchy    3686478       N   \n",
       "10                               Nuclear genetic code         32       N   \n",
       "11                         Mitochondrial genetic code         30       N   \n",
       "12                               Plastid genetic code          4       N   \n",
       "13                                   GenBank division         24       N   \n",
       "14         Hierarchical position (e.g., order, genus)         43       N   \n",
       "15        Date record first accessible through Entrez       9085       Y   \n",
       "16                                Date of last update       6974       Y   \n",
       "17  Property defined on particular node (e.g., ter...        147       N   \n",
       "18                   Free text associated with record   28258632       N   \n",
       "19         Name tokens associated with organism names    9743538       N   \n",
       "20                         Hydrogenosome genetic code          4       N   \n",
       "\n",
       "   Is Numerical Single Token Hierarchy Is Hidden  \n",
       "0             N            N         N         N  \n",
       "1             Y            Y         N         Y  \n",
       "2             N            Y         N         N  \n",
       "3             N            Y         N         N  \n",
       "4             N            Y         N         N  \n",
       "5             N            Y         N         N  \n",
       "6             N            Y         N         N  \n",
       "7             N            Y         N         N  \n",
       "8             N            Y         N         N  \n",
       "9             N            Y         N         N  \n",
       "10            N            Y         N         N  \n",
       "11            N            Y         N         N  \n",
       "12            N            Y         N         N  \n",
       "13            N            Y         N         N  \n",
       "14            N            Y         N         N  \n",
       "15            N            Y         N         N  \n",
       "16            N            Y         N         N  \n",
       "17            N            Y         N         N  \n",
       "18            N            Y         N         N  \n",
       "19            N            Y         N         N  \n",
       "20            N            Y         N         N  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "einfo_url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi?db=taxonomy\"\n",
    "einfo_response = requests.get(einfo_url)\n",
    "taxon_info = BeautifulSoup(einfo_response.content, features='xml')\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "# Extracting the field data\n",
    "field_tag = taxon_info.FieldList\n",
    "\n",
    "field = field_tag.find_all('Name')  \n",
    "full_name = field_tag.find_all('FullName')  \n",
    "desc = field_tag.find_all('Description')  \n",
    "term_count = field_tag.find_all('TermCount')\n",
    "is_date = field_tag.find_all('IsDate')\n",
    "is_numerical = field_tag.find_all('IsNumerical')\n",
    "single_token = field_tag.find_all('SingleToken')\n",
    "hierarchy = field_tag.find_all('Hierarchy')\n",
    "is_hidden = field_tag.find_all('IsHidden')\n",
    "\n",
    "field_data = []\n",
    "  \n",
    "# Loop to store the data in a list named 'field_data'\n",
    "for i in range(0, len(field)):\n",
    "    rows = [field[i].get_text(),full_name[i].get_text(),desc[i].get_text(),\n",
    "    term_count[i].get_text(), is_date[i].get_text(), is_numerical[i].get_text(), \n",
    "    single_token[i].get_text(),hierarchy[i].get_text(),is_hidden[i].get_text()]\n",
    "    field_data.append(rows)\n",
    "  \n",
    "# Converting the list into dataframe\n",
    "field_df = pd.DataFrame(field_data, columns=['Field Abbreviation','Full Field Name', 'Field Description',\n",
    "'Term Count', 'Is Date','Is Numerical','Single Token',\n",
    "'Hierarchy','Is Hidden'], dtype = str)\n",
    "\n",
    "display(field_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To learn about other Entrez databases associated with the <span style=\"color:#4682B4\">**_taxonomy_**</span> database entries, we call the EInfo utility again, this time specifying the LinkList:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Link Name</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>taxonomy_assembly_exp</td>\n",
       "      <td>Assembly records associated with taxonomy reco...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>taxonomy_bioproject_exp</td>\n",
       "      <td>BioProject records associated with taxonomy re...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>taxonomy_biosample_exp</td>\n",
       "      <td>BioSample records associated with taxonomy rec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>taxonomy_biosystems_exp</td>\n",
       "      <td>BioSystems records associated with taxonomy re...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>taxonomy_books</td>\n",
       "      <td>Books</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>taxonomy_cdd_exp</td>\n",
       "      <td>CDD records associated with taxonomy record (e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>taxonomy_clone_exp</td>\n",
       "      <td>Links to Clone DB (exploded for higher taxa)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>taxonomy_dbvar_exp</td>\n",
       "      <td>dbVar records associated with taxonomy record ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>taxonomy_gds_exp</td>\n",
       "      <td>GEO DataSet records associated with taxonomy r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>taxonomy_gene_exp</td>\n",
       "      <td>Gene records associated with taxonomy record (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>taxonomy_genome</td>\n",
       "      <td>Genome sequences associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>taxonomy_geoprofiles</td>\n",
       "      <td>GEO records associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>taxonomy_homologene_exp</td>\n",
       "      <td>HomoloGene records associated with taxonomy re...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>taxonomy_mesh</td>\n",
       "      <td>Related MeSH term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>taxonomy_nucest</td>\n",
       "      <td>EST sequences associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>taxonomy_nucgss</td>\n",
       "      <td>GSS sequences associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>taxonomy_nucleotide_exp</td>\n",
       "      <td>Nucleotide sequences associated with taxonomy ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>taxonomy_pcassay</td>\n",
       "      <td>Related PubChem BioAssay</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>taxonomy_pccompound</td>\n",
       "      <td>PubChem Compounds</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>taxonomy_pcsubstance</td>\n",
       "      <td>PubChem Substances</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>taxonomy_pmc</td>\n",
       "      <td>Free full text articles in PMC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>taxonomy_popset</td>\n",
       "      <td>Popset sequences associated with taxonomy record</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>taxonomy_probe_exp</td>\n",
       "      <td>Probe records associated with taxonomy record ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>taxonomy_protein_exp</td>\n",
       "      <td>Protein sequences associated with taxonomy rec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>taxonomy_protein_species2wp</td>\n",
       "      <td>Autonomous proteins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>taxonomy_proteinclusters_exp</td>\n",
       "      <td>Protein Clusters records associated with taxon...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>taxonomy_pubmed</td>\n",
       "      <td>Link to related PubMed articles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>taxonomy_pubmed_entrez</td>\n",
       "      <td>Related PubMed articles computed using other E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>taxonomy_snp_exp</td>\n",
       "      <td>Snp records associated with taxonomy record (e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>taxonomy_sra_exp</td>\n",
       "      <td>Sra records associated with taxonomy record (e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>taxonomy_structure_exp</td>\n",
       "      <td>Structure sequences associated with taxonomy r...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Link Name  \\\n",
       "0          taxonomy_assembly_exp   \n",
       "1        taxonomy_bioproject_exp   \n",
       "2         taxonomy_biosample_exp   \n",
       "3        taxonomy_biosystems_exp   \n",
       "4                 taxonomy_books   \n",
       "5               taxonomy_cdd_exp   \n",
       "6             taxonomy_clone_exp   \n",
       "7             taxonomy_dbvar_exp   \n",
       "8               taxonomy_gds_exp   \n",
       "9              taxonomy_gene_exp   \n",
       "10               taxonomy_genome   \n",
       "11          taxonomy_geoprofiles   \n",
       "12       taxonomy_homologene_exp   \n",
       "13                 taxonomy_mesh   \n",
       "14               taxonomy_nucest   \n",
       "15               taxonomy_nucgss   \n",
       "16       taxonomy_nucleotide_exp   \n",
       "17              taxonomy_pcassay   \n",
       "18           taxonomy_pccompound   \n",
       "19          taxonomy_pcsubstance   \n",
       "20                  taxonomy_pmc   \n",
       "21               taxonomy_popset   \n",
       "22            taxonomy_probe_exp   \n",
       "23          taxonomy_protein_exp   \n",
       "24   taxonomy_protein_species2wp   \n",
       "25  taxonomy_proteinclusters_exp   \n",
       "26               taxonomy_pubmed   \n",
       "27        taxonomy_pubmed_entrez   \n",
       "28              taxonomy_snp_exp   \n",
       "29              taxonomy_sra_exp   \n",
       "30        taxonomy_structure_exp   \n",
       "\n",
       "                                          Description  \n",
       "0   Assembly records associated with taxonomy reco...  \n",
       "1   BioProject records associated with taxonomy re...  \n",
       "2   BioSample records associated with taxonomy rec...  \n",
       "3   BioSystems records associated with taxonomy re...  \n",
       "4                                               Books  \n",
       "5   CDD records associated with taxonomy record (e...  \n",
       "6        Links to Clone DB (exploded for higher taxa)  \n",
       "7   dbVar records associated with taxonomy record ...  \n",
       "8   GEO DataSet records associated with taxonomy r...  \n",
       "9   Gene records associated with taxonomy record (...  \n",
       "10   Genome sequences associated with taxonomy record  \n",
       "11        GEO records associated with taxonomy record  \n",
       "12  HomoloGene records associated with taxonomy re...  \n",
       "13                                  Related MeSH term  \n",
       "14      EST sequences associated with taxonomy record  \n",
       "15      GSS sequences associated with taxonomy record  \n",
       "16  Nucleotide sequences associated with taxonomy ...  \n",
       "17                           Related PubChem BioAssay  \n",
       "18                                  PubChem Compounds  \n",
       "19                                 PubChem Substances  \n",
       "20                     Free full text articles in PMC  \n",
       "21   Popset sequences associated with taxonomy record  \n",
       "22  Probe records associated with taxonomy record ...  \n",
       "23  Protein sequences associated with taxonomy rec...  \n",
       "24                                Autonomous proteins  \n",
       "25  Protein Clusters records associated with taxon...  \n",
       "26                    Link to related PubMed articles  \n",
       "27  Related PubMed articles computed using other E...  \n",
       "28  Snp records associated with taxonomy record (e...  \n",
       "29  Sra records associated with taxonomy record (e...  \n",
       "30  Structure sequences associated with taxonomy r...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Extracting the associated links\n",
    "link_tag = taxon_info.LinkList\n",
    "\n",
    "link_name = link_tag.find_all('Name')  \n",
    "link_desc = link_tag.find_all('Description')  \n",
    "\n",
    "link_data = []\n",
    "  \n",
    "# Loop to store the data in a list named 'link_data'\n",
    "for i in range(0, len(link_name)):\n",
    "    link_rows = [link_name[i].get_text(), link_desc[i].get_text()]\n",
    "    link_data.append(link_rows)\n",
    "  \n",
    "# Converting the list into dataframe\n",
    "link_df = pd.DataFrame(link_data, columns=['Link Name', 'Description'], dtype = str)\n",
    "\n",
    "display(link_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To extract data about organisms from these fields, we created a Python package <span style=\"color:#4682B4\">**_ncbi._**</span> This module is composed of three main functions demonstrated below, <span style=\"color:#4682B4\">**_api_soup_**</span>, <span style=\"color:#4682B4\">**_id_search_**</span>, and <span style=\"color:#4682B4\">**_get_metadata._**</span>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>pre { line-height: 125%; }\n",
       "td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       ".output_html .hll { background-color: #ffffcc }\n",
       ".output_html { background: #f8f8f8; }\n",
       ".output_html .c { color: #3D7B7B; font-style: italic } /* Comment */\n",
       ".output_html .err { border: 1px solid #FF0000 } /* Error */\n",
       ".output_html .k { color: #008000; font-weight: bold } /* Keyword */\n",
       ".output_html .o { color: #666666 } /* Operator */\n",
       ".output_html .ch { color: #3D7B7B; font-style: italic } /* Comment.Hashbang */\n",
       ".output_html .cm { color: #3D7B7B; font-style: italic } /* Comment.Multiline */\n",
       ".output_html .cp { color: #9C6500 } /* Comment.Preproc */\n",
       ".output_html .cpf { color: #3D7B7B; font-style: italic } /* Comment.PreprocFile */\n",
       ".output_html .c1 { color: #3D7B7B; font-style: italic } /* Comment.Single */\n",
       ".output_html .cs { color: #3D7B7B; font-style: italic } /* Comment.Special */\n",
       ".output_html .gd { color: #A00000 } /* Generic.Deleted */\n",
       ".output_html .ge { font-style: italic } /* Generic.Emph */\n",
       ".output_html .gr { color: #E40000 } /* Generic.Error */\n",
       ".output_html .gh { color: #000080; font-weight: bold } /* Generic.Heading */\n",
       ".output_html .gi { color: #008400 } /* Generic.Inserted */\n",
       ".output_html .go { color: #717171 } /* Generic.Output */\n",
       ".output_html .gp { color: #000080; font-weight: bold } /* Generic.Prompt */\n",
       ".output_html .gs { font-weight: bold } /* Generic.Strong */\n",
       ".output_html .gu { color: #800080; font-weight: bold } /* Generic.Subheading */\n",
       ".output_html .gt { color: #0044DD } /* Generic.Traceback */\n",
       ".output_html .kc { color: #008000; font-weight: bold } /* Keyword.Constant */\n",
       ".output_html .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */\n",
       ".output_html .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */\n",
       ".output_html .kp { color: #008000 } /* Keyword.Pseudo */\n",
       ".output_html .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */\n",
       ".output_html .kt { color: #B00040 } /* Keyword.Type */\n",
       ".output_html .m { color: #666666 } /* Literal.Number */\n",
       ".output_html .s { color: #BA2121 } /* Literal.String */\n",
       ".output_html .na { color: #687822 } /* Name.Attribute */\n",
       ".output_html .nb { color: #008000 } /* Name.Builtin */\n",
       ".output_html .nc { color: #0000FF; font-weight: bold } /* Name.Class */\n",
       ".output_html .no { color: #880000 } /* Name.Constant */\n",
       ".output_html .nd { color: #AA22FF } /* Name.Decorator */\n",
       ".output_html .ni { color: #717171; font-weight: bold } /* Name.Entity */\n",
       ".output_html .ne { color: #CB3F38; font-weight: bold } /* Name.Exception */\n",
       ".output_html .nf { color: #0000FF } /* Name.Function */\n",
       ".output_html .nl { color: #767600 } /* Name.Label */\n",
       ".output_html .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */\n",
       ".output_html .nt { color: #008000; font-weight: bold } /* Name.Tag */\n",
       ".output_html .nv { color: #19177C } /* Name.Variable */\n",
       ".output_html .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */\n",
       ".output_html .w { color: #bbbbbb } /* Text.Whitespace */\n",
       ".output_html .mb { color: #666666 } /* Literal.Number.Bin */\n",
       ".output_html .mf { color: #666666 } /* Literal.Number.Float */\n",
       ".output_html .mh { color: #666666 } /* Literal.Number.Hex */\n",
       ".output_html .mi { color: #666666 } /* Literal.Number.Integer */\n",
       ".output_html .mo { color: #666666 } /* Literal.Number.Oct */\n",
       ".output_html .sa { color: #BA2121 } /* Literal.String.Affix */\n",
       ".output_html .sb { color: #BA2121 } /* Literal.String.Backtick */\n",
       ".output_html .sc { color: #BA2121 } /* Literal.String.Char */\n",
       ".output_html .dl { color: #BA2121 } /* Literal.String.Delimiter */\n",
       ".output_html .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */\n",
       ".output_html .s2 { color: #BA2121 } /* Literal.String.Double */\n",
       ".output_html .se { color: #AA5D1F; font-weight: bold } /* Literal.String.Escape */\n",
       ".output_html .sh { color: #BA2121 } /* Literal.String.Heredoc */\n",
       ".output_html .si { color: #A45A77; font-weight: bold } /* Literal.String.Interpol */\n",
       ".output_html .sx { color: #008000 } /* Literal.String.Other */\n",
       ".output_html .sr { color: #A45A77 } /* Literal.String.Regex */\n",
       ".output_html .s1 { color: #BA2121 } /* Literal.String.Single */\n",
       ".output_html .ss { color: #19177C } /* Literal.String.Symbol */\n",
       ".output_html .bp { color: #008000 } /* Name.Builtin.Pseudo */\n",
       ".output_html .fm { color: #0000FF } /* Name.Function.Magic */\n",
       ".output_html .vc { color: #19177C } /* Name.Variable.Class */\n",
       ".output_html .vg { color: #19177C } /* Name.Variable.Global */\n",
       ".output_html .vi { color: #19177C } /* Name.Variable.Instance */\n",
       ".output_html .vm { color: #19177C } /* Name.Variable.Magic */\n",
       ".output_html .il { color: #666666 } /* Literal.Number.Integer.Long */</style><div class=\"highlight\"><pre><span></span><span class=\"k\">def</span> <span class=\"nf\">api_soup</span><span class=\"p\">(</span><span class=\"n\">eutil</span><span class=\"p\">,</span> <span class=\"n\">params</span><span class=\"p\">):</span>\n",
       "    <span class=\"sd\">&quot;&quot;&quot;</span>\n",
       "<span class=\"sd\">    Retrieve NCBI Eutils response XML, and</span>\n",
       "<span class=\"sd\">    parse it into a beautifulsoup object</span>\n",
       "<span class=\"sd\">    &quot;&quot;&quot;</span>\n",
       "\n",
       "    <span class=\"n\">url</span> <span class=\"o\">=</span> <span class=\"sa\">f</span><span class=\"s2\">&quot;https://eutils.ncbi.nlm.nih.gov/entrez/eutils/</span><span class=\"si\">{</span><span class=\"n\">eutil</span><span class=\"si\">}</span><span class=\"s2\">.fcgi&quot;</span>\n",
       "    <span class=\"n\">response</span> <span class=\"o\">=</span> <span class=\"n\">requests</span><span class=\"o\">.</span><span class=\"n\">get</span><span class=\"p\">(</span><span class=\"n\">url</span><span class=\"p\">,</span> <span class=\"n\">params</span><span class=\"p\">)</span>\n",
       "    <span class=\"n\">soup</span> <span class=\"o\">=</span> <span class=\"n\">BeautifulSoup</span><span class=\"p\">(</span><span class=\"n\">response</span><span class=\"o\">.</span><span class=\"n\">content</span><span class=\"p\">,</span> <span class=\"n\">features</span><span class=\"o\">=</span><span class=\"s2\">&quot;xml&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"k\">return</span> <span class=\"n\">soup</span>\n",
       "</pre></div>\n"
      ],
      "text/latex": [
       "\\begin{Verbatim}[commandchars=\\\\\\{\\}]\n",
       "\\PY{k}{def} \\PY{n+nf}{api\\PYZus{}soup}\\PY{p}{(}\\PY{n}{eutil}\\PY{p}{,} \\PY{n}{params}\\PY{p}{)}\\PY{p}{:}\n",
       "    \\PY{l+s+sd}{\\PYZdq{}\\PYZdq{}\\PYZdq{}}\n",
       "\\PY{l+s+sd}{    Retrieve NCBI Eutils response XML, and}\n",
       "\\PY{l+s+sd}{    parse it into a beautifulsoup object}\n",
       "\\PY{l+s+sd}{    \\PYZdq{}\\PYZdq{}\\PYZdq{}}\n",
       "\n",
       "    \\PY{n}{url} \\PY{o}{=} \\PY{l+s+sa}{f}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{https://eutils.ncbi.nlm.nih.gov/entrez/eutils/}\\PY{l+s+si}{\\PYZob{}}\\PY{n}{eutil}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{.fcgi}\\PY{l+s+s2}{\\PYZdq{}}\n",
       "    \\PY{n}{response} \\PY{o}{=} \\PY{n}{requests}\\PY{o}{.}\\PY{n}{get}\\PY{p}{(}\\PY{n}{url}\\PY{p}{,} \\PY{n}{params}\\PY{p}{)}\n",
       "    \\PY{n}{soup} \\PY{o}{=} \\PY{n}{BeautifulSoup}\\PY{p}{(}\\PY{n}{response}\\PY{o}{.}\\PY{n}{content}\\PY{p}{,} \\PY{n}{features}\\PY{o}{=}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{xml}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "    \\PY{k}{return} \\PY{n}{soup}\n",
       "\\end{Verbatim}\n"
      ],
      "text/plain": [
       "def api_soup(eutil, params):\n",
       "    \"\"\"\n",
       "    Retrieve NCBI Eutils response XML, and\n",
       "    parse it into a beautifulsoup object\n",
       "    \"\"\"\n",
       "\n",
       "    url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/{eutil}.fcgi\"\n",
       "    response = requests.get(url, params)\n",
       "    soup = BeautifulSoup(response.content, features=\"xml\")\n",
       "\n",
       "    return soup"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Code(inspect.getsource(ncbi.api_soup), language='python')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>pre { line-height: 125%; }\n",
       "td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       ".output_html .hll { background-color: #ffffcc }\n",
       ".output_html { background: #f8f8f8; }\n",
       ".output_html .c { color: #3D7B7B; font-style: italic } /* Comment */\n",
       ".output_html .err { border: 1px solid #FF0000 } /* Error */\n",
       ".output_html .k { color: #008000; font-weight: bold } /* Keyword */\n",
       ".output_html .o { color: #666666 } /* Operator */\n",
       ".output_html .ch { color: #3D7B7B; font-style: italic } /* Comment.Hashbang */\n",
       ".output_html .cm { color: #3D7B7B; font-style: italic } /* Comment.Multiline */\n",
       ".output_html .cp { color: #9C6500 } /* Comment.Preproc */\n",
       ".output_html .cpf { color: #3D7B7B; font-style: italic } /* Comment.PreprocFile */\n",
       ".output_html .c1 { color: #3D7B7B; font-style: italic } /* Comment.Single */\n",
       ".output_html .cs { color: #3D7B7B; font-style: italic } /* Comment.Special */\n",
       ".output_html .gd { color: #A00000 } /* Generic.Deleted */\n",
       ".output_html .ge { font-style: italic } /* Generic.Emph */\n",
       ".output_html .gr { color: #E40000 } /* Generic.Error */\n",
       ".output_html .gh { color: #000080; font-weight: bold } /* Generic.Heading */\n",
       ".output_html .gi { color: #008400 } /* Generic.Inserted */\n",
       ".output_html .go { color: #717171 } /* Generic.Output */\n",
       ".output_html .gp { color: #000080; font-weight: bold } /* Generic.Prompt */\n",
       ".output_html .gs { font-weight: bold } /* Generic.Strong */\n",
       ".output_html .gu { color: #800080; font-weight: bold } /* Generic.Subheading */\n",
       ".output_html .gt { color: #0044DD } /* Generic.Traceback */\n",
       ".output_html .kc { color: #008000; font-weight: bold } /* Keyword.Constant */\n",
       ".output_html .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */\n",
       ".output_html .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */\n",
       ".output_html .kp { color: #008000 } /* Keyword.Pseudo */\n",
       ".output_html .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */\n",
       ".output_html .kt { color: #B00040 } /* Keyword.Type */\n",
       ".output_html .m { color: #666666 } /* Literal.Number */\n",
       ".output_html .s { color: #BA2121 } /* Literal.String */\n",
       ".output_html .na { color: #687822 } /* Name.Attribute */\n",
       ".output_html .nb { color: #008000 } /* Name.Builtin */\n",
       ".output_html .nc { color: #0000FF; font-weight: bold } /* Name.Class */\n",
       ".output_html .no { color: #880000 } /* Name.Constant */\n",
       ".output_html .nd { color: #AA22FF } /* Name.Decorator */\n",
       ".output_html .ni { color: #717171; font-weight: bold } /* Name.Entity */\n",
       ".output_html .ne { color: #CB3F38; font-weight: bold } /* Name.Exception */\n",
       ".output_html .nf { color: #0000FF } /* Name.Function */\n",
       ".output_html .nl { color: #767600 } /* Name.Label */\n",
       ".output_html .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */\n",
       ".output_html .nt { color: #008000; font-weight: bold } /* Name.Tag */\n",
       ".output_html .nv { color: #19177C } /* Name.Variable */\n",
       ".output_html .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */\n",
       ".output_html .w { color: #bbbbbb } /* Text.Whitespace */\n",
       ".output_html .mb { color: #666666 } /* Literal.Number.Bin */\n",
       ".output_html .mf { color: #666666 } /* Literal.Number.Float */\n",
       ".output_html .mh { color: #666666 } /* Literal.Number.Hex */\n",
       ".output_html .mi { color: #666666 } /* Literal.Number.Integer */\n",
       ".output_html .mo { color: #666666 } /* Literal.Number.Oct */\n",
       ".output_html .sa { color: #BA2121 } /* Literal.String.Affix */\n",
       ".output_html .sb { color: #BA2121 } /* Literal.String.Backtick */\n",
       ".output_html .sc { color: #BA2121 } /* Literal.String.Char */\n",
       ".output_html .dl { color: #BA2121 } /* Literal.String.Delimiter */\n",
       ".output_html .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */\n",
       ".output_html .s2 { color: #BA2121 } /* Literal.String.Double */\n",
       ".output_html .se { color: #AA5D1F; font-weight: bold } /* Literal.String.Escape */\n",
       ".output_html .sh { color: #BA2121 } /* Literal.String.Heredoc */\n",
       ".output_html .si { color: #A45A77; font-weight: bold } /* Literal.String.Interpol */\n",
       ".output_html .sx { color: #008000 } /* Literal.String.Other */\n",
       ".output_html .sr { color: #A45A77 } /* Literal.String.Regex */\n",
       ".output_html .s1 { color: #BA2121 } /* Literal.String.Single */\n",
       ".output_html .ss { color: #19177C } /* Literal.String.Symbol */\n",
       ".output_html .bp { color: #008000 } /* Name.Builtin.Pseudo */\n",
       ".output_html .fm { color: #0000FF } /* Name.Function.Magic */\n",
       ".output_html .vc { color: #19177C } /* Name.Variable.Class */\n",
       ".output_html .vg { color: #19177C } /* Name.Variable.Global */\n",
       ".output_html .vi { color: #19177C } /* Name.Variable.Instance */\n",
       ".output_html .vm { color: #19177C } /* Name.Variable.Magic */\n",
       ".output_html .il { color: #666666 } /* Literal.Number.Integer.Long */</style><div class=\"highlight\"><pre><span></span><span class=\"k\">def</span> <span class=\"nf\">id_search</span><span class=\"p\">(</span><span class=\"n\">name</span><span class=\"p\">):</span>\n",
       "    <span class=\"sd\">&quot;&quot;&quot;Get ID from text search, using NCBI esearch eutil&quot;&quot;&quot;</span>\n",
       "\n",
       "    <span class=\"n\">logger</span><span class=\"o\">.</span><span class=\"n\">info</span><span class=\"p\">(</span><span class=\"sa\">f</span><span class=\"s2\">&quot;Searching ncbi for term </span><span class=\"si\">{</span><span class=\"n\">name</span><span class=\"si\">}</span><span class=\"s2\">&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"n\">params</span> <span class=\"o\">=</span> <span class=\"p\">{</span><span class=\"s2\">&quot;db&quot;</span><span class=\"p\">:</span> <span class=\"s2\">&quot;Taxonomy&quot;</span><span class=\"p\">,</span> <span class=\"s2\">&quot;term&quot;</span><span class=\"p\">:</span> <span class=\"n\">name</span><span class=\"p\">}</span>\n",
       "\n",
       "    <span class=\"n\">soup</span> <span class=\"o\">=</span> <span class=\"n\">ncbi</span><span class=\"o\">.</span><span class=\"n\">api_soup</span><span class=\"p\">(</span><span class=\"s2\">&quot;esearch&quot;</span><span class=\"p\">,</span> <span class=\"n\">params</span><span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"k\">try</span><span class=\"p\">:</span>\n",
       "        <span class=\"n\">ncbi_id</span> <span class=\"o\">=</span> <span class=\"n\">soup</span><span class=\"o\">.</span><span class=\"n\">find</span><span class=\"p\">(</span><span class=\"s2\">&quot;Id&quot;</span><span class=\"p\">)</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">()</span>\n",
       "\n",
       "    <span class=\"k\">except</span> <span class=\"ne\">AttributeError</span><span class=\"p\">:</span>\n",
       "        <span class=\"n\">errors</span> <span class=\"o\">=</span> <span class=\"n\">soup</span><span class=\"o\">.</span><span class=\"n\">find</span><span class=\"p\">(</span><span class=\"s2\">&quot;ErrorList&quot;</span><span class=\"p\">)</span>\n",
       "        <span class=\"n\">warnings</span> <span class=\"o\">=</span> <span class=\"n\">soup</span><span class=\"o\">.</span><span class=\"n\">find</span><span class=\"p\">(</span><span class=\"s2\">&quot;WarningList&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "        <span class=\"k\">for</span> <span class=\"n\">error</span> <span class=\"ow\">in</span> <span class=\"n\">errors</span><span class=\"o\">.</span><span class=\"n\">children</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">logger</span><span class=\"o\">.</span><span class=\"n\">error</span><span class=\"p\">(</span><span class=\"sa\">f</span><span class=\"s2\">&quot;</span><span class=\"si\">{</span><span class=\"n\">error</span><span class=\"o\">.</span><span class=\"n\">name</span><span class=\"si\">}</span><span class=\"s2\">: </span><span class=\"si\">{</span><span class=\"n\">error</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">()</span><span class=\"si\">}</span><span class=\"s2\">&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "        <span class=\"k\">for</span> <span class=\"n\">warning</span> <span class=\"ow\">in</span> <span class=\"n\">warnings</span><span class=\"o\">.</span><span class=\"n\">children</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">logger</span><span class=\"o\">.</span><span class=\"n\">warning</span><span class=\"p\">(</span><span class=\"sa\">f</span><span class=\"s2\">&quot;</span><span class=\"si\">{</span><span class=\"n\">warning</span><span class=\"o\">.</span><span class=\"n\">name</span><span class=\"si\">}</span><span class=\"s2\">: </span><span class=\"si\">{</span><span class=\"n\">warning</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">()</span><span class=\"si\">}</span><span class=\"s2\">&quot;</span><span class=\"p\">)</span>\n",
       "\n",
       "        <span class=\"k\">return</span> <span class=\"kc\">None</span>\n",
       "\n",
       "    <span class=\"k\">return</span> <span class=\"n\">ncbi_id</span>\n",
       "</pre></div>\n"
      ],
      "text/latex": [
       "\\begin{Verbatim}[commandchars=\\\\\\{\\}]\n",
       "\\PY{k}{def} \\PY{n+nf}{id\\PYZus{}search}\\PY{p}{(}\\PY{n}{name}\\PY{p}{)}\\PY{p}{:}\n",
       "    \\PY{l+s+sd}{\\PYZdq{}\\PYZdq{}\\PYZdq{}Get ID from text search, using NCBI esearch eutil\\PYZdq{}\\PYZdq{}\\PYZdq{}}\n",
       "\n",
       "    \\PY{n}{logger}\\PY{o}{.}\\PY{n}{info}\\PY{p}{(}\\PY{l+s+sa}{f}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Searching ncbi for term }\\PY{l+s+si}{\\PYZob{}}\\PY{n}{name}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "    \\PY{n}{params} \\PY{o}{=} \\PY{p}{\\PYZob{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{db}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Taxonomy}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{,} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{term}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{name}\\PY{p}{\\PYZcb{}}\n",
       "\n",
       "    \\PY{n}{soup} \\PY{o}{=} \\PY{n}{ncbi}\\PY{o}{.}\\PY{n}{api\\PYZus{}soup}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{esearch}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{,} \\PY{n}{params}\\PY{p}{)}\n",
       "\n",
       "    \\PY{k}{try}\\PY{p}{:}\n",
       "        \\PY{n}{ncbi\\PYZus{}id} \\PY{o}{=} \\PY{n}{soup}\\PY{o}{.}\\PY{n}{find}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Id}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\n",
       "\n",
       "    \\PY{k}{except} \\PY{n+ne}{AttributeError}\\PY{p}{:}\n",
       "        \\PY{n}{errors} \\PY{o}{=} \\PY{n}{soup}\\PY{o}{.}\\PY{n}{find}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{ErrorList}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "        \\PY{n}{warnings} \\PY{o}{=} \\PY{n}{soup}\\PY{o}{.}\\PY{n}{find}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{WarningList}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "        \\PY{k}{for} \\PY{n}{error} \\PY{o+ow}{in} \\PY{n}{errors}\\PY{o}{.}\\PY{n}{children}\\PY{p}{:}\n",
       "            \\PY{n}{logger}\\PY{o}{.}\\PY{n}{error}\\PY{p}{(}\\PY{l+s+sa}{f}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+si}{\\PYZob{}}\\PY{n}{error}\\PY{o}{.}\\PY{n}{name}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{: }\\PY{l+s+si}{\\PYZob{}}\\PY{n}{error}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "        \\PY{k}{for} \\PY{n}{warning} \\PY{o+ow}{in} \\PY{n}{warnings}\\PY{o}{.}\\PY{n}{children}\\PY{p}{:}\n",
       "            \\PY{n}{logger}\\PY{o}{.}\\PY{n}{warning}\\PY{p}{(}\\PY{l+s+sa}{f}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+si}{\\PYZob{}}\\PY{n}{warning}\\PY{o}{.}\\PY{n}{name}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{: }\\PY{l+s+si}{\\PYZob{}}\\PY{n}{warning}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{l+s+si}{\\PYZcb{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{)}\n",
       "\n",
       "        \\PY{k}{return} \\PY{k+kc}{None}\n",
       "\n",
       "    \\PY{k}{return} \\PY{n}{ncbi\\PYZus{}id}\n",
       "\\end{Verbatim}\n"
      ],
      "text/plain": [
       "def id_search(name):\n",
       "    \"\"\"Get ID from text search, using NCBI esearch eutil\"\"\"\n",
       "\n",
       "    logger.info(f\"Searching ncbi for term {name}\")\n",
       "\n",
       "    params = {\"db\": \"Taxonomy\", \"term\": name}\n",
       "\n",
       "    soup = ncbi.api_soup(\"esearch\", params)\n",
       "\n",
       "    try:\n",
       "        ncbi_id = soup.find(\"Id\").getText()\n",
       "\n",
       "    except AttributeError:\n",
       "        errors = soup.find(\"ErrorList\")\n",
       "        warnings = soup.find(\"WarningList\")\n",
       "\n",
       "        for error in errors.children:\n",
       "            logger.error(f\"{error.name}: {error.getText()}\")\n",
       "\n",
       "        for warning in warnings.children:\n",
       "            logger.warning(f\"{warning.name}: {warning.getText()}\")\n",
       "\n",
       "        return None\n",
       "\n",
       "    return ncbi_id"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Code(inspect.getsource(ncbi.id_search), language='python')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>pre { line-height: 125%; }\n",
       "td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }\n",
       "td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       "span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }\n",
       ".output_html .hll { background-color: #ffffcc }\n",
       ".output_html { background: #f8f8f8; }\n",
       ".output_html .c { color: #3D7B7B; font-style: italic } /* Comment */\n",
       ".output_html .err { border: 1px solid #FF0000 } /* Error */\n",
       ".output_html .k { color: #008000; font-weight: bold } /* Keyword */\n",
       ".output_html .o { color: #666666 } /* Operator */\n",
       ".output_html .ch { color: #3D7B7B; font-style: italic } /* Comment.Hashbang */\n",
       ".output_html .cm { color: #3D7B7B; font-style: italic } /* Comment.Multiline */\n",
       ".output_html .cp { color: #9C6500 } /* Comment.Preproc */\n",
       ".output_html .cpf { color: #3D7B7B; font-style: italic } /* Comment.PreprocFile */\n",
       ".output_html .c1 { color: #3D7B7B; font-style: italic } /* Comment.Single */\n",
       ".output_html .cs { color: #3D7B7B; font-style: italic } /* Comment.Special */\n",
       ".output_html .gd { color: #A00000 } /* Generic.Deleted */\n",
       ".output_html .ge { font-style: italic } /* Generic.Emph */\n",
       ".output_html .gr { color: #E40000 } /* Generic.Error */\n",
       ".output_html .gh { color: #000080; font-weight: bold } /* Generic.Heading */\n",
       ".output_html .gi { color: #008400 } /* Generic.Inserted */\n",
       ".output_html .go { color: #717171 } /* Generic.Output */\n",
       ".output_html .gp { color: #000080; font-weight: bold } /* Generic.Prompt */\n",
       ".output_html .gs { font-weight: bold } /* Generic.Strong */\n",
       ".output_html .gu { color: #800080; font-weight: bold } /* Generic.Subheading */\n",
       ".output_html .gt { color: #0044DD } /* Generic.Traceback */\n",
       ".output_html .kc { color: #008000; font-weight: bold } /* Keyword.Constant */\n",
       ".output_html .kd { color: #008000; font-weight: bold } /* Keyword.Declaration */\n",
       ".output_html .kn { color: #008000; font-weight: bold } /* Keyword.Namespace */\n",
       ".output_html .kp { color: #008000 } /* Keyword.Pseudo */\n",
       ".output_html .kr { color: #008000; font-weight: bold } /* Keyword.Reserved */\n",
       ".output_html .kt { color: #B00040 } /* Keyword.Type */\n",
       ".output_html .m { color: #666666 } /* Literal.Number */\n",
       ".output_html .s { color: #BA2121 } /* Literal.String */\n",
       ".output_html .na { color: #687822 } /* Name.Attribute */\n",
       ".output_html .nb { color: #008000 } /* Name.Builtin */\n",
       ".output_html .nc { color: #0000FF; font-weight: bold } /* Name.Class */\n",
       ".output_html .no { color: #880000 } /* Name.Constant */\n",
       ".output_html .nd { color: #AA22FF } /* Name.Decorator */\n",
       ".output_html .ni { color: #717171; font-weight: bold } /* Name.Entity */\n",
       ".output_html .ne { color: #CB3F38; font-weight: bold } /* Name.Exception */\n",
       ".output_html .nf { color: #0000FF } /* Name.Function */\n",
       ".output_html .nl { color: #767600 } /* Name.Label */\n",
       ".output_html .nn { color: #0000FF; font-weight: bold } /* Name.Namespace */\n",
       ".output_html .nt { color: #008000; font-weight: bold } /* Name.Tag */\n",
       ".output_html .nv { color: #19177C } /* Name.Variable */\n",
       ".output_html .ow { color: #AA22FF; font-weight: bold } /* Operator.Word */\n",
       ".output_html .w { color: #bbbbbb } /* Text.Whitespace */\n",
       ".output_html .mb { color: #666666 } /* Literal.Number.Bin */\n",
       ".output_html .mf { color: #666666 } /* Literal.Number.Float */\n",
       ".output_html .mh { color: #666666 } /* Literal.Number.Hex */\n",
       ".output_html .mi { color: #666666 } /* Literal.Number.Integer */\n",
       ".output_html .mo { color: #666666 } /* Literal.Number.Oct */\n",
       ".output_html .sa { color: #BA2121 } /* Literal.String.Affix */\n",
       ".output_html .sb { color: #BA2121 } /* Literal.String.Backtick */\n",
       ".output_html .sc { color: #BA2121 } /* Literal.String.Char */\n",
       ".output_html .dl { color: #BA2121 } /* Literal.String.Delimiter */\n",
       ".output_html .sd { color: #BA2121; font-style: italic } /* Literal.String.Doc */\n",
       ".output_html .s2 { color: #BA2121 } /* Literal.String.Double */\n",
       ".output_html .se { color: #AA5D1F; font-weight: bold } /* Literal.String.Escape */\n",
       ".output_html .sh { color: #BA2121 } /* Literal.String.Heredoc */\n",
       ".output_html .si { color: #A45A77; font-weight: bold } /* Literal.String.Interpol */\n",
       ".output_html .sx { color: #008000 } /* Literal.String.Other */\n",
       ".output_html .sr { color: #A45A77 } /* Literal.String.Regex */\n",
       ".output_html .s1 { color: #BA2121 } /* Literal.String.Single */\n",
       ".output_html .ss { color: #19177C } /* Literal.String.Symbol */\n",
       ".output_html .bp { color: #008000 } /* Name.Builtin.Pseudo */\n",
       ".output_html .fm { color: #0000FF } /* Name.Function.Magic */\n",
       ".output_html .vc { color: #19177C } /* Name.Variable.Class */\n",
       ".output_html .vg { color: #19177C } /* Name.Variable.Global */\n",
       ".output_html .vi { color: #19177C } /* Name.Variable.Instance */\n",
       ".output_html .vm { color: #19177C } /* Name.Variable.Magic */\n",
       ".output_html .il { color: #666666 } /* Literal.Number.Integer.Long */</style><div class=\"highlight\"><pre><span></span><span class=\"k\">def</span> <span class=\"nf\">get_metadata</span><span class=\"p\">(</span><span class=\"n\">ncbi_id</span><span class=\"p\">):</span>\n",
       "    <span class=\"sd\">&quot;&quot;&quot;Request metadata by NCBI taxonomy ID, and return cleaned object&quot;&quot;&quot;</span>\n",
       "\n",
       "    <span class=\"n\">params</span> <span class=\"o\">=</span> <span class=\"p\">{</span><span class=\"s2\">&quot;db&quot;</span><span class=\"p\">:</span> <span class=\"s2\">&quot;Taxonomy&quot;</span><span class=\"p\">,</span> <span class=\"s2\">&quot;id&quot;</span><span class=\"p\">:</span> <span class=\"n\">ncbi_id</span><span class=\"p\">}</span>\n",
       "    <span class=\"n\">soup</span> <span class=\"o\">=</span> <span class=\"n\">ncbi</span><span class=\"o\">.</span><span class=\"n\">api_soup</span><span class=\"p\">(</span><span class=\"s2\">&quot;efetch&quot;</span><span class=\"p\">,</span> <span class=\"n\">params</span><span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"n\">taxon</span> <span class=\"o\">=</span> <span class=\"n\">soup</span><span class=\"o\">.</span><span class=\"n\">TaxaSet</span><span class=\"o\">.</span><span class=\"n\">Taxon</span>\n",
       "\n",
       "    <span class=\"n\">taxon_metadata</span> <span class=\"o\">=</span> <span class=\"p\">{</span>\n",
       "        <span class=\"s2\">&quot;ScientificName&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">ScientificName</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;ParentTaxId&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">ParentTaxId</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;Rank&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">Rank</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;Division&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">Division</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;GeneticCode&quot;</span><span class=\"p\">:</span> <span class=\"p\">{</span><span class=\"s2\">&quot;GCId&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">GCId</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span> <span class=\"s2\">&quot;GCName&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">GCName</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">()},</span>\n",
       "        <span class=\"s2\">&quot;MitoGeneticCode&quot;</span><span class=\"p\">:</span> <span class=\"p\">{</span>\n",
       "            <span class=\"s2\">&quot;MGCId&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">MGCId</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "            <span class=\"s2\">&quot;MGCName&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">MGCName</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"p\">},</span>\n",
       "        <span class=\"s2\">&quot;Lineage&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">Lineage</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;CreateDate&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">CreateDate</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;UpdateDate&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">UpdateDate</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "        <span class=\"s2\">&quot;PubDate&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">PubDate</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "    <span class=\"p\">}</span>\n",
       "\n",
       "    <span class=\"k\">if</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">otherNames</span><span class=\"p\">:</span>\n",
       "        <span class=\"n\">taxon</span><span class=\"p\">[</span><span class=\"s2\">&quot;OtherNames&quot;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"p\">(</span><span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">OtherNames</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),)</span>\n",
       "\n",
       "    <span class=\"c1\"># parse lineage</span>\n",
       "    <span class=\"n\">lineage_ex</span> <span class=\"o\">=</span> <span class=\"p\">[]</span>\n",
       "    <span class=\"k\">for</span> <span class=\"n\">taxon</span> <span class=\"ow\">in</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">LineageEx</span><span class=\"o\">.</span><span class=\"n\">children</span><span class=\"p\">:</span>\n",
       "        <span class=\"k\">if</span> <span class=\"nb\">isinstance</span><span class=\"p\">(</span><span class=\"n\">taxon</span><span class=\"p\">,</span> <span class=\"n\">Tag</span><span class=\"p\">):</span>\n",
       "            <span class=\"n\">lineage_ex</span><span class=\"o\">.</span><span class=\"n\">append</span><span class=\"p\">(</span>\n",
       "                <span class=\"p\">{</span>\n",
       "                    <span class=\"s2\">&quot;TaxId&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">TaxId</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "                    <span class=\"s2\">&quot;ScientificName&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">ScientificName</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "                    <span class=\"s2\">&quot;Rank&quot;</span><span class=\"p\">:</span> <span class=\"n\">taxon</span><span class=\"o\">.</span><span class=\"n\">Rank</span><span class=\"o\">.</span><span class=\"n\">getText</span><span class=\"p\">(),</span>\n",
       "                <span class=\"p\">}</span>\n",
       "            <span class=\"p\">)</span>\n",
       "\n",
       "    <span class=\"n\">taxon_metadata</span><span class=\"p\">[</span><span class=\"s2\">&quot;LineageEx&quot;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"n\">lineage_ex</span>\n",
       "\n",
       "    <span class=\"k\">return</span> <span class=\"n\">taxon_metadata</span>\n",
       "</pre></div>\n"
      ],
      "text/latex": [
       "\\begin{Verbatim}[commandchars=\\\\\\{\\}]\n",
       "\\PY{k}{def} \\PY{n+nf}{get\\PYZus{}metadata}\\PY{p}{(}\\PY{n}{ncbi\\PYZus{}id}\\PY{p}{)}\\PY{p}{:}\n",
       "    \\PY{l+s+sd}{\\PYZdq{}\\PYZdq{}\\PYZdq{}Request metadata by NCBI taxonomy ID, and return cleaned object\\PYZdq{}\\PYZdq{}\\PYZdq{}}\n",
       "\n",
       "    \\PY{n}{params} \\PY{o}{=} \\PY{p}{\\PYZob{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{db}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Taxonomy}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{,} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{id}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{ncbi\\PYZus{}id}\\PY{p}{\\PYZcb{}}\n",
       "    \\PY{n}{soup} \\PY{o}{=} \\PY{n}{ncbi}\\PY{o}{.}\\PY{n}{api\\PYZus{}soup}\\PY{p}{(}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{efetch}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{,} \\PY{n}{params}\\PY{p}{)}\n",
       "\n",
       "    \\PY{n}{taxon} \\PY{o}{=} \\PY{n}{soup}\\PY{o}{.}\\PY{n}{TaxaSet}\\PY{o}{.}\\PY{n}{Taxon}\n",
       "\n",
       "    \\PY{n}{taxon\\PYZus{}metadata} \\PY{o}{=} \\PY{p}{\\PYZob{}}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{ScientificName}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{ScientificName}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{ParentTaxId}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{ParentTaxId}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Rank}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{Rank}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Division}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{Division}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{GeneticCode}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{p}{\\PYZob{}}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{GCId}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{GCId}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,} \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{GCName}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{GCName}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{\\PYZcb{}}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{MitoGeneticCode}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{p}{\\PYZob{}}\n",
       "            \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{MGCId}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{MGCId}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "            \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{MGCName}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{MGCName}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{p}{\\PYZcb{}}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Lineage}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{Lineage}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{CreateDate}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{CreateDate}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{UpdateDate}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{UpdateDate}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "        \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{PubDate}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{PubDate}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "    \\PY{p}{\\PYZcb{}}\n",
       "\n",
       "    \\PY{k}{if} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{otherNames}\\PY{p}{:}\n",
       "        \\PY{n}{taxon}\\PY{p}{[}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{OtherNames}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{]} \\PY{o}{=} \\PY{p}{(}\\PY{n}{taxon}\\PY{o}{.}\\PY{n}{OtherNames}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\\PY{p}{)}\n",
       "\n",
       "    \\PY{c+c1}{\\PYZsh{} parse lineage}\n",
       "    \\PY{n}{lineage\\PYZus{}ex} \\PY{o}{=} \\PY{p}{[}\\PY{p}{]}\n",
       "    \\PY{k}{for} \\PY{n}{taxon} \\PY{o+ow}{in} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{LineageEx}\\PY{o}{.}\\PY{n}{children}\\PY{p}{:}\n",
       "        \\PY{k}{if} \\PY{n+nb}{isinstance}\\PY{p}{(}\\PY{n}{taxon}\\PY{p}{,} \\PY{n}{Tag}\\PY{p}{)}\\PY{p}{:}\n",
       "            \\PY{n}{lineage\\PYZus{}ex}\\PY{o}{.}\\PY{n}{append}\\PY{p}{(}\n",
       "                \\PY{p}{\\PYZob{}}\n",
       "                    \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{TaxId}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{TaxId}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "                    \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{ScientificName}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{ScientificName}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "                    \\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{Rank}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{:} \\PY{n}{taxon}\\PY{o}{.}\\PY{n}{Rank}\\PY{o}{.}\\PY{n}{getText}\\PY{p}{(}\\PY{p}{)}\\PY{p}{,}\n",
       "                \\PY{p}{\\PYZcb{}}\n",
       "            \\PY{p}{)}\n",
       "\n",
       "    \\PY{n}{taxon\\PYZus{}metadata}\\PY{p}{[}\\PY{l+s+s2}{\\PYZdq{}}\\PY{l+s+s2}{LineageEx}\\PY{l+s+s2}{\\PYZdq{}}\\PY{p}{]} \\PY{o}{=} \\PY{n}{lineage\\PYZus{}ex}\n",
       "\n",
       "    \\PY{k}{return} \\PY{n}{taxon\\PYZus{}metadata}\n",
       "\\end{Verbatim}\n"
      ],
      "text/plain": [
       "def get_metadata(ncbi_id):\n",
       "    \"\"\"Request metadata by NCBI taxonomy ID, and return cleaned object\"\"\"\n",
       "\n",
       "    params = {\"db\": \"Taxonomy\", \"id\": ncbi_id}\n",
       "    soup = ncbi.api_soup(\"efetch\", params)\n",
       "\n",
       "    taxon = soup.TaxaSet.Taxon\n",
       "\n",
       "    taxon_metadata = {\n",
       "        \"ScientificName\": taxon.ScientificName.getText(),\n",
       "        \"ParentTaxId\": taxon.ParentTaxId.getText(),\n",
       "        \"Rank\": taxon.Rank.getText(),\n",
       "        \"Division\": taxon.Division.getText(),\n",
       "        \"GeneticCode\": {\"GCId\": taxon.GCId.getText(), \"GCName\": taxon.GCName.getText()},\n",
       "        \"MitoGeneticCode\": {\n",
       "            \"MGCId\": taxon.MGCId.getText(),\n",
       "            \"MGCName\": taxon.MGCName.getText(),\n",
       "        },\n",
       "        \"Lineage\": taxon.Lineage.getText(),\n",
       "        \"CreateDate\": taxon.CreateDate.getText(),\n",
       "        \"UpdateDate\": taxon.UpdateDate.getText(),\n",
       "        \"PubDate\": taxon.PubDate.getText(),\n",
       "    }\n",
       "\n",
       "    if taxon.otherNames:\n",
       "        taxon[\"OtherNames\"] = (taxon.OtherNames.getText(),)\n",
       "\n",
       "    # parse lineage\n",
       "    lineage_ex = []\n",
       "    for taxon in taxon.LineageEx.children:\n",
       "        if isinstance(taxon, Tag):\n",
       "            lineage_ex.append(\n",
       "                {\n",
       "                    \"TaxId\": taxon.TaxId.getText(),\n",
       "                    \"ScientificName\": taxon.ScientificName.getText(),\n",
       "                    \"Rank\": taxon.Rank.getText(),\n",
       "                }\n",
       "            )\n",
       "\n",
       "    taxon_metadata[\"LineageEx\"] = lineage_ex\n",
       "\n",
       "    return taxon_metadata"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Code(inspect.getsource(ncbi.get_metadata), language='python')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <span style=\"color:gray\">_Accessing what is in the NCBI Taxonomy database_</span>\n",
    "\n",
    "The NCBI Taxonomy documentation indicates that the data model is built around a central framework called NameBank, and each entry includes a <span style=\"color:#4682B4\">**_primary name_**</span>, <span style=\"color:#4682B4\">**_secondary names_**</span>, a <span style=\"color:#4682B4\">**_taxonomy identifier_**</span>, <span style=\"color:#4682B4\">**_name entity identifiers_**</span> along with other various metadata about the lineage, genetic code, and linked Entrez records.\n",
    "\n",
    "We first tested our database access using a sample query with the 'esearch' utility for the Entrez NCBI Taxonomy database using formal organism names. \n",
    "\n",
    "<span style=\"color:#4682B4\">**_Input: Entrez text query (&term); Entrez database (&db)_**</span>\n",
    "\n",
    "<span style=\"color:#4682B4\">**_Expected Output: List of UIDs matching the Entrez query_**</span>\n",
    "\n",
    "\n",
    "Sample data: Get the Taxonomic IDs (TaxID) for 'influenza A subtype h1n1' 'alphainfluenzavirus', and 'orthomyxoviridae:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-05-05 12:21:25.391 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term influenza A subtype h1n1\n",
      "2022-05-05 12:21:25.534 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term alphainfluenzavirus\n",
      "2022-05-05 12:21:25.673 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term orthomyxoviridae\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['114727', '197911', '11308']\n"
     ]
    }
   ],
   "source": [
    "term_list = ['influenza A subtype h1n1','alphainfluenzavirus','orthomyxoviridae']\n",
    "id_list = []\n",
    "\n",
    "for each in term_list:\n",
    "    id_list.append(ncbi.id_search(each))\n",
    "\n",
    "print(id_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using this ID, we can then pass it through the 'efetch' utility to get the full records from the NCBI Taxonomy database about organisms in XML format.\n",
    "\n",
    "<span style=\"color:#4682B4\">**_Input: List of UIDs (&id); Entrez database (&db); Retrieval type (&rettype); Retrieval mode (&retmode)_**</span>\n",
    "\n",
    "<span style=\"color:#4682B4\">**_Expected Output: Formatted data records as specified_**</span>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "{'ScientificName': 'H1N1 subtype', 'ParentTaxId': '11320', 'Rank': 'serotype', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales; Orthomyxoviridae; Alphainfluenzavirus; Influenza A virus', 'CreateDate': '2000/02/07 12:42:00', 'UpdateDate': '2020/04/07 15:24:16', 'PubDate': '2000/02/07 12:42:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}, {'TaxId': '11308', 'ScientificName': 'Orthomyxoviridae', 'Rank': 'family'}, {'TaxId': '197911', 'ScientificName': 'Alphainfluenzavirus', 'Rank': 'genus'}, {'TaxId': '11320', 'ScientificName': 'Influenza A virus', 'Rank': 'species'}]}\n",
      "\n",
      "{'ScientificName': 'Alphainfluenzavirus', 'ParentTaxId': '11308', 'Rank': 'genus', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales; Orthomyxoviridae', 'CreateDate': '2002/05/08 12:00:00', 'UpdateDate': '2020/04/07 15:24:16', 'PubDate': '2002/06/12 19:01:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}, {'TaxId': '11308', 'ScientificName': 'Orthomyxoviridae', 'Rank': 'family'}]}\n",
      "\n",
      "{'ScientificName': 'Orthomyxoviridae', 'ParentTaxId': '2499411', 'Rank': 'family', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales', 'CreateDate': '1995/02/27 09:24:00', 'UpdateDate': '2002/05/08 12:00:00', 'PubDate': '1993/04/23 01:00:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}]}\n"
     ]
    }
   ],
   "source": [
    "for each in id_list: \n",
    "    print('\\n' + str(ncbi.get_metadata(each)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The query can handle multiple terms at once and return lists of IDs and full records for each ID. \n",
    "\n",
    "This same query can also be performed using informal names and synonyms for the organisms, as shown below. Together, these features will allow us to pass in disease and species names from other data sources and map them back to the formal Taxonomy."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-05-05 12:26:47.834 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term Influenzavirus A\n",
      "2022-05-05 12:26:47.974 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term h1n1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "{'ScientificName': 'Alphainfluenzavirus', 'ParentTaxId': '11308', 'Rank': 'genus', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales; Orthomyxoviridae', 'CreateDate': '2002/05/08 12:00:00', 'UpdateDate': '2020/04/07 15:24:16', 'PubDate': '2002/06/12 19:01:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}, {'TaxId': '11308', 'ScientificName': 'Orthomyxoviridae', 'Rank': 'family'}]}\n",
      "\n",
      "{'ScientificName': 'H1N1 subtype', 'ParentTaxId': '11320', 'Rank': 'serotype', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales; Orthomyxoviridae; Alphainfluenzavirus; Influenza A virus', 'CreateDate': '2000/02/07 12:42:00', 'UpdateDate': '2020/04/07 15:24:16', 'PubDate': '2000/02/07 12:42:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}, {'TaxId': '11308', 'ScientificName': 'Orthomyxoviridae', 'Rank': 'family'}, {'TaxId': '197911', 'ScientificName': 'Alphainfluenzavirus', 'Rank': 'genus'}, {'TaxId': '11320', 'ScientificName': 'Influenza A virus', 'Rank': 'species'}]}\n"
     ]
    }
   ],
   "source": [
    "term_list = ['Influenzavirus A','h1n1']\n",
    "id_list = []\n",
    "\n",
    "for each in term_list:\n",
    "    id_list.append(ncbi.id_search(each))\n",
    "\n",
    "\n",
    "for each in id_list:\n",
    "    print('\\n' + str(ncbi.get_metadata(each)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Diving deeper into Entrez with the species 'Influenzavirus A subtype H1N1' (H1N1):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-05-05 12:26:51.253 | INFO     | ncbi.id_search:id_search:8 - Searching ncbi for term Influenzavirus A subtype H1N1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: 114727\n",
      "\n",
      "{'ScientificName': 'H1N1 subtype', 'ParentTaxId': '11320', 'Rank': 'serotype', 'Division': 'Viruses', 'GeneticCode': {'GCId': '1', 'GCName': 'Standard'}, 'MitoGeneticCode': {'MGCId': '0', 'MGCName': 'Unspecified'}, 'Lineage': 'Viruses; Riboviria; Orthornavirae; Negarnaviricota; Polyploviricotina; Insthoviricetes; Articulavirales; Orthomyxoviridae; Alphainfluenzavirus; Influenza A virus', 'CreateDate': '2000/02/07 12:42:00', 'UpdateDate': '2020/04/07 15:24:16', 'PubDate': '2000/02/07 12:42:00', 'LineageEx': [{'TaxId': '10239', 'ScientificName': 'Viruses', 'Rank': 'superkingdom'}, {'TaxId': '2559587', 'ScientificName': 'Riboviria', 'Rank': 'clade'}, {'TaxId': '2732396', 'ScientificName': 'Orthornavirae', 'Rank': 'kingdom'}, {'TaxId': '2497569', 'ScientificName': 'Negarnaviricota', 'Rank': 'phylum'}, {'TaxId': '2497571', 'ScientificName': 'Polyploviricotina', 'Rank': 'subphylum'}, {'TaxId': '2497577', 'ScientificName': 'Insthoviricetes', 'Rank': 'class'}, {'TaxId': '2499411', 'ScientificName': 'Articulavirales', 'Rank': 'order'}, {'TaxId': '11308', 'ScientificName': 'Orthomyxoviridae', 'Rank': 'family'}, {'TaxId': '197911', 'ScientificName': 'Alphainfluenzavirus', 'Rank': 'genus'}, {'TaxId': '11320', 'ScientificName': 'Influenza A virus', 'Rank': 'species'}]}\n"
     ]
    }
   ],
   "source": [
    "h1n1_id = ncbi.id_search('Influenzavirus A subtype H1N1')\n",
    "print(f'ID: {h1n1_id}\\n')\n",
    "\n",
    "h1n1_data = ncbi.get_metadata(h1n1_id)\n",
    "print(h1n1_data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Specifically, we can pull out the lineage of H1N1 to visualize the hierarchy (especially as nodes) and obtain the IDs for higher ranks. Such queries allow us to use a shortest path analysis to identify the family which contains H1N1, where we specify the start point (H1N1) and the end node type (Family), following edges backwards to return exactly one parent."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TaxId</th>\n",
       "      <th>ScientificName</th>\n",
       "      <th>Rank</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10239</td>\n",
       "      <td>Viruses</td>\n",
       "      <td>superkingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2559587</td>\n",
       "      <td>Riboviria</td>\n",
       "      <td>clade</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2732396</td>\n",
       "      <td>Orthornavirae</td>\n",
       "      <td>kingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2497569</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>phylum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2497571</td>\n",
       "      <td>Polyploviricotina</td>\n",
       "      <td>subphylum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2497577</td>\n",
       "      <td>Insthoviricetes</td>\n",
       "      <td>class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2499411</td>\n",
       "      <td>Articulavirales</td>\n",
       "      <td>order</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>11308</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>197911</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>genus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11320</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>species</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     TaxId       ScientificName          Rank\n",
       "0    10239              Viruses  superkingdom\n",
       "1  2559587            Riboviria         clade\n",
       "2  2732396        Orthornavirae       kingdom\n",
       "3  2497569      Negarnaviricota        phylum\n",
       "4  2497571    Polyploviricotina     subphylum\n",
       "5  2497577      Insthoviricetes         class\n",
       "6  2499411      Articulavirales         order\n",
       "7    11308     Orthomyxoviridae        family\n",
       "8   197911  Alphainfluenzavirus         genus\n",
       "9    11320    Influenza A virus       species"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extracting the lineage\n",
    "lineage = h1n1_data[\"LineageEx\"]\n",
    "\n",
    "lineage_df = pd.DataFrame(lineage)\n",
    "lineage_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using the Taxonomy ID for H1N1, we can use the ELink utility to then access information from any of the other Entrez databases that link with Taxonomy. \n",
    "\n",
    "In the example below, we pull out gene sequence data from the NCBI Gene database:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n",
       "<!DOCTYPE Entrezgene-Set PUBLIC \"-//NLM//DTD NCBI-Entrezgene, 21st January 2005//EN\" \"https://www.ncbi.nlm.nih.gov/data_specs/dtd/NCBI_Entrezgene.dtd\">\n",
       "<Entrezgene-Set>\n",
       "<Entrezgene>\n",
       "<Entrezgene_track-info>\n",
       "<Gene-track>\n",
       "<Gene-track_geneid>10090</Gene-track_geneid>\n",
       "<Gene-track_status value=\"live\">0</Gene-track_status>\n",
       "<Gene-track_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>1999</Date-std_year>\n",
       "<Date-std_month>5</Date-std_month>\n",
       "<Date-std_day>25</Date-std_day>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-track_create-date>\n",
       "<Gene-track_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2022</Date-std_year>\n",
       "<Date-std_month>4</Date-std_month>\n",
       "<Date-std_day>24</Date-std_day>\n",
       "<Date-std_hour>18</Date-std_hour>\n",
       "<Date-std_minute>0</Date-std_minute>\n",
       "<Date-std_second>40</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-track_update-date>\n",
       "</Gene-track>\n",
       "</Entrezgene_track-info>\n",
       "<Entrezgene_type value=\"protein-coding\">6</Entrezgene_type>\n",
       "<Entrezgene_source>\n",
       "<BioSource>\n",
       "<BioSource_genome value=\"genomic\">1</BioSource_genome>\n",
       "<BioSource_origin value=\"natural\">1</BioSource_origin>\n",
       "<BioSource_org>\n",
       "<Org-ref>\n",
       "<Org-ref_taxname>Homo sapiens</Org-ref_taxname>\n",
       "<Org-ref_common>human</Org-ref_common>\n",
       "<Org-ref_db>\n",
       "<Dbtag>\n",
       "<Dbtag_db>taxon</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>9606</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Org-ref_db>\n",
       "<Org-ref_orgname>\n",
       "<OrgName>\n",
       "<OrgName_name>\n",
       "<OrgName_name_binomial>\n",
       "<BinomialOrgName>\n",
       "<BinomialOrgName_genus>Homo</BinomialOrgName_genus>\n",
       "<BinomialOrgName_species>sapiens</BinomialOrgName_species>\n",
       "</BinomialOrgName>\n",
       "</OrgName_name_binomial>\n",
       "</OrgName_name>\n",
       "<OrgName_attrib>specified</OrgName_attrib>\n",
       "<OrgName_lineage>Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia; Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae; Homo</OrgName_lineage>\n",
       "<OrgName_gcode>1</OrgName_gcode>\n",
       "<OrgName_mgcode>2</OrgName_mgcode>\n",
       "<OrgName_div>PRI</OrgName_div>\n",
       "</OrgName>\n",
       "</Org-ref_orgname>\n",
       "</Org-ref>\n",
       "</BioSource_org>\n",
       "<BioSource_subtype>\n",
       "<SubSource>\n",
       "<SubSource_subtype value=\"chromosome\">1</SubSource_subtype>\n",
       "<SubSource_name>6</SubSource_name>\n",
       "</SubSource>\n",
       "</BioSource_subtype>\n",
       "</BioSource>\n",
       "</Entrezgene_source>\n",
       "<Entrezgene_gene>\n",
       "<Gene-ref>\n",
       "<Gene-ref_locus>UST</Gene-ref_locus>\n",
       "<Gene-ref_desc>uronyl 2-sulfotransferase</Gene-ref_desc>\n",
       "<Gene-ref_maploc>6q25.1</Gene-ref_maploc>\n",
       "<Gene-ref_db>\n",
       "<Dbtag>\n",
       "<Dbtag_db>HGNC</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>HGNC:17223</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Ensembl</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>ENSG00000111962</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "<Dbtag>\n",
       "<Dbtag_db>MIM</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>610752</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "<Dbtag>\n",
       "<Dbtag_db>AllianceGenome</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>HGNC:17223</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Gene-ref_db>\n",
       "<Gene-ref_syn>\n",
       "<Gene-ref_syn_E>2OST</Gene-ref_syn_E>\n",
       "</Gene-ref_syn>\n",
       "</Gene-ref>\n",
       "</Entrezgene_gene>\n",
       "<Entrezgene_prot>\n",
       "<Prot-ref>\n",
       "<Prot-ref_name>\n",
       "<Prot-ref_name_E>dermatan/chondroitin sulfate 2-sulfotransferase</Prot-ref_name_E>\n",
       "</Prot-ref_name>\n",
       "<Prot-ref_desc>uronyl 2-sulfotransferase</Prot-ref_desc>\n",
       "</Prot-ref>\n",
       "</Entrezgene_prot>\n",
       "<Entrezgene_summary>Uronyl 2-sulfotransferase transfers sulfate to the 2-position of uronyl residues, such as iduronyl residues in dermatan sulfate and glucuronyl residues in chondroitin sulfate (Kobayashi et al., 1999 [PubMed 10187838]).[supplied by OMIM, Mar 2008]</Entrezgene_summary>\n",
       "<Entrezgene_location>\n",
       "<Maps>\n",
       "<Maps_display-str>6q25.1</Maps_display-str>\n",
       "<Maps_method>\n",
       "<Maps_method_map-type value=\"cyto\"/>\n",
       "</Maps_method>\n",
       "</Maps>\n",
       "</Entrezgene_location>\n",
       "<Entrezgene_gene-source>\n",
       "<Gene-source>\n",
       "<Gene-source_src>LocusLink</Gene-source_src>\n",
       "<Gene-source_src-int>10090</Gene-source_src-int>\n",
       "<Gene-source_src-str2>10090</Gene-source_src-str2>\n",
       "</Gene-source>\n",
       "</Entrezgene_gene-source>\n",
       "<Entrezgene_locus>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference GRCh38.p14 Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Chromosome 6 Reference GRCh38.p14 Primary Assembly</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"miscRNA\">14</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>transcript variant X1</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XR_001743088</Gene-commentary_accession>\n",
       "<Gene-commentary_version>3</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148990374</Seq-interval_from>\n",
       "<Seq-interval_to>148990428</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148992983</Seq-interval_from>\n",
       "<Seq-interval_to>148993065</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149011956</Seq-interval_from>\n",
       "<Seq-interval_to>149012826</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359218</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_rna>\n",
       "<RNA-ref>\n",
       "<RNA-ref_type value=\"miscRNA\"/>\n",
       "<RNA-ref_ext>\n",
       "<RNA-ref_ext_gen>\n",
       "<RNA-gen>\n",
       "<RNA-gen_product>uronyl 2-sulfotransferase, transcript variant X1</RNA-gen_product>\n",
       "</RNA-gen>\n",
       "</RNA-ref_ext_gen>\n",
       "</RNA-ref_ext>\n",
       "</RNA-ref>\n",
       "</Gene-commentary_rna>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"miscRNA\">14</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>transcript variant X2</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XR_007059195</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148990374</Seq-interval_from>\n",
       "<Seq-interval_to>148990428</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148992983</Seq-interval_from>\n",
       "<Seq-interval_to>148993065</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149005302</Seq-interval_from>\n",
       "<Seq-interval_to>149005588</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359219</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_rna>\n",
       "<RNA-ref>\n",
       "<RNA-ref_type value=\"miscRNA\"/>\n",
       "<RNA-ref_ext>\n",
       "<RNA-ref_ext_gen>\n",
       "<RNA-gen>\n",
       "<RNA-gen_product>uronyl 2-sulfotransferase, transcript variant X2</RNA-gen_product>\n",
       "</RNA-gen>\n",
       "</RNA-ref_ext_gen>\n",
       "</RNA-ref_ext>\n",
       "</RNA-ref>\n",
       "</Gene-commentary_rna>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>NM_005715</Gene-commentary_accession>\n",
       "<Gene-commentary_version>3</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149019138</Seq-interval_from>\n",
       "<Seq-interval_to>149019235</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149021323</Seq-interval_from>\n",
       "<Seq-interval_to>149021480</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149073832</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>1519243657</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>NP_005706</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>CCDS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>CCDS5213.1</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>CCDS5213.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747430</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149019138</Seq-interval_from>\n",
       "<Seq-interval_to>149019235</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149021323</Seq-interval_from>\n",
       "<Seq-interval_to>149021480</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149073832</Seq-interval_from>\n",
       "<Seq-interval_to>149074115</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>5032219</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>transcript variant X3</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XM_047418047</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148990374</Seq-interval_from>\n",
       "<Seq-interval_to>148990428</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149019138</Seq-interval_from>\n",
       "<Seq-interval_to>149019152</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359220</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>isoform X1</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XP_047274003</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747430</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148990374</Seq-interval_from>\n",
       "<Seq-interval_to>148990424</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359221</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>transcript variant X4</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XM_047418048</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148990374</Seq-interval_from>\n",
       "<Seq-interval_to>148990428</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149005302</Seq-interval_from>\n",
       "<Seq-interval_to>149005588</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359222</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>isoform X1</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XP_047274004</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747430</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148990374</Seq-interval_from>\n",
       "<Seq-interval_to>148990424</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359223</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>transcript variant X5</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XM_011535378</Gene-commentary_accession>\n",
       "<Gene-commentary_version>4</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149005302</Seq-interval_from>\n",
       "<Seq-interval_to>149005588</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359224</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>isoform X2</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XP_011533680</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747430</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148964409</Seq-interval_from>\n",
       "<Seq-interval_to>148964562</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149005302</Seq-interval_from>\n",
       "<Seq-interval_to>149005313</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>767941753</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>transcript variant X6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XM_017010152</Gene-commentary_accession>\n",
       "<Gene-commentary_version>2</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148963912</Seq-interval_from>\n",
       "<Seq-interval_to>148964402</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359225</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_label>isoform X3</Gene-commentary_label>\n",
       "<Gene-commentary_accession>XP_016865641</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747430</Seq-interval_from>\n",
       "<Seq-interval_to>148747676</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148886985</Seq-interval_from>\n",
       "<Seq-interval_to>148887028</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148941278</Seq-interval_from>\n",
       "<Seq-interval_to>148941433</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148953871</Seq-interval_from>\n",
       "<Seq-interval_to>148953950</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148963912</Seq-interval_from>\n",
       "<Seq-interval_to>148964059</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>1034648104</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Alternate T2T-CHM13v2.0</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Chromosome 6 Alternate T2T-CHM13v2.0</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_060930</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149946825</Seq-interval_from>\n",
       "<Seq-interval_to>150276838</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>NM_005715</Gene-commentary_accession>\n",
       "<Gene-commentary_version>3</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149946825</Seq-interval_from>\n",
       "<Seq-interval_to>149947472</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150086793</Seq-interval_from>\n",
       "<Seq-interval_to>150086836</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150141091</Seq-interval_from>\n",
       "<Seq-interval_to>150141246</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150153684</Seq-interval_from>\n",
       "<Seq-interval_to>150153763</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150164223</Seq-interval_from>\n",
       "<Seq-interval_to>150164376</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150218947</Seq-interval_from>\n",
       "<Seq-interval_to>150219044</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150221132</Seq-interval_from>\n",
       "<Seq-interval_to>150221289</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150273682</Seq-interval_from>\n",
       "<Seq-interval_to>150276838</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>1519243657</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>NP_005706</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_genomic-coords>\n",
       "<Seq-loc>\n",
       "<Seq-loc_mix>\n",
       "<Seq-loc-mix>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149947226</Seq-interval_from>\n",
       "<Seq-interval_to>149947472</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150086793</Seq-interval_from>\n",
       "<Seq-interval_to>150086836</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150141091</Seq-interval_from>\n",
       "<Seq-interval_to>150141246</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150153684</Seq-interval_from>\n",
       "<Seq-interval_to>150153763</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150164223</Seq-interval_from>\n",
       "<Seq-interval_to>150164376</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150218947</Seq-interval_from>\n",
       "<Seq-interval_to>150219044</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150221132</Seq-interval_from>\n",
       "<Seq-interval_to>150221289</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>150273682</Seq-interval_from>\n",
       "<Seq-interval_to>150273965</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Seq-loc-mix>\n",
       "</Seq-loc_mix>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_genomic-coords>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>5032219</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "</Entrezgene_locus>\n",
       "<Entrezgene_properties>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_label>Nomenclature</Gene-commentary_label>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_anchor>HUGO Gene Nomenclature Committee</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_properties>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"property\">16</Gene-commentary_type>\n",
       "<Gene-commentary_label>Official Symbol</Gene-commentary_label>\n",
       "<Gene-commentary_text>UST</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"property\">16</Gene-commentary_type>\n",
       "<Gene-commentary_label>Official Full Name</Gene-commentary_label>\n",
       "<Gene-commentary_text>uronyl 2-sulfotransferase</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_properties>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"property\">16</Gene-commentary_type>\n",
       "<Gene-commentary_label>Exon count</Gene-commentary_label>\n",
       "<Gene-commentary_text>13</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GeneOntology</Gene-commentary_heading>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_pre-text>Provided by</Other-source_pre-text>\n",
       "<Other-source_anchor>GOA</Other-source_anchor>\n",
       "<Other-source_url>http://www.ebi.ac.uk/GOA/</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_label>Function</Gene-commentary_label>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>21873635</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GO</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>8146</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>enables</Other-source_pre-text>\n",
       "<Other-source_anchor>sulfotransferase activity</Other-source_anchor>\n",
       "<Other-source_post-text>evidence: IBA</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GO</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>8146</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>enables</Other-source_pre-text>\n",
       "<Other-source_anchor>sulfotransferase activity</Other-source_anchor>\n",
       "<Other-source_post-text>evidence: TAS</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_label>Process</Gene-commentary_label>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GO</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>30208</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>involved_in</Other-source_pre-text>\n",
       "<Other-source_anchor>dermatan sulfate biosynthetic process</Other-source_anchor>\n",
       "<Other-source_post-text>evidence: TAS</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GO</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>30010</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>involved_in</Other-source_pre-text>\n",
       "<Other-source_anchor>establishment of cell polarity</Other-source_anchor>\n",
       "<Other-source_post-text>evidence: IEA</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>10187838</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GO</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>6477</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>involved_in</Other-source_pre-text>\n",
       "<Other-source_anchor>protein sulfation</Other-source_anchor>\n",
       "<Other-source_post-text>evidence: TAS</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GO</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>50770</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>involved_in</Other-source_pre-text>\n",
       "<Other-source_anchor>regulation of axonogenesis</Other-source_anchor>\n",
       "<Other-source_post-text>evidence: IEA</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_label>Component</Gene-commentary_label>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GO</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>139</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>located_in</Other-source_pre-text>\n",
       "<Other-source_anchor>Golgi membrane</Other-source_anchor>\n",
       "<Other-source_post-text>evidence: TAS</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GO</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>16021</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>located_in</Other-source_pre-text>\n",
       "<Other-source_anchor>integral component of membrane</Other-source_anchor>\n",
       "<Other-source_post-text>evidence: IEA</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Entrezgene_properties>\n",
       "<Entrezgene_comments>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>RefSeq Status</Gene-commentary_heading>\n",
       "<Gene-commentary_label>VALIDATED</Gene-commentary_label>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"gene-group\">23</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Orthologs from Annotation Pipeline</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"gene-group\">23</Gene-commentary_type>\n",
       "<Gene-commentary_label>Primary</Gene-commentary_label>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>10090</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>UST</Other-source_pre-text>\n",
       "<Other-source_anchor>human</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>taxon</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>9606</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"gene-group\">23</Gene-commentary_type>\n",
       "<Gene-commentary_label>Additional</Gene-commentary_label>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>338362</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_pre-text>UST</Other-source_pre-text>\n",
       "<Other-source_anchor>mouse</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>taxon</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>10090</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>27511124</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>25456346</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>23620144</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>21873635</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>20379614</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>20360315</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>19240061</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>17227754</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>16192264</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>15489334</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>14574404</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>12477932</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>10187838</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Phenotypes</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"phenotype\">19</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genome-wide pharmacogenetics of antidepressant response in the GENDEP project.</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"phenotype\">19</Gene-commentary_type>\n",
       "<Gene-commentary_heading>EBI GWAS Catalog</Gene-commentary_heading>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>20360315</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_anchor>EBI GWAS Catalog</Other-source_anchor>\n",
       "<Other-source_url>https://www.ebi.ac.uk/gwas/publications/20360315</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"phenotype\">19</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genome-wide scan of job-related exhaustion with three replication studies implicate a susceptibility variant at the UST gene locus.</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"phenotype\">19</Gene-commentary_type>\n",
       "<Gene-commentary_heading>EBI GWAS Catalog</Gene-commentary_heading>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>23620144</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_anchor>EBI GWAS Catalog</Other-source_anchor>\n",
       "<Other-source_url>https://www.ebi.ac.uk/gwas/publications/23620144</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"phenotype\">19</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genomic predictors of combat stress vulnerability and resilience in U.S. Marines: A genome-wide association study across multiple ancestries implicates PRTFDC1 as a potential PTSD gene.</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"phenotype\">19</Gene-commentary_type>\n",
       "<Gene-commentary_heading>EBI GWAS Catalog</Gene-commentary_heading>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>25456346</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_anchor>EBI GWAS Catalog</Other-source_anchor>\n",
       "<Other-source_url>https://www.ebi.ac.uk/gwas/publications/25456346</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>NCBI Reference Sequences (RefSeq)</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>RefSeqs maintained independently of Annotated Genomes</Gene-commentary_heading>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA Sequence</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>NM_005715</Gene-commentary_accession>\n",
       "<Gene-commentary_version>3</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>1519243657</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>NM_005715</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>1519243657</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Product</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>NP_005706</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>5032219</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>NP_005706</Other-source_anchor>\n",
       "<Other-source_post-text>uronyl 2-sulfotransferase</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>5032219</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Conserved Domains</Gene-commentary_heading>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_pre-text>(1)</Other-source_pre-text>\n",
       "<Other-source_anchor>summary</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>CDD</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>328972</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>cl22970: Sulfotransfer_2; Sulfotransferase family</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_text>Location: 104 - 356</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Related</Gene-commentary_heading>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Ensembl</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>ENSP00000356433.4</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_heading>UniProtKB</Gene-commentary_heading>\n",
       "<Gene-commentary_source/>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniProtKB/Swiss-Prot</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>Q9Y2C2</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>Q9Y2C2</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Consensus CDS (CCDS)</Gene-commentary_heading>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>CCDS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>CCDS5213.1</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>CCDS5213.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_label>RefSeq Status</Gene-commentary_label>\n",
       "<Gene-commentary_text>VALIDATED</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Source Sequence</Gene-commentary_heading>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>AB020316,AI570697,AL807246,BC093668,CA842761,DB496757</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AB020316,AI570697,AL807246,BC093668,CA842761,DB496757</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Related</Gene-commentary_heading>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Ensembl</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>ENST00000367463.5</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>RefSeqs of Annotated Genomes: Homo sapiens Annotation Release 110</Gene-commentary_heading>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Reference GRCh38.p14 Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>568815592</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"miscRNA\">14</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XR_001743088</Gene-commentary_accession>\n",
       "<Gene-commentary_version>3</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2217359218</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359218</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"miscRNA\">14</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XR_007059195</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2217359219</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359219</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XM_047418047</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2217359220</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359220</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XP_047274003</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2217359221</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>XP_047274003</Other-source_anchor>\n",
       "<Other-source_post-text>uronyl 2-sulfotransferase isoform X1</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359221</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XM_047418048</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2217359222</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359222</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XP_047274004</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2217359223</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>XP_047274004</Other-source_anchor>\n",
       "<Other-source_post-text>uronyl 2-sulfotransferase isoform X1</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359223</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XM_011535378</Gene-commentary_accession>\n",
       "<Gene-commentary_version>4</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2217359224</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359224</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XP_011533680</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>767941753</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>XP_011533680</Other-source_anchor>\n",
       "<Other-source_post-text>uronyl 2-sulfotransferase isoform X2</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>767941753</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Conserved Domains</Gene-commentary_heading>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_pre-text>(1)</Other-source_pre-text>\n",
       "<Other-source_anchor>summary</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>CDD</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>304664</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>cl22970: Sulfotransfer_2; Sulfotransferase family</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_text>Location: 104 - 211</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XM_017010152</Gene-commentary_accession>\n",
       "<Gene-commentary_version>2</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2217359225</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2217359225</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>XP_016865641</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>1034648104</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>XP_016865641</Other-source_anchor>\n",
       "<Other-source_post-text>uronyl 2-sulfotransferase isoform X3</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>1034648104</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Alternate T2T-CHM13v2.0</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>NC_060930</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>2194974009</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149946825</Seq-interval_from>\n",
       "<Seq-interval_to>150276838</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products/>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Related Sequences</Gene-commentary_heading>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genomic</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AL357992</Gene-commentary_accession>\n",
       "<Gene-commentary_version>14</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>11611020</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AL357992.14</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>29219</Seq-interval_from>\n",
       "<Seq-interval_to>154929</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"minus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>11611020</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genomic</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AL359252</Gene-commentary_accession>\n",
       "<Gene-commentary_version>17</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>12584421</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AL359252.17</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>2000</Seq-interval_from>\n",
       "<Seq-interval_to>178928</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>12584421</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genomic</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AL589705</Gene-commentary_accession>\n",
       "<Gene-commentary_version>10</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>14018307</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AL589705.10</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genomic</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AL590485</Gene-commentary_accession>\n",
       "<Gene-commentary_version>7</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>14329598</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AL590485.7</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>100</Seq-interval_from>\n",
       "<Seq-interval_to>10479</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>14329598</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genomic</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AL807246</Gene-commentary_accession>\n",
       "<Gene-commentary_version>11</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>22138729</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AL807246.11</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>121088</Seq-interval_from>\n",
       "<Seq-interval_to>124213</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>22138729</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genomic</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>CH471051</Gene-commentary_accession>\n",
       "<Gene-commentary_version>2</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>74230054</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>CH471051.2</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>74230054</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>EAW47812</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>119568197</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>EAW47812.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>119568197</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Genomic</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>CP068272</Gene-commentary_accession>\n",
       "<Gene-commentary_version>2</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>2033694209</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>CP068272.2</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149946825</Seq-interval_from>\n",
       "<Seq-interval_to>150276838</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2033694209</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AB020316</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>4803734</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AB020316.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>4803734</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>BAA77510</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>4803735</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BAA77510.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>4803735</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AI570697</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>4534071</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AI570697.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>4534071</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AJ420554</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>17066418</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AJ420554.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>17066418</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AK292922</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>158259304</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AK292922.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>158259304</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>BAF85611</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>158259305</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BAF85611.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>158259305</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>AK315320</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>164694916</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AK315320.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>164694916</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>BAG37723</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>189067528</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BAG37723.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>189067528</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>BC093668</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>62739432</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BC093668.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>62739432</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>AAH93668</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>62739433</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AAH93668.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>62739433</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>BC093694</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>62739709</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BC093694.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>62739709</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>AAH93694</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>62739710</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AAH93694.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>62739710</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>CA842761</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>26672692</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>CA842761.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>26672692</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"mRNA\">3</Gene-commentary_type>\n",
       "<Gene-commentary_heading>mRNA</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>DB496757</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Nucleotide</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>92191650</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>DB496757.1</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>92191650</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_text>None</Gene-commentary_text>\n",
       "<Gene-commentary_products>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"peptide\">8</Gene-commentary_type>\n",
       "<Gene-commentary_accession>Q9Y2C2</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>Protein</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>68052988</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniProtKB/Swiss-Prot</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>Q9Y2C2.1</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>UniProtKB/Swiss-Prot:Q9Y2C2</Other-source_anchor>\n",
       "<Other-source_url>https://www.uniprot.org/uniprot/Q9Y2C2</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_whole>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>68052988</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-loc_whole>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_products>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Additional Links</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_text>MIM</Gene-commentary_text>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>MIM</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>610752</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>610752</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>AceView</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>10090</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>AceView</Other-source_anchor>\n",
       "<Other-source_url>https://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/av.cgi?c=geneid&amp;org=9606&amp;l=10090</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_text>UCSC</Gene-commentary_text>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UCSC</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>UCSC</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>UCSC</Other-source_anchor>\n",
       "<Other-source_url>http://genome.ucsc.edu/cgi-bin/hgTracks?org=human&amp;position=NM_005715</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>HuGE Navigator</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>10090</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>HuGE Navigator</Other-source_anchor>\n",
       "<Other-source_url>https://phgkb.cdc.gov/PHGKB/huGEPedia.action?firstQuery=UST&amp;geneID=10090&amp;typeOption=gene&amp;which=2&amp;pubOrderType=pubD&amp;typeSubmit=GO&amp;check=y</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_text>UniGene</Gene-commentary_text>\n",
       "<Gene-commentary_xtra-properties>\n",
       "<Xtra-Terms>\n",
       "<Xtra-Terms_tag>UNIGENE</Xtra-Terms_tag>\n",
       "<Xtra-Terms_value>Hs.657370</Xtra-Terms_value>\n",
       "</Xtra-Terms>\n",
       "</Gene-commentary_xtra-properties>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniGene</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>Hs.657370</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>Hs.657370</Other-source_anchor>\n",
       "<Other-source_url>https://www.ncbi.nlm.nih.gov/UniGene/clust.cgi?ORG=Hs&amp;CID=657370</Other-source_url>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Results show that overexpression of 2OST alone in the HEK293 cells did not appreciably affect heparan sulfate (HS) chain length. Also, its simultaneous overexpression with Hsepi abolished the effect of Hsepi overexpression on the HS chain lengh.</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>27511124</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2018</Date-std_year>\n",
       "<Date-std_month>5</Date-std_month>\n",
       "<Date-std_day>5</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>2</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2018</Date-std_year>\n",
       "<Date-std_month>5</Date-std_month>\n",
       "<Date-std_day>5</Date-std_day>\n",
       "<Date-std_hour>12</Date-std_hour>\n",
       "<Date-std_minute>21</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Statistical analysis showed the strongest evidence for a variant located in an intron of UST associated with job-related exhaustion.</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>23620144</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2014</Date-std_year>\n",
       "<Date-std_month>2</Date-std_month>\n",
       "<Date-std_day>8</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>1</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2014</Date-std_year>\n",
       "<Date-std_month>2</Date-std_month>\n",
       "<Date-std_day>8</Date-std_day>\n",
       "<Date-std_hour>12</Date-std_hour>\n",
       "<Date-std_minute>4</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Clinical trial of gene-disease association and gene-environment interaction. (HuGE Navigator)</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>20379614</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>HuGENet</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>10090.20379614</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>HuGENet</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2010</Date-std_year>\n",
       "<Date-std_month>6</Date-std_month>\n",
       "<Date-std_day>30</Date-std_day>\n",
       "<Date-std_hour>22</Date-std_hour>\n",
       "<Date-std_minute>4</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2010</Date-std_year>\n",
       "<Date-std_month>6</Date-std_month>\n",
       "<Date-std_day>30</Date-std_day>\n",
       "<Date-std_hour>22</Date-std_hour>\n",
       "<Date-std_minute>7</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Clinical trial and genome-wide association study of gene-disease association, gene-environment interaction, and pharmacogenomic / toxicogenomic. (HuGE Navigator)</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>20360315</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>HuGENet</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>10090.20360315</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>HuGENet</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2010</Date-std_year>\n",
       "<Date-std_month>6</Date-std_month>\n",
       "<Date-std_day>30</Date-std_day>\n",
       "<Date-std_hour>22</Date-std_hour>\n",
       "<Date-std_minute>4</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2010</Date-std_year>\n",
       "<Date-std_month>6</Date-std_month>\n",
       "<Date-std_day>30</Date-std_day>\n",
       "<Date-std_hour>22</Date-std_hour>\n",
       "<Date-std_minute>7</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Observational study of gene-disease association. (HuGE Navigator)</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>19240061</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>HuGENet</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>10090.19240061</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>HuGENet</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2009</Date-std_year>\n",
       "<Date-std_month>4</Date-std_month>\n",
       "<Date-std_day>1</Date-std_day>\n",
       "<Date-std_hour>21</Date-std_hour>\n",
       "<Date-std_minute>31</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2009</Date-std_year>\n",
       "<Date-std_month>4</Date-std_month>\n",
       "<Date-std_day>1</Date-std_day>\n",
       "<Date-std_hour>21</Date-std_hour>\n",
       "<Date-std_minute>32</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>2OST transfers sulfate preferentially to the GlcA residue located in a unique sequence, -GalNAc(4SO(4))-GlcA-GalNAc(6SO(4))-.</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>16192264</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2007</Date-std_year>\n",
       "<Date-std_month>8</Date-std_month>\n",
       "<Date-std_day>25</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>2</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2010</Date-std_year>\n",
       "<Date-std_month>1</Date-std_month>\n",
       "<Date-std_day>21</Date-std_day>\n",
       "<Date-std_hour>0</Date-std_hour>\n",
       "<Date-std_minute>0</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>analysis of differences and similarities various residues play in the biological roles of the HS-2OST and CS-2OST enzymes</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>17227754</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2007</Date-std_year>\n",
       "<Date-std_month>8</Date-std_month>\n",
       "<Date-std_day>25</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>2</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2010</Date-std_year>\n",
       "<Date-std_month>1</Date-std_month>\n",
       "<Date-std_day>21</Date-std_day>\n",
       "<Date-std_hour>0</Date-std_hour>\n",
       "<Date-std_minute>0</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Interactions</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>286133</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SCARA5</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>130309</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>148867</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SLC30A7</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>127176</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>112609</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>MRAP2</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>125195</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>85319</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BAGE2</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>124473</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>83729</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>INHBE</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>123742</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>79877</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>DCAKD</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>122965</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>64132</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>XYLT2</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>122081</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>55830</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>GLT8D1</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>120935</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>54940</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>OCIAD1</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>120280</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>51280</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>GOLM1</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>119432</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>28965</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SLC27A6</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>118790</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>27333</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>GOLIM4</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>118146</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>23556</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>PIGN</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>117100</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>10874</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>NMU</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>116082</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>9528</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>TMEM59</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>114904</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>9374</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>PPT2</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>114775</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>6768</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>ST14</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>112645</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>5445</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>PON2</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>111441</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>4697</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>NDUFA4</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>110777</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>4121</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>MAN1A1</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>110294</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>3949</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>LDLR</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>110141</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>2734</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>GLG1</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>108996</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>1945</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>EFNA4</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>108265</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>1047</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>CLGN</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>107477</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>33961781</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>821</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>CANX</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>107271</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>9</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2021</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Two-hybrid</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28298427</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>154</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>ADRB2</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>106663</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2020</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>8</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>20</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2020</Date-std_year>\n",
       "<Date-std_month>11</Date-std_month>\n",
       "<Date-std_day>8</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>58</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Proximity Label-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>30021884</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>51449</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>PCYOX1</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>119547</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2020</Date-std_year>\n",
       "<Date-std_month>10</Date-std_month>\n",
       "<Date-std_day>4</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>19</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2020</Date-std_year>\n",
       "<Date-std_month>10</Date-std_month>\n",
       "<Date-std_day>4</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>56</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115209</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>OMA1</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>125420</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2017</Date-std_year>\n",
       "<Date-std_month>6</Date-std_month>\n",
       "<Date-std_day>4</Date-std_day>\n",
       "<Date-std_hour>12</Date-std_hour>\n",
       "<Date-std_minute>26</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2018</Date-std_year>\n",
       "<Date-std_month>8</Date-std_month>\n",
       "<Date-std_day>5</Date-std_day>\n",
       "<Date-std_hour>14</Date-std_hour>\n",
       "<Date-std_minute>59</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>56605</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>ERO1B</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>121155</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2017</Date-std_year>\n",
       "<Date-std_month>6</Date-std_month>\n",
       "<Date-std_day>4</Date-std_day>\n",
       "<Date-std_hour>12</Date-std_hour>\n",
       "<Date-std_minute>26</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2018</Date-std_year>\n",
       "<Date-std_month>8</Date-std_month>\n",
       "<Date-std_day>5</Date-std_day>\n",
       "<Date-std_hour>11</Date-std_hour>\n",
       "<Date-std_minute>44</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>54928</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BPNT2</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>120268</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2017</Date-std_year>\n",
       "<Date-std_month>6</Date-std_month>\n",
       "<Date-std_day>4</Date-std_day>\n",
       "<Date-std_hour>12</Date-std_hour>\n",
       "<Date-std_minute>26</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2018</Date-std_year>\n",
       "<Date-std_month>8</Date-std_month>\n",
       "<Date-std_day>5</Date-std_day>\n",
       "<Date-std_hour>11</Date-std_hour>\n",
       "<Date-std_minute>44</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>28514442</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>3714</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>JAG2</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>109918</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2017</Date-std_year>\n",
       "<Date-std_month>6</Date-std_month>\n",
       "<Date-std_day>4</Date-std_day>\n",
       "<Date-std_hour>12</Date-std_hour>\n",
       "<Date-std_minute>26</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2018</Date-std_year>\n",
       "<Date-std_month>8</Date-std_month>\n",
       "<Date-std_day>5</Date-std_day>\n",
       "<Date-std_hour>11</Date-std_hour>\n",
       "<Date-std_minute>44</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-MS</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>21145461</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>8452</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>CUL3</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>114030</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2012</Date-std_year>\n",
       "<Date-std_month>10</Date-std_month>\n",
       "<Date-std_day>7</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>20</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2018</Date-std_year>\n",
       "<Date-std_month>8</Date-std_month>\n",
       "<Date-std_day>5</Date-std_day>\n",
       "<Date-std_hour>11</Date-std_hour>\n",
       "<Date-std_minute>44</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"generif\">18</Gene-commentary_type>\n",
       "<Gene-commentary_text>Affinity Capture-RNA</Gene-commentary_text>\n",
       "<Gene-commentary_refs>\n",
       "<Pub>\n",
       "<Pub_pmid>\n",
       "<PubMedId>19322201</PubMedId>\n",
       "</Pub_pmid>\n",
       "</Pub>\n",
       "</Gene-commentary_refs>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>BioGRID</Other-source_anchor>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>115398</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>GeneID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>1994</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>ELAVL1</Other-source_anchor>\n",
       "</Other-source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>BioGRID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>108309</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "<Gene-commentary_create-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2012</Date-std_year>\n",
       "<Date-std_month>7</Date-std_month>\n",
       "<Date-std_day>8</Date-std_day>\n",
       "<Date-std_hour>10</Date-std_hour>\n",
       "<Date-std_minute>24</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_create-date>\n",
       "<Gene-commentary_update-date>\n",
       "<Date>\n",
       "<Date_std>\n",
       "<Date-std>\n",
       "<Date-std_year>2018</Date-std_year>\n",
       "<Date-std_month>8</Date-std_month>\n",
       "<Date-std_day>5</Date-std_day>\n",
       "<Date-std_hour>11</Date-std_hour>\n",
       "<Date-std_minute>44</Date-std_minute>\n",
       "<Date-std_second>0</Date-std_second>\n",
       "</Date-std>\n",
       "</Date_std>\n",
       "</Date>\n",
       "</Gene-commentary_update-date>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Markers (Sequence Tagged Sites/STS)</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>485655</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>NoName</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>70501</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>D6S1564</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR), detects polymorphism</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>AFMa218wh1</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>GDB:602031</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>GDB:607092</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>HSA218WH1</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH30982</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH37608</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH9738</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>SHGC-20385</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>stSG14620</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>79979</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>RH46094</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>stSG21951</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>83075</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>D6S1148E</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>Cda01a07</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>GDB:443316</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>GDB:448716</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH9870</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>172676</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SHGC-146752</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>61326</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>D6S1329</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>GDB:458620</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>MR10993</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH37205</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>stSG6237</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>WI-5676</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>74135</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SGC44468</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>EST465115</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH60628</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>184476</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SHGC-83979</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>61354</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SHGC-44468</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH97033</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>177605</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SHGC-152794</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>173034</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SHGC-148649</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>94443</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SHGC-58640</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH101358</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>RH99555</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>184307</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>SHGC-111955</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_source>\n",
       "<Other-source>\n",
       "<Other-source_src>\n",
       "<Dbtag>\n",
       "<Dbtag_db>UniSTS</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>85238</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Other-source_src>\n",
       "<Other-source_anchor>RH92645</Other-source_anchor>\n",
       "<Other-source_post-text>(e-PCR)</Other-source_post-text>\n",
       "</Other-source>\n",
       "</Gene-commentary_source>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"other\">255</Gene-commentary_type>\n",
       "<Gene-commentary_label>Alternate name</Gene-commentary_label>\n",
       "<Gene-commentary_text>stSG47471</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Gene Location History</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Annotation Release 110</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p14</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_text>current</Gene-commentary_text>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>40</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>16</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148747029</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>T2T-CHM13v2.0</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Alternate</Gene-commentary_label>\n",
       "<Gene-commentary_text>current</Gene-commentary_text>\n",
       "<Gene-commentary_accession>GCF_009914755</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_009914825</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_060930</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149946825</Seq-interval_from>\n",
       "<Seq-interval_to>150276838</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>2194974009</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20211119</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20210514</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20210226</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20201120</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20200815</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20200522</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20200228</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20191205</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20190905</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 109.20190607</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>39</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Annotation Release 109</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh38.p12</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>38</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>15</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>12</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>148745890</Seq-interval_from>\n",
       "<Seq-interval_to>149076989</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>568815592</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Updated Annotation Release 105.20201022</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh37.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_text>previous assembly</Gene-commentary_text>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>25</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>13</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>11</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149068165</Seq-interval_from>\n",
       "<Seq-interval_to>149398125</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>224589818</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Homo sapiens Annotation Release 105</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>GRCh37.p13</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Reference</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000001405</Gene-commentary_accession>\n",
       "<Gene-commentary_version>25</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000001305</Gene-commentary_accession>\n",
       "<Gene-commentary_version>13</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_000006</Gene-commentary_accession>\n",
       "<Gene-commentary_version>11</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149068062</Seq-interval_from>\n",
       "<Seq-interval_to>149398125</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>224589818</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>HuRef</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Alternate</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000002125</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000000025</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>AC_000138</Gene-commentary_accession>\n",
       "<Gene-commentary_version>1</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>146637094</Seq-interval_from>\n",
       "<Seq-interval_to>146966116</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>157734152</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "<Seq-interval_fuzz-from>\n",
       "<Int-fuzz>\n",
       "<Int-fuzz_lim value=\"lt\"/>\n",
       "</Int-fuzz>\n",
       "</Seq-interval_fuzz-from>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly\">24</Gene-commentary_type>\n",
       "<Gene-commentary_heading>CHM1_1.1</Gene-commentary_heading>\n",
       "<Gene-commentary_label>Alternate</Gene-commentary_label>\n",
       "<Gene-commentary_accession>GCF_000306695</Gene-commentary_accession>\n",
       "<Gene-commentary_version>2</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"assembly-unit\">25</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Primary Assembly</Gene-commentary_heading>\n",
       "<Gene-commentary_accession>GCF_000306705</Gene-commentary_accession>\n",
       "<Gene-commentary_version>2</Gene-commentary_version>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"genomic\">1</Gene-commentary_type>\n",
       "<Gene-commentary_label>Chromosome 6</Gene-commentary_label>\n",
       "<Gene-commentary_accession>NC_018917</Gene-commentary_accession>\n",
       "<Gene-commentary_version>2</Gene-commentary_version>\n",
       "<Gene-commentary_seqs>\n",
       "<Seq-loc>\n",
       "<Seq-loc_int>\n",
       "<Seq-interval>\n",
       "<Seq-interval_from>149333028</Seq-interval_from>\n",
       "<Seq-interval_to>149662872</Seq-interval_to>\n",
       "<Seq-interval_strand>\n",
       "<Na-strand value=\"plus\"/>\n",
       "</Seq-interval_strand>\n",
       "<Seq-interval_id>\n",
       "<Seq-id>\n",
       "<Seq-id_gi>528476637</Seq-id_gi>\n",
       "</Seq-id>\n",
       "</Seq-interval_id>\n",
       "</Seq-interval>\n",
       "</Seq-loc_int>\n",
       "</Seq-loc>\n",
       "</Gene-commentary_seqs>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_heading>Representative Expression</Gene-commentary_heading>\n",
       "<Gene-commentary_comment>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_label>Text Summary</Gene-commentary_label>\n",
       "<Gene-commentary_text>Broad expression in thyroid (RPKM 10.7), adrenal (RPKM 4.5) and 21 other tissues</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_label>Tissue List</Gene-commentary_label>\n",
       "<Gene-commentary_text>thyroid; adrenal; ovary; brain; urinary bladder; testis; endometrium; gall bladder; spleen; skin; prostate; lung; fat; esophagus; placenta; lymph node; appendix; colon; kidney; stomach; pancreas; small intestine; heart; </Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "<Gene-commentary>\n",
       "<Gene-commentary_type value=\"comment\">254</Gene-commentary_type>\n",
       "<Gene-commentary_label>Category</Gene-commentary_label>\n",
       "<Gene-commentary_text>Broad expression</Gene-commentary_text>\n",
       "</Gene-commentary>\n",
       "</Gene-commentary_comment>\n",
       "</Gene-commentary>\n",
       "</Entrezgene_comments>\n",
       "<Entrezgene_unique-keys>\n",
       "<Dbtag>\n",
       "<Dbtag_db>HGNC</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_str>HGNC:17223</Object-id_str>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "<Dbtag>\n",
       "<Dbtag_db>LocusID</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>10090</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "<Dbtag>\n",
       "<Dbtag_db>MIM</Dbtag_db>\n",
       "<Dbtag_tag>\n",
       "<Object-id>\n",
       "<Object-id_id>610752</Object-id_id>\n",
       "</Object-id>\n",
       "</Dbtag_tag>\n",
       "</Dbtag>\n",
       "</Entrezgene_unique-keys>\n",
       "<Entrezgene_xtra-index-terms>\n",
       "<Entrezgene_xtra-index-terms_E>LOC10090</Entrezgene_xtra-index-terms_E>\n",
       "</Entrezgene_xtra-index-terms>\n",
       "<Entrezgene_xtra-properties>\n",
       "<Xtra-Terms>\n",
       "<Xtra-Terms_tag>PROP</Xtra-Terms_tag>\n",
       "<Xtra-Terms_value>phenotype</Xtra-Terms_value>\n",
       "</Xtra-Terms>\n",
       "</Entrezgene_xtra-properties>\n",
       "</Entrezgene>\n",
       "</Entrezgene-Set>"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h1n1_gene_id = ncbi.api_soup('elink',{\"db\":\"taxonomy\",\"dbfrom\":\"gene\",\"id\":h1n1_id})\n",
    "h1n1_gene_id = h1n1_gene_id.LinkSetDb.Id.getText()\n",
    "\n",
    "ncbi.api_soup(\"efetch\",{\"db\":\"gene\",\"id\":h1n1_gene_id,\"retmode\":\"xml\"})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also use the EGQuery utility to see how many records in all Entrez databases refer to the formal term \"Influenza A virus subtype H1N1\" and compare that with how many mention the informal term \"H1N1\":"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Database Name</th>\n",
       "      <th>Records with Term \"Influenza A virus subtype H1N1\"</th>\n",
       "      <th>Records with Term \"H1N1\"</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pubmed</td>\n",
       "      <td>16901</td>\n",
       "      <td>22948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pmc</td>\n",
       "      <td>6917</td>\n",
       "      <td>63269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mesh</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>books</td>\n",
       "      <td>135</td>\n",
       "      <td>1515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pubmedhealth</td>\n",
       "      <td>Error</td>\n",
       "      <td>Error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>omim</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ncbisearch</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>nuccore</td>\n",
       "      <td>4213</td>\n",
       "      <td>226250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>nucgss</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>nucest</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>protein</td>\n",
       "      <td>1651</td>\n",
       "      <td>307623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>genome</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>structure</td>\n",
       "      <td>28</td>\n",
       "      <td>451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>taxonomy</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>snp</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>dbvar</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>gene</td>\n",
       "      <td>22</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>sra</td>\n",
       "      <td>2077</td>\n",
       "      <td>8805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>biosystems</td>\n",
       "      <td>Error</td>\n",
       "      <td>Error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>unigene</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>cdd</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>clone</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>popset</td>\n",
       "      <td>593</td>\n",
       "      <td>628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>geoprofiles</td>\n",
       "      <td>0</td>\n",
       "      <td>279625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>gds</td>\n",
       "      <td>1042</td>\n",
       "      <td>11865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>homologene</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>pccompound</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>pcsubstance</td>\n",
       "      <td>0</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>pcassay</td>\n",
       "      <td>183</td>\n",
       "      <td>3405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>nlmcatalog</td>\n",
       "      <td>97</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>probe</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>gap</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>proteinclusters</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>bioproject</td>\n",
       "      <td>6</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>biosample</td>\n",
       "      <td>913</td>\n",
       "      <td>6801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>biocollections</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Database Name Records with Term \"Influenza A virus subtype H1N1\"  \\\n",
       "0            pubmed                                              16901   \n",
       "1               pmc                                               6917   \n",
       "2              mesh                                                  1   \n",
       "3             books                                                135   \n",
       "4      pubmedhealth                                              Error   \n",
       "5              omim                                                  0   \n",
       "6        ncbisearch                                                  6   \n",
       "7           nuccore                                               4213   \n",
       "8            nucgss                                                  0   \n",
       "9            nucest                                                  0   \n",
       "10          protein                                               1651   \n",
       "11           genome                                                  0   \n",
       "12        structure                                                 28   \n",
       "13         taxonomy                                                  1   \n",
       "14              snp                                                  0   \n",
       "15            dbvar                                                  0   \n",
       "16             gene                                                 22   \n",
       "17              sra                                               2077   \n",
       "18       biosystems                                              Error   \n",
       "19          unigene                                                  0   \n",
       "20              cdd                                                  0   \n",
       "21            clone                                                  0   \n",
       "22           popset                                                593   \n",
       "23      geoprofiles                                                  0   \n",
       "24              gds                                               1042   \n",
       "25       homologene                                                  0   \n",
       "26       pccompound                                                  0   \n",
       "27      pcsubstance                                                  0   \n",
       "28          pcassay                                                183   \n",
       "29       nlmcatalog                                                 97   \n",
       "30            probe                                                  0   \n",
       "31              gap                                                  0   \n",
       "32  proteinclusters                                                  0   \n",
       "33       bioproject                                                  6   \n",
       "34        biosample                                                913   \n",
       "35   biocollections                                                  5   \n",
       "\n",
       "   Records with Term \"H1N1\"  \n",
       "0                     22948  \n",
       "1                     63269  \n",
       "2                        17  \n",
       "3                      1515  \n",
       "4                     Error  \n",
       "5                        13  \n",
       "6                         9  \n",
       "7                    226250  \n",
       "8                         0  \n",
       "9                         0  \n",
       "10                   307623  \n",
       "11                        1  \n",
       "12                      451  \n",
       "13                        1  \n",
       "14                        0  \n",
       "15                        0  \n",
       "16                      158  \n",
       "17                     8805  \n",
       "18                    Error  \n",
       "19                        0  \n",
       "20                        0  \n",
       "21                        0  \n",
       "22                      628  \n",
       "23                   279625  \n",
       "24                    11865  \n",
       "25                        0  \n",
       "26                        3  \n",
       "27                      489  \n",
       "28                     3405  \n",
       "29                      158  \n",
       "30                        0  \n",
       "31                       98  \n",
       "32                        0  \n",
       "33                      304  \n",
       "34                     6801  \n",
       "35                        0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "query_list = ['H1N1', 'Influenza A virus subtype H1N1']\n",
    "dfs = []\n",
    "\n",
    "for each in query_list:\n",
    "\n",
    "    query = ncbi.api_soup('egquery', {\"term\": each})\n",
    "    query_tag = query.eGQueryResult\n",
    "\n",
    "    query_name = query_tag.find_all('DbName')  \n",
    "    query_count = query_tag.find_all('Count')  \n",
    "\n",
    "    query_data = []\n",
    "    \n",
    "    # Loop to store the data in a list named 'lineage_data'\n",
    "    for i in range(0, len(query_name)):\n",
    "        query_rows = [query_name[i].get_text(), query_count[i].get_text()]\n",
    "        query_data.append(query_rows)\n",
    "    \n",
    "    # Converting the list into dataframe\n",
    "    query_df = pd.DataFrame(query_data, columns=['Database Name', 'Records with Term \"' + each + '\"'], dtype = str)\n",
    "\n",
    "    dfs.append(query_df)\n",
    "\n",
    "df = pd.merge(right=dfs[0], left=dfs[1])\n",
    "\n",
    "display(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It should be noted that the taxonomy database only has 1 record that refers to either, and that they are the same record. This shows the core mapping function of the taxonomy database -- all other Entrez linkages map back to the same taxonomy.\n",
    "\n",
    "Additional tests reveal that NCBI is strict about terms used for implementation of the ESearch utility. For instance, searching for the term 'Influenza Virus A' reveals that the phrase is not found in the NCBI Taxonomy:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<ErrorList>\n",
      " <PhraseNotFound>\n",
      "  Influenza\n",
      " </PhraseNotFound>\n",
      " <PhraseNotFound>\n",
      "  Virus\n",
      " </PhraseNotFound>\n",
      " <PhraseNotFound>\n",
      "  A\n",
      " </PhraseNotFound>\n",
      "</ErrorList>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "correction = ncbi.api_soup('esearch', {\"term\": 'Influenza Virus A', \"db\":\"taxonomy\"})\n",
    "print(correction.ErrorList.prettify())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "However, the terms 'Influenzavirus A' and 'Influenza A virus' can both be found using ESearch:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<IdList>\n",
      " <Id>\n",
      "  197911\n",
      " </Id>\n",
      "</IdList>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "id_response = ncbi.api_soup('esearch', {\"term\": 'Influenzavirus A', \"db\":\"taxonomy\"})\n",
    "print(id_response.IdList.prettify())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<IdList>\n",
      " <Id>\n",
      "  11320\n",
      " </Id>\n",
      "</IdList>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "id_response = ncbi.api_soup('esearch', {\"term\": 'Influenza A virus', \"db\":\"taxonomy\"})\n",
    "print(id_response.IdList.prettify())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "While at first glance, it may appear that ESearch doesn't handle synonyms as expected. However, these tests actually reveal that NCBI has a highly controlled taxonomy which means we can rely upon it as a data source. The reason 'Influenza Virus A' is not supported by ESearch is because it could be a misnomer for either the genus (Alphainfluenzavirus, informal equivalent is \"Influenzavirus A\") or the species (Influenza A Virus). \n",
    "\n",
    "The ESpell utility provided by NCBI offers a way to view possible spellings for a particular organism to help reduce the risk of using invalid search terms in ESearch, as demonstrated below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<CorrectedQuery>\n",
      " influenzavirus a\n",
      "</CorrectedQuery>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "correction = ncbi.api_soup('espell', {\"term\": 'Influenzavirus A', \"db\":\"taxonomy\"})\n",
    "print(correction.CorrectedQuery.prettify())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "However, ESpell does not function perfectly. Human work is required for review of many synonyms in NCBI, as there are unpredictable mistakes which could pose issues to data integrity. \n",
    "\n",
    "For instance, searching \"Influenza Virus B\" (expected output: \"Influenzavirus B\") returns \"Influenza Virus A\":"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<CorrectedQuery>\n",
      " influenza virus a\n",
      "</CorrectedQuery>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "correction = ncbi.api_soup('espell', {\"term\": 'Influenza Virus B', \"db\":\"taxonomy\"})\n",
    "print(correction.CorrectedQuery.prettify())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "d7315c49e695b70a80fa10d86a25c9d146976f258accde30ac4a8950a94324f1"
  },
  "kernelspec": {
   "display_name": "Python 3.9.12 ('env': venv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
